{
 "metadata": {
  "name": "",
  "signature": "sha256:a82ce9a9e669ea56ee134ce86edd312b1b26d2e7a09b34a23167b04828597119"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "# Development using test dataset"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "__grinder run__\n",
      "\n",
      "@ system76-server:~/notebook/grinderSIP/t/example\n",
      "\n",
      "__with abundance profile__\n",
      "\n",
      "~~~\n",
      "grinder -reference_file genome3.fna -fr 515Fm-927Rm.fna \\\n",
      "    -rd 215 normal 50 -tr 1000 -length_bias 0 -unidirectional 1 \\\n",
      "    -af abnd_prof.txt\n",
      "~~~    \n",
      "\n",
      "__with abundance model__\n",
      "\n",
      "~~~\n",
      "grinder -reference_file genome3.fna -fr 515Fm-927Rm.fna -am exponential \\\n",
      "    -rd 215 normal 50 -tr 1000 -length_bias 0 -unidirectional 1 \n",
      "~~~\n",
      "\n",
      "__with multiple samples__\n",
      "\n",
      "~~~\n",
      "grinder -reference_file genome3.fna -fr 515Fm-927Rm.fna -am exponential \\\n",
      "-rd 215 normal 50 -tr 1000 -length_bias 0 -unidirectional 1 -num_libraries 2 \\\n",
      "-shared_perc 100\n",
      "    \n",
      "~~~\n",
      "\n",
      "                "
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "## testing script with 3 genome\n",
      "\n",
      "@ system76-server:~/notebook/grinderSIP/dev\n",
      "\n",
      "### simulating isotope incorporation (max of 100%)\n",
      "\n",
      "    $ perl ../bin/makeIncorp_phylo.pl -r grinder-ranks.txt -t genome3.nwk > genome3_inc100.txt\n",
      "\n",
      "### simulating isotope incorporation (max of 50%)\n",
      "\n",
      "    $ perl ../bin/makeIncorp_phylo.pl -r grinder-ranks.txt -t genome3.nwk -range 0-50 > genome3_inc50.txt\n",
      "\n",
      "### grinderSIP.pl\n",
      "\n",
      "CHANGES to grindeSIP.pl:\n",
      "Fragments written to each fraction file, then grinder is run on each \n",
      "    \n",
      "* fragments made by genome fragmentation:\n",
      " * genomes fragmented based on determined or defined relative abundance (RA)\n",
      "  * fragmentation size defined by size distribution\n",
      "  * fragment start defined by poisson? or uniform? distribution\n",
      " * total number of genome copies (TNGC) defined\n",
      " * TNGC x RA(each taxon) \n",
      " * fragments distributed into fractions based on fragment GC\n",
      "  * naming issues?\n",
      "   * may need to rename by taxon\n",
      "* foreach fraction:\n",
      " * wrapper to run grinder\n",
      "  * need option file (options based on grinder)\n",
      " \n",
      "\n",
      "    $ perl ../bin/grinderSIP.pl -g genome3.fna -r grinder-reads.fa -inc genome3_inc100.txt | less\n",
      "\n",
      "    \n",
      "\n",
      "__checking that all reads written__\n",
      "\n",
      "    $ nseq readsByFrac/isoIncorp/*fasta | perl -ne 'BEGIN{$sum=0} /.+:(\\d+)/; $sum+=$1; END{print $sum,\"\\n\"}'\n",
      "        * yes, all written\n",
      "\n",
      "\n",
      "    \n"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "# Larger dataset of 10 genomes\n",
      "\n",
      "## Making the dataset\n",
      "\n",
      "@ system76-server:~/notebook/grinderSIP/dev/bac_genome10\n",
      "\n",
      "__Randomly selected 10 genomes__\n",
      "* Selecting from  ../../../deltaGC_notes/data/genome_data/prok-bac-genomes_Ngaps\n",
      "~~~\n",
      "Bacillus_cereus_F837_76.fasta\n",
      "Cyanobium_gracile_PCC_6307.fasta\n",
      "Escherichia_coli_APEC_O78.fasta\n",
      "Geobacter_lovleyi_SZ.fasta\n",
      "Idiomarina_loihiensis_GSL_199.fasta\n",
      "Leuconostoc_citreum_KM20.fasta\n",
      "Micrococcus_luteus_NCTC_2665.fasta\n",
      "Nitrosomonas_europaea_ATCC_19718.fasta\n",
      "Roseobacter_litoralis_Och_149.fasta\n",
      "Xanthomonas_axonopodis_Xac29-1.fasta\n",
      "~~~\n",
      "\n",
      "Getting genomes from collection\n",
      "~~~\n",
      "$ printf \"Bacillus_cereus_F837_76.fasta\\nCyanobium_gracile_PCC_6307.fasta\\nEscherichia_coli_APEC_O78.fasta\\nGeobacter_lovleyi_SZ.fasta\\nIdiomarina_loihiensis_GSL_199.fasta\\nLeuconostoc_citreum_KM20.fasta\\nMicrococcus_luteus_NCTC_2665.fasta\\nNitrosomonas_europaea_ATCC_19718.fasta\\nRoseobacter_litoralis_Och_149.fasta\\nXanthomonas_axonopodis_Xac29-1.fasta\" | xargs -n 1 -I % bash -c \"grep -c '>' ~/notebook/deltaGC_notes/data/genome_data/prok-bac-genomes_Ngaps/%\"\n",
      "    * all have 1 genome\n",
      "\n",
      "$ printf \"Bacillus_cereus_F837_76.fasta\\nCyanobium_gracile_PCC_6307.fasta\\nEscherichia_coli_APEC_O78.fasta\\nGeobacter_lovleyi_SZ.fasta\\nIdiomarina_loihiensis_GSL_199.fasta\\nLeuconostoc_citreum_KM20.fasta\\nMicrococcus_luteus_NCTC_2665.fasta\\nNitrosomonas_europaea_ATCC_19718.fasta\\nRoseobacter_litoralis_Och_149.fasta\\nXanthomonas_axonopodis_Xac29-1.fasta\" | xargs -n 1 -I % bash -c \"ln -s  ~/notebook/deltaGC_notes/data/genome_data/prok-bac-genomes_Ngaps/% genomes/\"\n",
      "    * all genomes symlinked\n",
      "\n",
      "~~~\n",
      "\n",
      "Combining and editing names (using file names)\n",
      "      \n",
      "    $ find genomes/ -name \"*fasta\" | xargs -n 1 -I % bash -c \"perl -p -e 'BEGIN{\\$in=\\$ARGV[0]; \\$in=~s/.+\\/|\\.[^.]+\\$//g} s/^>.+/>\\$in/' % \" > genome10.fna\n",
      "\n",
      "    \n",
      "\n",
      "## Grinder simulation (10000 reads)\n",
      "\n",
      "__exponential abundance model__\n",
      "\n",
      "~~~\n",
      "grinder -reference_file genome10.fna -fr ../515Fm-927Rm.fna -am exponential \\\n",
      "    -rd 215 normal 50 -tr 10000 -length_bias 0 -unidirectional 1 -base_name genome10\n",
      "~~~\n",
      "\n",
      "## Making a tree of the genomes\n",
      "\n",
      "16S rRNA tree\n",
      "\n",
      "### Using RNAmmer to get all 16S rRNA genes from each genome\n",
      "\n",
      "Getting already created rnammer output files\n",
      "\n",
      "    $ mkdir rnammer\n",
      "    \n",
      "    $ find genomes/ -name \"*fasta\" | perl -pe 's/.+\\///; s/\\.fasta/_rrn.fna/' | xargs -n 1 -I % bash -c \"ln -s /var/seq_data/ncbi_db/genome/rnammer/bacteria_rrn/% rnammer/\"\n",
      "    \n",
      "Filtering out poor hits\n",
      "    \n",
      "    $ find rnammer/ -name \"*_rrn.fna\" | perl -pe 's/(.+)(\\.fna)/$1$2\\n$1\\_s2k$2/g' | parallel -q -N 2 bash -c 'perl ~/dev/NY_misc/bin/rnammer_filt.py -s 2000 - <{1} > {2}'\n",
      "        * score >= 2000\n",
      "\n",
      "### Making consensus 16S sequence for each \n",
      "\n",
      "    $ find rnammer/ -name \"*_s2k.fna\" | perl -pe 's/(.+)(\\.fna)/$1$2\\n$1\\_con$2/g' | parallel -q -N 2 bash -c \"~/dev/NY_misc_perl/bin/consensus_seq.pl -i - -t 30 -n {1} < {1} > {2}\"\n",
      "        * low threshold used to minimize ambiguous\n",
      "\n",
      "Editing names & combining ('?' -> 'N')\n",
      "\n",
      "    $ find rnammer/ -name \"*con.fna\" | xargs -n 1 -I % bash -c \"perl -pe 's/.+\\//>/;s/_rrn_s2k\\.fna//; s/\\?/N/g' %\" > genome10_rrn-con.fna\n",
      "\n",
      "### Aligning with mafft\n",
      "\n",
      "    $ mafft-linsi --thread 20 genome10_rrn-con.fna > genome10_rrn-con_aln.fna\n",
      "\n",
      "### Making ML tree with RAxML\n",
      "\n",
      "@ system76-server:~/notebook/grinderSIP/dev/bac_genome10/raxml\n",
      "\n",
      "    $ alignIO.py ../genome10_rrn-con_aln.fna genome10_rrn-con_aln.phy\n",
      "\n",
      "    $ raxmlHPC-PTHREADS-AVX -f a -x 0318 -p 0911 -# 100 -m GTRGAMMA -s genome10_rrn-con_aln.phy -n genome10_rrn-con_ML -T 20\n",
      "\n",
      "    $ mv RAxML_bipartitions.genome10_rrn-con_ML genome10_rrn-con_ML.nwk\n",
      "\n",
      "\n",
      "## Making incorp file (min-max incorp: 0-100)\n",
      "\n",
      "@ system76-server:~/notebook/grinderSIP/dev/bac_genome10\n",
      "\n",
      "__Incorp: min0, max100; weight=0.5__\n",
      "\n",
      "    $ perl ../../bin/makeIncorp_phylo.pl -r genome10-ranks.txt -t raxml/genome10_rrn-con_ML.nwk > genome10_inc0-100_w0.5.txt\n",
      "\n",
      "__Incorp: min0, max100; weight=1__\n",
      "\n",
      "    $ perl ../../bin/makeIncorp_phylo.pl -r genome10-ranks.txt -t raxml/genome10_rrn-con_ML.nwk -w 1 > genome10_inc0-100_w1.txt\n",
      "\n",
      "__Incorp: min0, max100; weight=0__\n",
      "\n",
      "    $ perl ../../bin/makeIncorp_phylo.pl -r genome10-ranks.txt -t raxml/genome10_rrn-con_ML.nwk -w 0 > genome10_inc0-100_w0.txt\n",
      "    \n",
      "## running grinderSIP.pl\n",
      "\n",
      "    $ perl ../../bin/grinderSIP.pl -g genome10.fna -r genome10-reads.fa -inc genome10_inc0-100_w1.txt -out readsByFrac_inc0-100_w1 | less\n",
      "\n",
      "__checking that all reads written__\n",
      "\n",
      "    $ nseq readsByFrac_inc0-100_w1/isoIncorp/*fasta | perl -ne 'BEGIN{$sum=0} /.+:(\\d+)/; $sum+=$1; END{print $sum,\"\\n\"}'\n",
      "        * yes, all written\n",
      "\n"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Checking on character trait evolution through ordering by tree\n",
      "import os\n",
      "import glob\n",
      "\n",
      "pwd = '/home/nick/notebook/grinderSIP/dev/bac_genome10/'\n",
      "\n",
      "## getting tree labels\n",
      "cmd = 'nw_labels -I ' + pwd + 'raxml/genome10_rrn-con_ML.nwk'\n",
      "ret = os.popen(cmd, 'r')\n",
      "labels = [i.strip() for i in ret]\n",
      "\n",
      "## ordering & printing tables\n",
      "def order_table(csv_file, order_list):\n",
      "    fh = open(csv_file, 'r')\n",
      "    tbl = [i.rstrip().split('\\t') for i in fh]\n",
      "    tbl = dict(tbl)\n",
      "    for i in labels:\n",
      "        print '\\t'.join([csv_file, i, tbl[i]])\n",
      "        \n",
      "# writing dict(tbl) in order of labels\n",
      "files = glob.glob(pwd + '*_w*.txt')\n",
      "for f in files:\n",
      "    order_table(f, labels)\n",
      "\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.txt\tEscherichia_coli_APEC_O78\t89.57\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.txt\tIdiomarina_loihiensis_GSL_199\t69.70\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.txt\tXanthomonas_axonopodis_Xac29-1\t57.66\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.txt\tNitrosomonas_europaea_ATCC_19718\t8.87\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.txt\tBacillus_cereus_F837_76\t100.00\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.txt\tGeobacter_lovleyi_SZ\t70.05\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.txt\tLeuconostoc_citreum_KM20\t27.61\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.txt\tMicrococcus_luteus_NCTC_2665\t0.00\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.txt\tCyanobium_gracile_PCC_6307\t31.37\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.txt\tRoseobacter_litoralis_Och_149\t87.14\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w1.txt\tEscherichia_coli_APEC_O78\t41.00\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w1.txt\tIdiomarina_loihiensis_GSL_199\t49.54\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w1.txt\tXanthomonas_axonopodis_Xac29-1\t22.18\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w1.txt\tNitrosomonas_europaea_ATCC_19718\t26.64\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w1.txt\tBacillus_cereus_F837_76\t100.00\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w1.txt\tGeobacter_lovleyi_SZ\t62.53\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w1.txt\tLeuconostoc_citreum_KM20\t53.11\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w1.txt\tMicrococcus_luteus_NCTC_2665\t54.00\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w1.txt\tCyanobium_gracile_PCC_6307\t24.64\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w1.txt\tRoseobacter_litoralis_Och_149\t0.00\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.5.txt\tEscherichia_coli_APEC_O78\t35.12\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.5.txt\tIdiomarina_loihiensis_GSL_199\t38.34\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.5.txt\tXanthomonas_axonopodis_Xac29-1\t54.47\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.5.txt\tNitrosomonas_europaea_ATCC_19718\t81.63\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.5.txt\tBacillus_cereus_F837_76\t66.40\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.5.txt\tGeobacter_lovleyi_SZ\t3.79\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.5.txt\tLeuconostoc_citreum_KM20\t100.00\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.5.txt\tMicrococcus_luteus_NCTC_2665\t0.00\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.5.txt\tCyanobium_gracile_PCC_6307\t23.78\n",
        "/home/nick/notebook/grinderSIP/dev/bac_genome10/genome10_inc0-100_w0.5.txt\tRoseobacter_litoralis_Och_149\t50.25\n"
       ]
      }
     ],
     "prompt_number": 50
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": []
    }
   ],
   "metadata": {}
  }
 ]
}