{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Major questions:\n",
    "\n",
    "1. How specific & sensitive is the our current HR-SIP analytical method?\n",
    "\n",
    "1. What are some major factors affecting accuracy of our current HR-SIP analytical method?\n",
    "\n",
    "1. How can we increase the accuracy of HR-SIP?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Simulation methodology validation\n",
    "\n",
    "\n",
    "## Diagram of simulation methodology\n",
    "\n",
    "* Figure 1\n",
    "\n",
    "## Supplemental example plots of key aspects in simulation\n",
    "\n",
    "### Genomic fragment simulation\n",
    "\n",
    "* Sup Figure:\n",
    "  * 3 taxa of differing GC\n",
    "  * Fragment length distributions\n",
    "    * 'realistic' = skewed normal\n",
    "    * 'large' = ~50 kb\n",
    "    * 'small' = ~1-2 kb\n",
    "  * faceted heatmap:\n",
    "    * taxa ~ (fragment length distribution)\n",
    "\n",
    "### Diffusion calculations\n",
    "\n",
    "* Sup Figure:\n",
    "  * 3 taxa of differing GC (same as last)\n",
    "  * histograms:\n",
    "    * BD distribution for pre & post diffusion \n",
    "  * Shows effect of diffusion & how it relates to fragment length\n",
    "\n",
    "### Real HR-SIP dataset validation\n",
    "\n",
    "* Dataset: Ashley's priming experiment\n",
    "* Select OTUs with close genome representative\n",
    "  * BLASTn of OTU 16S vs genome 16S blast db\n",
    "* Simulation of just genomes of interest\n",
    "* Simulation of all genomes \n",
    "* Plot abundance distributions: real vs simulated\n",
    "\n",
    "### Mock community validation?\n",
    "\n",
    "* __Goal:__ validate the simulation model vs a ground-truth real dataset\n",
    "* __Is this necessary???__\n",
    "* 10-20 isolates\n",
    "* 'Control' community:\n",
    "  * All DNA is 12C\n",
    "* 'Treatment' community:\n",
    "  * 2 to 4 isolates grown with 13C (~100% 13C incorporation)\n",
    "* 2 gradients (full HR-SIP pipeline)\n",
    "  * add to MiSeq run\n",
    "* __Marquessa project???__\n",
    "* Downsides:\n",
    "  * Costly in time and money\n",
    "  * Would takes months to complete\n",
    "\n",
    "* (Sup) Figure:\n",
    "  * Taxon abundances of real & simulated communities\n",
    "* Table:\n",
    "  * Accuracy of detecting incorporators (simulation vs real dataset)\n",
    "\n",
    "\n",
    "## Example figure: absolute & relative taxon abundances\n",
    "\n",
    "* All taxa simulated (n=1210); just top 100 plotted\n",
    "\n",
    "* Figure:\n",
    "  * 1 community\n",
    "  * faceted:\n",
    "    * (absolute abundance or relative abundance) ~ (total community or subsampled community)  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1) How specific & sensitive is the our current HR-SIP analytical method?\n",
    "# 2) What some major factors affecting accuracy of identifying incorporators?\n",
    "\n",
    "\n",
    "## Accuracy as a function of isotope incorporation\n",
    "\n",
    "### Variable parameters:\n",
    "\n",
    "* % isotope incorporation\n",
    "  * 0, 25, 50, 100\n",
    "* % taxa that incorporate\n",
    "  * __1, 5, 10, 25, 50__\n",
    "* n-reps (stocastic: taxon abundances & which incorporate)\n",
    "  * 10 (20?)\n",
    "* total simulations\n",
    "  * `4 * 3 * 10 = 120`\n",
    "  * `time = 120 * 1.6 [hr] = 192 [hr] = 8 [days]`\n",
    "\n",
    "### Set parameters:\n",
    "\n",
    "* Community rank-abundance:\n",
    "  * lognormal\n",
    "* Total community abundance:\n",
    "  * 1e9\n",
    "* 'heavy' BD range:\n",
    "  * 1.71 - 1.75\n",
    "\n",
    "### Analysis:\n",
    "\n",
    "* Figure:\n",
    "  * faceted boxplot\n",
    "    * x-axis = percent isotope incorporation\n",
    "    * color/group = percent of taxa to incorporate\n",
    "    * facet = sensitivity/specificity (and balanced accuracy?)\n",
    "\n",
    "\n",
    "### Possible extensions:\n",
    "\n",
    "* % isotope incorporation\n",
    "  * normal distribution, varying standard deviation\n",
    "     * mean = 50\n",
    "     * sd = (1, 10, 25)\n",
    "  * uniform distribution\n",
    "\n",
    "\n",
    "## Accuracy as a function of community evenness and isotope incorporation\n",
    "\n",
    "### Variable parameters:\n",
    "\n",
    "* Community rank-abundance:\n",
    "  * lognormal abundance distribution\n",
    "    * very uneven\n",
    "    * moderately uneven\n",
    "    * slightly uneven\n",
    "* % incorp\n",
    "  * 0:100:5\n",
    "* XCommunity richness:\n",
    "  * X100 taxa\n",
    "  * X300 taxa\n",
    "  * X900 taxa\n",
    "* n-reps (stocastic: taxon abundances & which incorporate)\n",
    "  * 20\n",
    "* total simulations\n",
    "\n",
    "\n",
    "### Set parameters:\n",
    "\n",
    "* % taxa that incorporate\n",
    "  * __10%?__\n",
    "* Total community abundance:\n",
    "  * 1e9\n",
    "* 'heavy' BD range:\n",
    "  * 1.71 - 1.75\n",
    "\n",
    "\n",
    "### Analysis:\n",
    "\n",
    "* Figure:\n",
    "  * pointrange (percent_incorp ~ accuracy_value; color by community evenness)\n",
    "\n",
    "\n",
    "\n",
    "## Accuracy as a function taxon abundance\n",
    "\n",
    "### Variable parameters:\n",
    "\n",
    "* Partition communities into 'dominant' and 'rare' subsets\n",
    "  * __dominant/rare cutoff: 1% relative abundance?__\n",
    "* Assess sensitivity & specificity for each subset\n",
    "* __Note:__ \n",
    "  * Much less time to perform if using data from past simulation run datasets\n",
    "\n",
    "### Analysis:\n",
    "\n",
    "* Sup Figures:\n",
    "  * Same as above simulation runs, but split into 'dominant' vs 'rare'\n",
    "\n",
    "\n",
    "## Accuracy as a function of BD window\n",
    "\n",
    "### Variable parameters:\n",
    "\n",
    "* __Note:__ \n",
    "  * Much less time to perform if using data from past simulation run datasets\n",
    "* 'Heavy' BD range:\n",
    "  * BD-min\n",
    "    * range = [1.67:1.77:0.02]\n",
    "  * BD-max\n",
    "    * BD-min + 0.04\n",
    "  * product: BD-min ~ BD-max\n",
    "\n",
    "### Analysis:\n",
    "\n",
    "* Figure:\n",
    "  * Plot showing sensitivities/specificities for each 'heavy' BD range\n",
    "* Figure:\n",
    "  * Taxon abundances of true positives & false negatives\n",
    "  * __Why__: displays the issue with selecting one 'heavy' BD window"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# How can we increase the accuracy of our method?\n",
    "\n",
    "## Multiple 'heavy' fraction windows\n",
    "\n",
    "* General sliding window approach\n",
    "\n",
    "* Binning OTUs by log2fc ~ BD\n",
    "\n",
    "* Extrapolating GC for each OTU, with 2-3 different 'heavy' BD windows\n",
    "\n",
    "## Time series analysis\n",
    "\n",
    "* MetagenomeSeq method"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
