{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Description:\n",
    "\n",
    "Notebooks in this directory describe isopycnicGenomes workflow where probability density functions fit to the fragment BD distribution simulated for each genome.\n",
    "\n",
    "* PDF = probability density function\n",
    "* KDE = kernel density estimation\n",
    "* Implementing some of the workflow in python (except simulating the initial communities, which is done with grinder).\n",
    "\n",
    "\n",
    "___\n",
    "## Experiments for testing incorporation using the modeling tool\n",
    "\n",
    "### atom % 13C\n",
    "\n",
    "* Simulates isotope dilution or short incubations\n",
    "  * Method  \n",
    "    * incorporation % treatments: 10, 20, 40, 60, 80, 100%\n",
    "    * Total treatments: 6\n",
    "    * Title: perc_incorp\n",
    "    \n",
    "\n",
    "### Variable incorporation\n",
    "\n",
    "* Simulate variable percentages of incorporation and see which taxa are ID'ed\n",
    "* A fuzzy cutoff of percent incorp that is detectable?\n",
    "  * Method\n",
    "    * 25% of taxa incorporate\n",
    "    * incorporation % distribution: \n",
    "      * uniform: 0-100, 50-100, 0-50\n",
    "      * normal: mean=50, sd=c(0,10,20,30)\n",
    " \n",
    "\n",
    "### Few incorporators vs many (e.g., ~10% vs ~50%) \n",
    "\n",
    "* Simulates communities of specialists vs generalists\n",
    "  * Or a more recalcitrant vs labile substrate\n",
    "  * Method\n",
    "    * factorial\n",
    "    * % incorporator treatments: 5, 10, 20, 40, 60%\n",
    "    * % incorporation: 20, 50, or 100%\n",
    "    * Total treatments: 5 x 3 = 15\n",
    "    * Title: perc_tax-incorp\n",
    "\n",
    "\n",
    "### Community evenness\n",
    "\n",
    "* differing levels of evenness\n",
    "  * Method\n",
    "    * evenness levels: uniform, log (differing params)\n",
    "\n",
    "\n",
    "### Community richness\n",
    "\n",
    "* soil vs simpler habitat\n",
    "  * Method\n",
    "    * N-taxa treatments: 100, 500, 1000 taxa    \n",
    "\n",
    "\n",
    "### Relative abundance of incorporators\n",
    "\n",
    "* dominant vs rare responders\n",
    "  * Could just do a post-hoc analysis on which taxa were detected by DESeq2\n",
    "\n",
    "\n",
    "### split populations: active vs dead/dormant \n",
    "\n",
    "* only some individuals incorporate isotope\n",
    "  * Method\n",
    "    * 25% of taxa incorporate\n",
    "    * factorial\n",
    "    * % split populations: 10, 50, 100%\n",
    "    * incorporation % treatments: 10, 50, 100%\n",
    "\n",
    "### split populations: full incorp vs partial\n",
    "\n",
    "* only some individuals incorporate isotope\n",
    "  * Method\n",
    "    * 25% of taxa incorporate\n",
    "    * factorial\n",
    "    * % split populations: 100%\n",
    "    * all in 1 sub-population incorporate 100%\n",
    "    * partial incorporation % treatments: 10, 20, 40, 60, 80, 100%\n",
    "\n",
    "    \n",
    "### Differences in rank-abundance between control and treatment\n",
    "\n",
    "* differing numbers of taxa with varying rank-abundances in the community\n",
    "  * Method\n",
    "    * Simulate communities with different levels of shared rank-abundances\n",
    "    * `SIPSim gradientComms --perm_perc`\n",
    "\n",
    "\n",
    "### Number of biological replicates\n",
    "\n",
    "* N-replicates to test: 1, 3, 5\n",
    "* Create 5 replicate control and treatment communities\n",
    "  * Test for incorp ID accuracy with differing number of replicates\n",
    "  * How to combine replicates?\n",
    "\n",
    "\n",
    "### Cross feeding\n",
    "\n",
    "* Nearly 100% incorporators with some partial 2ndary feeders\n",
    "    * Method\n",
    "      * Incorporators: normal dist, mean=90, sd=5\n",
    "      * Secondary feeders: normal dist, mean=c(10,20,30,40,50), sd=10\n",
    "\n",
    "\n",
    "### Phylogenetic signal vs random trait distribution\n",
    "\n",
    "* Random trait distributions coupled with high levels of microdiversity could cause 'split populations' and dilute the signal of incorporation \n",
    "* Clustering phylotypes (genomes) to produce course-grained phylotypes and see if the incorporation signal is lost\n",
    "  * Method\n",
    "    * 25% of taxa incorporate\n",
    "    * taxonomic clustering: genus, family, order, class, phylum\n",
    "    * % incorporation: 10, 20, 40, 60, 80, 100"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# General Workflow\n",
    "\n",
    "## Fragment GC distributions\n",
    "\n",
    "fragSim.py\n",
    "\n",
    "* input:\n",
    "  * genomes\n",
    "  * number of fragments per genome\n",
    "  * [primers]\n",
    "* workflow:\n",
    "  * Foreach genome:\n",
    "    * select amplicons (if needed)\n",
    "    * simulate fragments\n",
    "    * calculate GC content\n",
    "* output: table of fragment GC contents\n",
    "\n",
    " \n",
    "## Community abundance and incorporation\n",
    "\n",
    "### Simulate communities for each gradient fraction\n",
    "\n",
    "gradientComms.py\n",
    "\n",
    "* Goal:\n",
    "  * simulate abundance of taxa in >=1 community\n",
    "* Input:\n",
    "  * Genome list file\n",
    "  * Grinder options (profile file)\n",
    "* workflow:\n",
    "  * call modified Grinder with options\n",
    "* output:\n",
    "  * table of taxon abundances for each sample\n",
    "\n",
    "### Simulate isotope incorp\n",
    "\n",
    "isoIncorp.py\n",
    "\n",
    "* Goal:\n",
    "  * for each taxon in community, how isotope incorp distributed across individuals?\n",
    "    * example: taxonX incorp is normally distributed with mean of X and s.d. of Y\n",
    "    * user defines distribution and params\n",
    "    * can these distributions be 'evolved' across the tree (if provided)?\n",
    "      * brownian evolution of each param?\n",
    "    * 'Fragments' assumed to be pulled randomly from taxon population\n",
    "\n",
    "* Input:\n",
    "  * community file\n",
    "  * [isotope]\n",
    "  * [percent taxa with any incorp]\n",
    "  * [intra-taxon incorp: specify distribution and params]\n",
    "    * special: GMM: [GMM, weights, Norm1_params, Norm2_params]\n",
    "  * inter-taxon incorp:\n",
    "    * specify distributions of how intra-taxon params vary\n",
    "    * OR phylogeny: distribution params 'evolved' across tree\n",
    "* Workflow:\n",
    "  * Load community file\n",
    "  * if phylogeny:\n",
    "    * call R script to get which taxa incorporate based on tree (brownian motion)\n",
    "    * % of taxa defined by user\n",
    "  * else:\n",
    "    * random selection of taxa\n",
    "    * % of taxa defined by user\n",
    "  * For incorporators (randomly ordered!):\n",
    "    * if phylogeny:\n",
    "      * brownian motion evo. of intra-taxon incorp distribution params\n",
    "    * else:\n",
    "      * select intra-taxon incorp distribution params from inter-taxon param distribution\n",
    "  * For non-incorps:\n",
    "    * uniform distribution with min = 0 & max = 0\n",
    "* Output (incorp file):\n",
    "  * tab-delim table:\n",
    "    * sample, taxon, intra-taxon_incorp_disribution_type, distribution_params...\n",
    "\n",
    "\n",
    "## Creating OTU tables\n",
    "\n",
    "make_OTU_table.py\n",
    "\n",
    "* input:\n",
    "  * frag GC file\n",
    "  * community file\n",
    "  * incorp file\n",
    "  * [BD distribution [default: cauchy, params...]]\n",
    "  * [weight (multiplier) for incorp by abundance] \n",
    "\n",
    "* class creation:\n",
    "  * fragGC\n",
    "      * KDE fit to fragment G+C values for each taxon\n",
    "        * http://scikit-learn.org/stable/modules/density.html\n",
    "      * class:\n",
    "        * library : taxon : KDE fit\n",
    "      * simulate gradients (BD fractions)\n",
    "        * min-max based on theoretical min-max BD\n",
    "        * class: \n",
    "          * library : bin : otu : count\n",
    "          * function: place_in_bin(self, library, otu, BD)\n",
    "  * comm\n",
    "      * subclassed pandas dataframe?\n",
    "  * incorp file\n",
    "      * parse into distributions for each lib-taxon\n",
    "      * dict-like:\n",
    "          * library : taxon : pymix_distribution\n",
    "  * otu_table\n",
    "      * dict-like -- library : fraction : OTU : OTU_count\n",
    "\n",
    "* workflow (for each sample):\n",
    "  * load community file, incorp file, fragGC file\n",
    "  * Foreach sample:\n",
    "    * Foreach taxon (from comm class): \n",
    "      * sample GC value from KDE\n",
    "        * option: Log GC values from KDE\n",
    "      * select perc. incorp from intra-taxon incorp-distribution\n",
    "        * option: Log perc. incorp\n",
    "      * calculate BD (GC + perc. incorp)\n",
    "      * simulate gradient noise:\n",
    "        * select value (new BD) from (cauchy) distribution (mean = calculated BD)\n",
    "      * bin BD value in corresponding fraction\n",
    "        * save: sample => fraction => OTU\n",
    "            * save as pandas dataframe?\n",
    "  * write out OTU table (frag counts for taxon in fraction = OTU/taxon abundance)\n",
    "* output:\n",
    "  * OTU table:\n",
    "    * rows=OTU, columns=library-fraction\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Validation\n",
    "\n",
    "## Fragment simulation\n",
    "\n",
    "* E. coli\n",
    "* Plotting fragment length distributions under different conditions\n",
    "\n",
    "## KDE simulation\n",
    "\n",
    "* E. coli\n",
    "\n",
    "### Bandwidth\n",
    "\n",
    "* Testing effect bandwidth param\n",
    "* Plotting/comparing distributions made with differing params\n",
    "\n",
    "### Monte Carlo estimation\n",
    "\n",
    "* Testing effect of Monte Carlo estimation of distributions\n",
    "* Plotting/comparing distributions made with differing params\n",
    "\n",
    "## Community\n",
    "\n",
    "* Plotting community rank-abundances with different params\n",
    "* Calculating beta diversity with different params\n",
    "\n",
    "## Incorporation\n",
    "\n",
    "* E. coli\n",
    "* Plotting BD distribution with differing incorporation settings\n",
    "* Settings:\n",
    "  * Uniform incorporation\n",
    "  * Normal incorporation\n",
    "  * Split populations\n",
    "  \n",
    "## Gradient fractions\n",
    "\n",
    "* Plotting gradient fractions with differing params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***\n",
    "#~OLD~"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gaussian mixture models with pymix\n",
    "\n",
    "    n1 = mixture.NormalDistribution(-2,0.4)\n",
    "    n2 = mixture.NormalDistribution(2,0.6)\n",
    "    m = mixture.MixtureModel(2,[0.5,0.5], [n1,n2])\n",
    "    print m\n",
    "    m.sample()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fragment GC distributions\n",
    "\n",
    "fragSim.py\n",
    "\n",
    "* input:\n",
    "  * genomes\n",
    "  * number of fragments per genome\n",
    "  * [primers]\n",
    "* workflow:\n",
    "  * loading genomes as flat-file db\n",
    "  * loading primers as biopython seq records\n",
    "  * Foreach genome:\n",
    "    * find amplicons (if needed)\n",
    "      * load genome seq as Dseqrecord\n",
    "      * in-silico PCR with pydna\n",
    "    * simulate fragments\n",
    "      * get position from amplicon, simulate fragment around amplicon, pull out sequence from genome\n",
    "    * calculate GC content\n",
    "    * apply KDE to GC values (non-parametric approach)\n",
    "* output:\n",
    "  * pickle: genome => KDE_object\n",
    "        \n",
    "        \n",
    "### TODO:\n",
    "* in-silico PCR\n",
    "  * pydna (http://pydna.readthedocs.org/en/latest/pydna.html)\n",
    "* genome flat-file indexing\n",
    "  * pyfasta (https://pypi.python.org/pypi/pyfasta/)\n",
    "  \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***\n",
    "# ~OLD~\n",
    "***"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Specific Workflow\n",
    "\n",
    "## Simulate isotope incorporation\n",
    "* __makeIncorp_phylo.pl__\n",
    "* already complete\n",
    "\n",
    "## Define pre-isopycnic community \n",
    "* __preIsopycnicComm.pl__ \n",
    "* really, just utilizing Grinder\n",
    "* complete?\n",
    " \n",
    "## Simulate genome fragments and calculate BD\n",
    "* __simFrags.py__\n",
    "* simulation of certain number of fragment from each genome\n",
    "  * simulation of amplicon or metagenome fragments\n",
    "  * foreach fragment:\n",
    "    * calculate GC\n",
    "    * calculate BD based on isotope incorp of genome\n",
    "* Output:\n",
    "  * csv: sample, genome, GC, BD\n",
    "  \n",
    "## Create gradient communities \n",
    "* __simGradientComms.py__\n",
    "* Simulate communities for each gradient fraction\n",
    " * __simGradientComms.py__ \n",
    " * simulate fractions\n",
    "   * difference fractions for each gradient (each sample)\n",
    " * Foreach sample: Foreach genome: \n",
    "   * apply KDE to BD values (non-parametric approach)\n",
    "   * simulate number of 'fragments' needed to meet abs abund as defined in pre-isopynic community\n",
    "   * bin sim-frags by fraction\n",
    " * Output:\n",
    "   * csv: sample, genome, abs_abundance, rel_abundance\n",
    "  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***\n",
    "#~OLD~\n",
    "***"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## questions:\n",
    "\n",
    "* Can biopython simulate PCR (as done with bioperl)?\n",
    "\n",
    "## TODO:\n",
    "\n",
    "* simFractions.py\n",
    "  * simulate gradient fractions\n",
    "  * input: simulated community\n",
    "  * output: table\n",
    "    * sample, fraction_num, BD_start, BD_end\n",
    "  * params:\n",
    "    * isotope(s)\n",
    "      * determines possible max BD\n",
    "    * gradient min-max\n",
    "    * fraction size dist: mean, stdev"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# isopycnic.py scheme\n",
    "\n",
    "* input:\n",
    " * genome fasta\n",
    " * sim community file\n",
    " * isotope incorp file\n",
    " * fraction file\n",
    "\n",
    "* main:\n",
    " * foreach genome (workflow independent by genome; parallel)\n",
    "   * bio.sequence instance for genome\n",
    "   * isopycnic instance (incorp, simComm, script_args)\n",
    "     * both tables loaded as pandas DataFrames\n",
    "     * incorp table melted; column of sample index\n",
    "   * creating read index (genome location where reads originated)\n",
    "     * artificial PCR to select regions if amplicon fragment (primers provided)\n",
    "     * random fragments if shotgun \n",
    "     * generate by calling grinder?\n",
    "   * simulating fragment start-end\n",
    "     * from read start-end\n",
    "   * fragmenting genome\n",
    "     * load genome\n",
    "     * select sequence for each frag start-end\n",
    "     * calculate GC\n",
    "     * calculate BD\n",
    "   * array of BD values:\n",
    "     * write out [if wanted]\n",
    "     * fit to distributions\n",
    "     \n",
    " \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Specific Workflow - take 2\n",
    "\n",
    "## Simulate isotope incorporation\n",
    "* __makeIncorp_phylo.pl__\n",
    "* already complete\n",
    "\n",
    "## Define pre-isopycnic community \n",
    "* __preIsopycnicComm.pl__ \n",
    "* really, just utilizing Grinder\n",
    "* complete?\n",
    " \n",
    "## Simulate genome fragments and calculate BD\n",
    "* __isopycnic.pl__\n",
    "  * alter:\n",
    "    * write out the BD of each simulated fragment\n",
    "  * output: sample, genome, frag_scaf, frag_start, frag_end, GC, BD\n",
    "    \n",
    "## Fitting BD values to a PDF\n",
    "* __fitBD.py__\n",
    "  * load table as dict {genome: [BD_values,]}\n",
    "  * foreach genome (parallel):\n",
    "    * fit data to distributions\n",
    "    * output table: \n",
    "      * genome, AIC, BIC, distribution, l-moment(s)\n",
    "    \n",
    "## Create gradient communities \n",
    "* __makeGradientComms.py__\n",
    "* input:\n",
    " * fitBD.py output\n",
    " * community abundance file\n",
    "* Foreach genome: random draws from PDFs\n",
    "  * N-draws based on total community abundance & relative abundances of taxa\n",
    "  * using scipy for drawing from distribution\n",
    "   \n",
    "* output table of OTU abs abundance\n",
    "   * each genome = OTU\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Specific Workflow - take 3\n",
    "\n",
    "## Simulate isotope incorporation\n",
    "* __makeIncorp_phylo.pl__\n",
    "* already complete\n",
    "\n",
    "## Define pre-isopycnic community \n",
    "* __preIsopycnicComm.pl__ \n",
    "* really, just utilizing Grinder\n",
    "* complete?\n",
    "\n",
    "## simulate reads\n",
    "* __grinderSE.pl__\n",
    "  * altered grinder that just outputs start-end of reads (not read itself)\n",
    "  * parallel:\n",
    "    * for each sample-genome\n",
    "  * Output: sample, genome, scaffold, read_start, read_end\n",
    "\n",
    "## simulate fragments\n",
    "* __fragSim.py__\n",
    "  * parallel:\n",
    "    * for each sample-genome\n",
    "    * determine frag start-end from genome\n",
    "    * get fragment, calculate G+C and BD\n",
    "\n",
    "## Simulate genome fragments and calculate BD\n",
    "* __isopycnic.py__\n",
    "  * output: sample, genome, frag_scaf, frag_start, frag_end, GC, BD\n",
    "    \n",
    "## Fitting BD values to a PDF\n",
    "* __fitBD.py__\n",
    "  * load table as dict {genome: [BD_values,]}\n",
    "  * foreach genome (parallel):\n",
    "    * fit data to distributions\n",
    "    * output table: \n",
    "      * genome, AIC, BIC, distribution, l-moment(s)\n",
    "    \n",
    "## Create gradient communities \n",
    "* __makeGradientComms.py__\n",
    "* input:\n",
    " * fitBD.py output\n",
    " * community abundance file\n",
    "* Foreach genome: random draws from PDFs\n",
    "  * N-draws based on total community abundance & relative abundances of taxa\n",
    "  * using scipy for drawing from distribution\n",
    "   \n",
    "* output table of OTU abs abundance\n",
    "   * each genome = OTU\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "hide_input": true,
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
