{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "\n",
    "* Use validation dataset to test the power of HR-SIP with multiple 'heavy' BD windows\n",
    "  * global adjustment of all hypotheses for all windows\n",
    "  * using 2-4 windows and assessing the power"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Setting variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "workDir = '/home/nick/notebook/SIPSim/dev/bac_genome1147/validation/multiWindow_HRSIP/'\n",
    "genomeDir = '/var/seq_data/ncbi_db/genome/Jan2016/bac_complete_spec-rep1_rn/'\n",
    "figureDir = '/home/nick/notebook/SIPSim/figures/bac_genome_n1147/'\n",
    "\n",
    "# OTU_abs{abs}_PCR_sub.physeq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n",
      "The pushnote extension is already loaded. To reload it, use:\n",
      "  %reload_ext pushnote\n"
     ]
    }
   ],
   "source": [
    "import glob\n",
    "import nestly\n",
    "import os\n",
    "%load_ext rpy2.ipython\n",
    "%load_ext pushnote"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "library(ggplot2)\n",
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(gridExtra)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/nick/notebook/SIPSim/dev/bac_genome1147/validation/multiWindow_HRSIP\n"
     ]
    }
   ],
   "source": [
    "if not os.path.isdir(workDir):\n",
    "    os.makedirs(workDir)\n",
    "        \n",
    "%cd $workDir    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HR-SIP with multiple windows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def HRSIP_multi_window(physeq, BD_shift, BDs, outname, padj=0.1, log2=0.25):\n",
    "    # HR-SIP for each window\n",
    "    occurs = ','.join([str(x/100.0) for x in range(0,55,5)])\n",
    "    outname2 = outname + '_DS2.txt'\n",
    "    !SIPSimR phyloseq_DESeq2 --occur_all $occurs -w $BDs $physeq > $outname2\n",
    "    # making confusion matrix\n",
    "    !SIPSimR DESeq2_confuseMtx --padj $padj --log2 $log2 $BD_shift -o $outname $outname2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.70-1.74,1.72-1.76"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1070\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 75 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1056\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1056\n",
      " post-filter: number of taxa: 1041\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 74 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 990\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 990\n",
      " post-filter: number of taxa: 978\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 74 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 930\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 930\n",
      " post-filter: number of taxa: 919\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 71 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 855\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 855\n",
      " post-filter: number of taxa: 849\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 66 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 781\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 781\n",
      " post-filter: number of taxa: 777\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 63 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 715\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 715\n",
      " post-filter: number of taxa: 715\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 60 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.35\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 638\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 638\n",
      " post-filter: number of taxa: 638\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 53 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.4\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 555\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 555\n",
      " post-filter: number of taxa: 555\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 49 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.45\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 502\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 502\n",
      " post-filter: number of taxa: 502\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 44 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.5\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 438\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 438\n",
      " post-filter: number of taxa: 438\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 33 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.76\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 19\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 973\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 4 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 100\n",
      "\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1056\n",
      "# \"heavy\" BD window: 1.72-1.76\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 19\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1056\n",
      " post-filter: number of taxa: 964\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 5 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 100\n",
      "\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 990\n",
      "# \"heavy\" BD window: 1.72-1.76\n",
      " pre-filter: number of samples: 56\n",
      "**OUTPUT MUTED**"
     ]
    }
   ],
   "source": [
    "BDs = '1.70-1.74,1.72-1.76'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW1')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.70-1.75,1.72-1.77"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1076\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 82 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 21\n",
      "\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1056\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1056\n",
      " post-filter: number of taxa: 1047\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 82 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 21\n",
      "\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 990\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 990\n",
      " post-filter: number of taxa: 981\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 81 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 23\n",
      "\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 930\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 930\n",
      " post-filter: number of taxa: 922\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 81 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 28\n",
      "\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 855\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 855\n",
      " post-filter: number of taxa: 851\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 74 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 40\n",
      "\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 781\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 781\n",
      " post-filter: number of taxa: 779\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 64 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 37\n",
      "\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 715\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 715\n",
      " post-filter: number of taxa: 715\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 61 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 33\n",
      "\n",
      "# occurance cutoff: 0.35\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 638\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 638\n",
      " post-filter: number of taxa: 638\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 60 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 32\n",
      "\n",
      "# occurance cutoff: 0.4\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 555\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 555\n",
      " post-filter: number of taxa: 555\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 46 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 29\n",
      "\n",
      "# occurance cutoff: 0.45\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 502\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 502\n",
      " post-filter: number of taxa: 502\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 43 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 29\n",
      "\n",
      "# occurance cutoff: 0.5\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 438\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 438\n",
      " post-filter: number of taxa: 438\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 38 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 20\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 25\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1000\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 21 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 103\n",
      "\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1056\n",
      "# \"heavy\" BD window: 1.72-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 25\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1056\n",
      " post-filter: number of taxa: 991\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 21 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 103\n",
      "\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 990\n",
      "# \"heavy\" BD window: 1.72-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 25\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 990\n",
      " post-filter: number of taxa: 947\n",
      "# DESeq2 run\n",
      "**OUTPUT MUTED**"
     ]
    }
   ],
   "source": [
    "BDs = '1.70-1.75,1.72-1.77'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW2')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.70-1.73,1.72-1.75,1.74-1.77"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1064\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 13 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1056\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1056\n",
      " post-filter: number of taxa: 1037\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 14 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 990\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 990\n",
      " post-filter: number of taxa: 977\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 13 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 930\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 930\n",
      " post-filter: number of taxa: 919\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 15 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 855\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 855\n",
      " post-filter: number of taxa: 849\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 14 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 781\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 781\n",
      " post-filter: number of taxa: 777\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 13 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 715\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 715\n",
      " post-filter: number of taxa: 715\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 14 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.35\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 638\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 638\n",
      " post-filter: number of taxa: 638\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 15 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.4\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 555\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 555\n",
      " post-filter: number of taxa: 555\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 13 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.45\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 502\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 502\n",
      " post-filter: number of taxa: 502\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 10 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.5\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 438\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 438\n",
      " post-filter: number of taxa: 438\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 10 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 15\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 944\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 2 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 79\n",
      "\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1056\n",
      "# \"heavy\" BD window: 1.72-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 15\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1056\n",
      " post-filter: number of taxa: 935\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 2 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 80\n",
      "\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 990\n",
      "# \"heavy\" BD window: 1.72-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 15\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 990\n",
      " post-filter: number of taxa: 899\n",
      "# DESeq2 run\n",
      "**OUTPUT MUTED**"
     ]
    }
   ],
   "source": [
    "BDs = '1.70-1.73,1.72-1.75,1.74-1.77'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW3')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.70-1.72,1.71-1.73,1.72-1.74,1.73-1.75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1040\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1056\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1056\n",
      " post-filter: number of taxa: 1020\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 990\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 990\n",
      " post-filter: number of taxa: 967\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 930\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 930\n",
      " post-filter: number of taxa: 913\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 855\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 855\n",
      " post-filter: number of taxa: 845\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 781\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 781\n",
      " post-filter: number of taxa: 774\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 715\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 715\n",
      " post-filter: number of taxa: 714\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.35\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 638\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 638\n",
      " post-filter: number of taxa: 637\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.4\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 555\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 555\n",
      " post-filter: number of taxa: 555\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.45\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 502\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 502\n",
      " post-filter: number of taxa: 502\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.5\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 438\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 438\n",
      " post-filter: number of taxa: 438\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 979\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1056\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1056\n",
      " post-filter: number of taxa: 959\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 990\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 990\n",
      " post-filter: number of taxa: 914\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 930\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 930\n",
      " post-filter: number of taxa: 867\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 855\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 855\n",
      " post-filter: number of taxa: 809\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 781\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 781\n",
      " post-filter: number of taxa: 746\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 715\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 715\n",
      " post-filter: number of taxa: 694\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "**OUTPUT MUTED**"
     ]
    }
   ],
   "source": [
    "BDs = '1.70-1.72,1.71-1.73,1.72-1.74,1.73-1.75'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW4')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.71-1.75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1025\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 17 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 59\n",
      "\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1056\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1056\n",
      " post-filter: number of taxa: 1003\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 17 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 59\n",
      "\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 990\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 990\n",
      " post-filter: number of taxa: 949\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 17 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 63\n",
      "\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 930\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 930\n",
      " post-filter: number of taxa: 897\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 15 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 62\n",
      "\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 855\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 855\n",
      " post-filter: number of taxa: 832\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 11 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 63\n",
      "\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 781\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 781\n",
      " post-filter: number of taxa: 767\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 10 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 54\n",
      "\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 715\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 715\n",
      " post-filter: number of taxa: 709\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 8 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 51\n",
      "\n",
      "# occurance cutoff: 0.35\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 638\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 638\n",
      " post-filter: number of taxa: 637\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 9 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 47\n",
      "\n",
      "# occurance cutoff: 0.4\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 555\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 555\n",
      " post-filter: number of taxa: 555\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 8 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 38\n",
      "\n",
      "# occurance cutoff: 0.45\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 502\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 502\n",
      " post-filter: number of taxa: 502\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 8 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 34\n",
      "\n",
      "# occurance cutoff: 0.5\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 438\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 438\n",
      " post-filter: number of taxa: 438\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 6 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 27\n",
      "\n",
      "# A file of all DESeq2 run results was written to: DESeq2_all_runs.txt\n",
      "\n",
      "#- Greatest number of rejected hypotheses for each BD window -#\n",
      "heavy_BD_min\theavy_BD_max\tmax_n_rej_hypo\n",
      "1.71\t1.75\t63\n",
      "\n",
      "# Global adjustment of p-values for all BW windows with \"best\" occurance cutoff\n",
      "\n",
      "#- Number of rejected hypotheses for each BD window (post global p-value adjustment-#\n",
      "heavy_BD_min\theavy_BD_max\tn_rej_hypo\n",
      "1.71\t1.75\t63\n",
      "\n",
      "Log2Fold cutoff: -Inf\n",
      "padj cutoff: 0.1\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW5\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW5_data.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW5_table.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW5_overall.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW5_byClass.txt\n"
     ]
    }
   ],
   "source": [
    "BDs = '1.71-1.75'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW5')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.71-1.8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1053\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 69 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 88\n",
      "\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1056\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1056\n",
      " post-filter: number of taxa: 1031\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 68 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 88\n",
      "\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 990\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 990\n",
      " post-filter: number of taxa: 974\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 65 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 91\n",
      "\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 930\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 930\n",
      " post-filter: number of taxa: 920\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 66 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 89\n",
      "\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 855\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 855\n",
      " post-filter: number of taxa: 851\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 64 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 85\n",
      "\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 781\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 781\n",
      " post-filter: number of taxa: 778\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 63 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 72\n",
      "\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 715\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 715\n",
      " post-filter: number of taxa: 714\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 62 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 67\n",
      "\n",
      "# occurance cutoff: 0.35\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 638\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 638\n",
      " post-filter: number of taxa: 638\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 62 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 62\n",
      "\n",
      "# occurance cutoff: 0.4\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 555\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 555\n",
      " post-filter: number of taxa: 555\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 59 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 52\n",
      "\n",
      "# occurance cutoff: 0.45\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 502\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 502\n",
      " post-filter: number of taxa: 502\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 54 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 47\n",
      "\n",
      "# occurance cutoff: 0.5\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 438\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 438\n",
      " post-filter: number of taxa: 438\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 47 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 38\n",
      "\n",
      "# A file of all DESeq2 run results was written to: DESeq2_all_runs.txt\n",
      "\n",
      "#- Greatest number of rejected hypotheses for each BD window -#\n",
      "heavy_BD_min\theavy_BD_max\tmax_n_rej_hypo\n",
      "1.71\t1.8\t91\n",
      "\n",
      "# Global adjustment of p-values for all BW windows with \"best\" occurance cutoff\n",
      "\n",
      "#- Number of rejected hypotheses for each BD window (post global p-value adjustment-#\n",
      "heavy_BD_min\theavy_BD_max\tn_rej_hypo\n",
      "1.71\t1.8\t91\n",
      "\n",
      "Log2Fold cutoff: -Inf\n",
      "padj cutoff: 0.1\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW6\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW6_data.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW6_table.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW6_overall.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW6_byClass.txt\n"
     ]
    }
   ],
   "source": [
    "BDs = '1.71-1.8'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW6')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Plotting results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['OTU_n2_abs1e9_PCR_subNorm_MW1_byClass.txt',\n",
       " 'OTU_n2_abs1e9_PCR_subNorm_MW2_byClass.txt',\n",
       " 'OTU_n2_abs1e9_PCR_subNorm_MW3_byClass.txt',\n",
       " 'OTU_n2_abs1e9_PCR_subNorm_MW4_byClass.txt',\n",
       " 'OTU_n2_abs1e9_PCR_subNorm_MW5_byClass.txt',\n",
       " 'OTU_n2_abs1e9_PCR_subNorm_MW6_byClass.txt']"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files = ['OTU_n2_abs1e9_PCR_subNorm_MW{}_byClass.txt'.format(i) for i in xrange(1,7)]\n",
    "files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "  library      variables    values file             BD_window\n",
       "1       2    Sensitivity 0.8990826  MW1 1.70-1.74,\\n1.72-1.76\n",
       "2       2    Specificity 1.0000000  MW1 1.70-1.74,\\n1.72-1.76\n",
       "3       2 Pos Pred Value 1.0000000  MW1 1.70-1.74,\\n1.72-1.76\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -i files\n",
    "\n",
    "renames = data.frame(file = c('MW1', 'MW2', 'MW3', 'MW4', 'MW5', 'MW6'),\n",
    "                     BD_window = c('1.70-1.74,\\n1.72-1.76', \n",
    "                                   '1.70-1.75,\\n1.72-1.77',\n",
    "                                   '1.70-1.73,\\n1.72-1.75,\\n1.74-1.77',\n",
    "                                   '1.70-1.72,\\n1.71-1.73,\\n1.72-1.74,\\n1.73-1.75',\n",
    "                                   '1.71-1.75', '1.71-1.8'))\n",
    "\n",
    "df = list()\n",
    "for(F in files){\n",
    "    tmp = read.delim(F, sep='\\t')\n",
    "    df[[F]] = tmp\n",
    "}\n",
    "df = do.call(rbind, df)\n",
    "df$file = gsub('.+(MW[0-9]+).+', '\\\\1', rownames(df))\n",
    "\n",
    "df = inner_join(df, renames, c('file' = 'file')) \n",
    "\n",
    "df %>% head(n=3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAImCAMAAAC8U6mTAAADAFBMVEUAAAABAQECAgIDAwMEBAQF\nBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcY\nGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKior\nKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+\nPj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBR\nUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2Nk\nZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3\nd3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmK\nioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJyd\nnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+w\nsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLD\nw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW\n1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp\n6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8\n/Pz9/f3+/v7////isF19AAAgAElEQVR4nO19C3wV1bX3qe1VUURRpHrQrzwiip8giLW1/azWT/G2\nCn70a28ErmCsEG19tRSpVfoQaWnhaqu1udoKPloeQREU1EiCeQgIGaWoFVFRSzBQhQSSQGJCztyZ\nOTNnZs7Ze89as2efR1j/38+czMxi+c/8z8zsWWuvtWM6oUcjlmsCBLUggXs4SOAeDhK4h0MkcGL0\nVvu35m8f/+3m1IeJroM8HOAekQLCLYZBPrvd+f52GXwWIHBiSXHMEfj7xZ8Ufz/1YaLu5ToOqngH\nMlFdDbdFuF1bq8QtwrR2bRRuf1Urg1X/DBD4UGmpI3B3n1f1jSck7I+kwAd5/65NcE9IQ+dncFuE\n24PdStwiTLu5Zwfj9l5NBlVBApsHbYH3xvbpzbFm+0PXN86d+0DrIQ5aeAcy0dEOt0W4betU4hZh\n2tkWhdvsCfx+rEvvir1vf+h6xbRp9zZ3cLCPdyATbW1wW4TbloOcAxNgkGfQvj8KBtkTeE9sv3Hp\n7rU/rF15c4sGnjK0rdQfxr1FoxhkT+DuPq/p9X0S9oe1iwQWQbXAm+qDt8ACL9tvDJ+/3379DakP\nEySwCKoFLrk/eAsssPmj+Vt9r2xOfZgggUVQILDvMi0pDt6CCCyALbDcKSOBhQx8AtsXZtzGxYFb\nJDCWbY4FTl6YKxxoQVuSAtcm34OBhDkvelG8B2MYyLGN4j0YxcAW2H9hgiErcFu3CSDhbjY62jkH\nGGjl7McwkGPLY8BA14EI2HbbAvsvzGwJTLdoEQrlPVgAElgEdQLHH4BuBQvsJgfLYiaK95g/r04e\nJIFDUUAxYAm8oha6FSywmxxsbWho2DGqasMQ4/PT5EESOBQFFANnkOW7MMEIFNibHDSwcKb+5FXu\nURI4FAUUA2eQVZupXt29gVuBAqeSg8mtEW36z//3wD5XfZjcJoFDUUAxYN2i1yQ/KuKBW4ECp5KD\nFqaV6fpdExt2X/NlY+OBvn1LPm01ASTcysa+fZwDDOzl7McwkGPLY4Bxi2LgFzh5Yca/PnONK6Jo\nK1Bgb3JQbzzR/v42xj4xhl/bt6+mKzgMBRSDexkX7Zo/TB56fXXqmhVsAZ7BbnJQn20mkcq26/on\nsRZrB92iQ1FAMXAGWb4LU9Nemf21eS86G/wtwCg6mRw004X6uU8bP274xjufXHtl8iAJHIoCioFz\nBfsuTEvGmy9yN3hbgPfgZHLQTBc2xnYbO1omHn/S5D3JgyRwKAooBu4z2H+ZGni0WAvaokgWlm1O\nR9HOhTniu8uMa3rCvE2eY26quL56VbWzQQJj2eb4NSl5Ycannr1Mq/vJiB95jjhzOGpKiuLxeNF1\nNZEIfCBhAkg4wcZnHZwDDLRy9mMYyLHlMcC4RTGwBa5/cI7x87errI14xdyvb9S05cO9AttzOMZO\nXlyzsWZR8bgoBK7d32kCSLiTjYMHOAcY2M/Zj2Egx5bHAOMWxcAWeMZwU+Axg+5PClw/xthfNyQp\nty9VPHRDUtkiukWHYpurW/Rw46n7RL228PykwNpTZz5YU3qldcifKh47ZWntxtrFxWNJ4FBscyXw\nwHVabbxW2zDIFlh7qCh+wbNaJqqnDDGfwZOrSeBQbHMl8KV3vjJrwJx1d1n34RXGA1hbX1HP0NfA\nprWrqp3xNSYf7CSCPeWjJHAoCigGtsDl58aHPX5e/Oy/2sr5B9fcLUw+2EkEe8tHSeAwFFAMHLU2\nVWzS6itSb75xn6TcLUw+2E4EM8pH5U4ZCSxk4F6O93JElBLYkw+2E8GpPe+Ulz9OAoehgGJwL1vF\niRpoC5MPthPBqT3ll102Y+8BE0DCB9ho2c85wEATZz+GgRxbHgOMWxQDjsBaRMVnvnywlQhmlI8C\nCXP+F3QFCxm4Avsv04iKzzz5YDsRzCgflTtlJLCQgeriMzcf7CSCM8tH5U4ZCSxkoLr4zM0HO4ng\nzPJRuVNGAgsZ5LT4jAQORQHFIKfFZyRwKAooBjktPqPy0VAUUAxyWnxG5aOhKKAYqC4+E4Ju0aEo\noBioLj4TggQORQHFQHHxmSc52H33gN5XbNOpfFSaAoqB4uIzT3Jw4elvt9wyLEHlo9IUUAwUF595\nkoMT7tH1pthOKh+VpoBioLj4zJMu3NWi68v7tFP5qDQFFAPFxWe+8tGusn4r3fLRJ0aPvo3ShWEo\noBgoLj7zJgc3j7xkS3KvVT760Zo1i+kKDkMBxUBx8ZknObi5/wLzk8pHpSmgGKguPnPTheNvNobP\nDZ1UPipNAcWAVXymad8uqUvt9VW1pBWmYdKFp1ltlLZS+ag0BRQDVvGZpo28/YLFzj5fVUtaYRpF\nsrBscyVwXckc7fbp8xdV2wJrT55/hz2H1lfVklaYRgJj2eZK4IljVmkTvjruvLgjsFZ77eXPW7/7\nqlrSCtNIYCzbXAlctNJ4Rxplj6Kd7H+ygtBX1ZJWmEblo1i2uSofPbNK0yqLbIFXrDjHk/33V7X4\nC9PoCsayzdUVPGlS1Zprrk69B8/yjrx8VS3+wjQSWGibB24dgdeOjccvXq2tHa+hkHWBgabY84Aw\nLTC37mtSbaX9y+p13tyhaAuXD2asPkoCK3fLSBfeOtSb/Rdt4fLBrNVHSWDVblV3fPfkg/mrj0b/\nt5HAE7IksCcfnL766G5NW04Cq3arWmBPPjh99dFHBg++8VNs318GomgnzEBTixK3CNOWpijcKhbY\nkw/O59VHGaAFokECe/LBrNVHG5s42Mk7kIlP/gW3Rbht3KPELcJ0D/fsYNzOViuwt51w5uqjG+YS\nVOPXVXUSWLsnSGBPO+HM1UcJBYKwkSxCgYAE7uEggXs4RAInRm+1f8uMRRMKBHyBE0uKY47AmbHo\n915aw8GLvAOZeInrRMotwlSV24oo3C54oUICz/4rQOBDpaWOwPxYNAMU6CiQQIeefEMykR6L1klg\nMQpN4PRY9O9isYkUixagIGLRHoHTY9HtTU2VPXzKDoJBoWaTdFdgViyaBA5DAcUgSwKzY9EkcCgK\nKAaBAouazsIFZseiSeBQFFAMAgUWNZ2labNotvkgsP8qFTWdJYHRbPNBYOcqFbWZjUbgWlraLgwF\nFAOWwPZVKmozG5HALV0mgIS72Gg/yDnAQAtnP4aBHFseA4xbFAP/UhzYnrN0i8ayze0tGt1zlgTG\nss2HZzACJDCWbT4JLGozCxbYzf6WWT06iqlXpTQFFAOBwKI2s2CB3exva0NDw45RVdSrUpoCioF/\nkIXtOYupTTKxcKZOvSqlKaAY+AdZrKtU1HQWU5tkbo1o01O9KjfOnfsAtfQPQwHFQHCLFrWZhQrs\n61WpTyvT3V6VFdOm3dvcYQJIuIONtjbOAQb2cfZjGMix5THAuEUxYAlsX6WiNrNQgX1L2zWeaN+g\nrF6VOt2iQ1JAMfAJnHbNCtrMQgX2Zn/12WaWkHpVSlNAMfAPstKvUlHTWZDA3tok/dynjR/Uq1Ka\nAoqB/wrOvEpFTWdxvSqNG/NuYwf1qpSmgGKQ/gzOvEpFTWcpkoVlm3OBPVepv7GsptVXr6r2T+gg\ngdFs80Dg1FXqbyxbU1Jkdje8riZCgWlpu1AUUAzSBPY2ivY3lh07eXHNxppFxeOiFJiWtgtDAcUg\nTWBvo2h/Y9mhG5KSFtEtWoZtrm/R3kbR/sayY6csrd1Yu7h4LAkswzb3AruNov2NZaunDDGfwZOr\ncQK76UInTyjVypABEljIgDllJ3kb9jeW1bRNa1dVb9I0nMBuutDJE0q1MmSABBYySJuyc45/yo7/\n+s4ccWPShXaeUK6VIQMksJBBmmiz/JtxwRZIYE+60M4TMspH5U4ZCSxkkH5V+relBfakC+08YWrP\nA337liTLR4GEOfWRUZSPYhjIseXWeSLcohikC+wXcaJgCySwL11o5QlTe0I1I2WArmAhA7HAmmzx\nmSddaOcJqXxUmgKKQbrAGVepZPGZmy508oRUPipLAcVAdfGZmy508oRUPipLAcUgp8VnJHAoCigG\nOS0+I4FDUUAxyG3xGZWPhqGAYpDT4jNa2i4UBRQDKj7DMpBjmw/PYAeRFJ8JQQKHooBioLr4TAgS\nOBQFFAPFxWee7G/33QN6X7FNp/JRaQooBoqLzzzZ34Wnv91yy7AElY9KU0AxUFx85sn+TrhH15ti\nO6l8VJoCioHi4jNP9ndXi64v79OeKh99p7z8cRI4DAUUA8XFZ77y0a6yfivd8tHyyy6bsfeACSDh\nA2y07OccYKCJsx/DQI4tjwHGLYqB4uIzbz5488hLtiT3UvmoFAUUA8XFZ57s7+b+C8xPKh+VpoBi\noLr4zM0Hj7/ZGD43dFL5qDQFFANB8VldyRzt9unzFyVnQtc/OMf4+dtVPltMPvg0q43SVioflaaA\nYiAoPps4ZpU24avjzkvKPWO4KfCYQb4pHRTJwrLNC4Ht16SilcZ4a5TzRB6+TNOeqNcWnk8Cy7DN\np9ekM6s0rbLI2Rq4TquN12obBkUoMJWPhqKAYiB4TZo0qWrNNVc7W5fe+cqsAXPW3RVpv2gSOAwF\nFAPBa9LasfH4xau1teOtrfJz48MePy9+9l/pFi3DNufPYN9rUm2l58imik1afYW/+izrAgNNsecB\nYVpgbkWvSQBg0oWM1UdJYOVu+a9Jq9d5k4f+LYTAnmLRzNVHSWDlbvnpwluHetP//i24wJ50IX/1\n0ej/NhJ4QqDAIGDShfzVR6P/20jgCVkS2JMuTF999InRo29Dl48CTbFVlgjTAnOrevVRT7owffXR\nj9asWUzrBwtQEOsHe9KF/NVHUYQzQAIL3aqe+O7pNstYfXRxOQfcA5lYsgRui3C7eKkat3DTpZG4\n/dWzMliK6TabufroLqlvFwGCTW0HZdAdJDChJ4AE7uEggXs4RAInRm+1f8uMRRMKBHyBE0uKY47A\nmbHo9iYe9nCPZGDvXrgtxi3cFOMWYYqhwHe79Y23ZHAwQOBDpaWOwKxY9Abe2G8dfJi48VW4LcLt\nhvpgmxBuEab13LODcXvPehlUAPLBjsD8WDQDFOgolECHK3B6LNpd+QyzQFgmolj5jIGWg0rcIkzb\n90fhNnsCp8eizbULuR0ruAcyEUWPDgZaOzgHgKF+eQYdrVEwyJ7A+RyLBp4ytK3UH8a9RaMYZElg\nTiyaBBYguwJvYg4qwQKzY9E0q1KE7Aqc1oUUIbAAJLAIqgUW9SQlgcOyzSOB2T1JSWA5tjmf+J55\nzYr6G8qWrrR0mQAS7mKj/SDnAAMtnP0YBnJseQwwblEMWALDe5LKCkzNSMNQQDFgCQzvSUq3aCzb\nPLpFQ0ACY9nmocCi/oaY2qQyq4VDMbUylKaAYhAsMLMoCSywm/1tbWho2DGqiloZSlNAMWAOssA9\nSTG1SSYWztSplaE0BRQD5iAruCcpVGBv9tfYGtGmp1oZ6iRwSAooBsG3aGYXUqjAvlaG+rQy3W1l\n+LtYbCK1MgxDAcVAIDCzJylSYN/Sdo0n2t9fq5Vhe1NTJV3BYSigGLAEFvUkRQrszf7qs80sIbUy\nlKaAYsAcZIl7kmIE9tYm6ec+bfygVobSFFAMmFdwUE9ShMCe2qTG2G5jB7UylKaAYsB5Bgf1JAUL\nLAQJHIoCigF3kOW5ZuurV1WzZwmTwFi2+SNw6pqtKSmKx+NF19UoEJiySWEooBiwBB7x3WXGg3jC\nvGTTs7GTF9dsrFlUPC56gSkfHIoCigFzFD317GVa3U9G/MjaGpqsoKgqolt0BGzz4hYdr5j79Y2a\ntnx48gqesrR2Y+3i4rEkcARs80Tg+jHG/roh1lb1lCHmM3hyNQkcAds8EVh76swHa0qvtLc3rV1V\nvYlhh8sHO4lgqV6VDJDAQgYcgbWHiuIXPOvs4A61MflgJxEs1auSARJYyICZLjQewNr6CvfVl9t9\nFpMPthPBcr0qGSCBhQw4F6d/d3iBPflgOxGc2mNWF1LH9zAUUAw4AvslnRhaYE8+2E4E++qDm6xe\nTEDCnE5OrS3wrk/NnP0YBnJseQwwblEMQAJroYvPfPlgKxHs3UO36FAUUAw4AmdcsyGLzzz5YDsR\nzKgPljtlJLCQgeriMzcf7CSCM+uD5U4ZCSxkoLr4zM0HO4lgqg+WpYBikNPiMxI4FAUUg5wWn5HA\noSigGOS0+IxWPgtFAcUgp8VntW3dJoCEu9noaOccYKCVsx/DQI4tjwHGLYqB6uIzIegWHYoCioHq\n4jMhSOBQFFAMFBefeZKD3XcP6H3FNp3KR6UpoBgoLj7zJAcXnv52yy3DElQ+Kk0BxUBx8ZknOTjh\nHl1viu2k8lFpCigGiovPPOnCXS26vrxPO5WPSlNAMVBcfOYrH+0q67fSLR99ZPDgG6l8NAwFFAPF\nxWfe5ODmkZdsSe61ykd3a9pyuoLDUEAxiL74LDH/wpN33bU4ueFJDm7uv8D8pPJRaQooBtEXn/1X\n/+diu1Ye/2hyy00Xjr/ZGD43dFL5qDQFFANI8dm3S+o4RiyBBz2gx3bpvz4rueWmC0+z2ihtpfJR\naQooBsHFZ5o28vYLFsMFPvolU+DVxwDONwkcigKKAVvgTS+vqnVncIzUnjz/DubMd5bAw39rCnzn\neYDzTQKHooBiwBL45UlmscqwH6xPCazVXnv580CBy3r/d6z8ji/8FXC+qXw0FAUUA5bAV91QWX3T\nzUu/O8nactL/0OrC7t/0jsVOXwDQV6/dH8E6JlGsuoJhIMeWu+oKwi2KAUvgITXGJTtUqzzD2lqx\n4hzklJ3uj5og8tItOiQFFAOWwGNuern69otrbnCm7Mxi2IgEBqOnC5wHbpkCr740Hv/KU498ayVX\nWJHAW20AzjdeYKAp9jwgTAvMLWcUXb3GGUOvXodN+MdsJLc8xaKZy8uSwMrdBqYLbx0aZspO6/P/\n54Pkb55i0czlZUlg5W6VdXxffKn14ckHs5aXJYFVu1UmcF1v68OTD05fXvad8vLHSWDVbhUIbI2w\n6i87x9rw5IPTl5ctv+yyGXuxiVsGosgHM7CvTYlbhGlbcxRuoxc4OcQaVGNtePLB6cvLurdoBmiB\n6AJZINqTD2YtL/vBxxy8zzuQiX9+BLdFuP2gQYlbhGkD9+xg3M5WK7C3nXDm8rL1v+fhfu4RKeTe\nbdYZzF8fLCMfr+7zC7zVg+QeTzvhzOVlCQWClMAxD3JJiBAtSMweDobA7R9lnwZBFfwCt5hP4IeO\nt7cSo520Q2YsmlAg8Am89PPmE/iIO62NxJLimCNwZix660trOHiRdyATFRVwW4RbhKkity9F4vZP\nz6yUwNLGTIGHTW358hsfjVxvbRwqLXUE5seiGaBAR/4GOo5cod/1mL7kEmfbETg9Ft28fftqEliA\nfBW47yN6+ff19cc5247A6bHoB/r2LSnEWDQw1C/PgBuLRjGIXuAxwza9d0rj7EHpAudzLBqTn8HY\nSv1h3CsYxSB6gTefOlOf/rkjlzjb7jM4Mxbdw+dkIRgwkK8C6937jAeu+4dYArNj0SSwCNkVGNZt\ntpNBxxKYHYsmgUNRQDGACgzrNnvi9S8gTjcJHI4CikG03WZLT4mdcN1q8BkngUNRQDGIuNts94aZ\nZ8aOn/xcB+h8U6e7UBRQDBR0m936m6/E+vznSojAtLRdGAooBmq6zX78w89BUoh0iw5FAcVAQbfZ\nxJa7z4ydVAo43yRwKAooBlG/Byc2/+yM2PHXvcB6YcoACRyKAopBtN1mX/vpkNixE1Z4hlhu9rfM\nmshTTL0qpSmgGETbbTZ21HeWHvDxdrO/rQ0NDTtGVVGvSmkKKAbRdpt9Yn8abW/218DCmTr1qpSm\ngGKguNuspzbJ3BrRpqd6VdbMnDmvucMEkHAHG21tnAMM7OPsxzCQY8tjgHGLYqC426yvV6U+rUx3\ne1WaAlOPjjAUUAwUd5v1LW3XeKJ9g7J6Vep0iw5JAcVAcbdZb/ZXn21mCalXpTQFFAPF3Wa9tUn6\nuU8bP6hXpTQFFIPou8364alNaoztNnZQr0ppCigG0XebRYAEDkUBxQDSbba+elU1c0IHCYxmmz8C\np67ZmpIis5HhdTXRC0zpwlAUUAxYAo/47jLjQTxhXrLB7NjJi2s21iwqHqdAYGpGGoYCigFzFD31\n7GVa3U9G/MjaGrohqSW0GSkCdIsORQHFgClwxdyvb9S05cOTV/CUpbUbaxcXjyWBI2CbJwLXjzH2\n1w2xtqqnmN2jiyZXhxLYTRc6eUKpVoYMkMBCBmyBtafOfLCm9Ep7e9PaVdXMhu+ope2cPKFUK0MG\nSGAhA47A2kNF8QuedXZwh9qYpe3sPKFcK0MGSGAhA2a60HgAa+sr3FdfVhALJrAnXWjnCb0JRBI4\nFAUUA87F6d8dXmBPutDOE6b2/C4Wm0hL24WhgGLAEdgv6cTQAvvShVaeMLWnvampkq7gMBRQDEAC\na9Diswx40oV2npDKR6UpoBhwBM64ZmHFZ5lw04VOnpDKR2UpoBhEW3yWCTdd6OQJqXxUlgKKQcTF\nZziQwKEooBgoKD6DgwQORQHFQE3xGRC0tF0oCigGCorP4KB8cCgKKAaKO74HXMF0iw5DAcUg2uIz\nJEjgUBRQDKItPkOCBA5FAcUg2uKzTLjZ3+67B/S+YptO5aPSFFAMFBefebK/C09/u+WWYQkqH5Wm\ngGKguPjMk/2dcI+uN8V2UvmoNAUUA8XFZ57s764WXV/epz1VPvr3hx9+uPWQCSDhQ2x0tHMOMNDC\n2Y9hIMeWxwDjFsVAcfGZr3y0q6zfSrd8dOX3vjer6aAJIOGDbLS2cA4w0MzZj2Egx5bHAOMWxUBx\n8Zk3H7x55CVbknupfFSKAoqB4uIzT/Z3c/8F5ieVj0pTQDFQXXzm5oPH32wMnxs6qXxUmgKKQXDx\nWV3JHO326fMXhZwXncoHn2a1UdpK5aPSFFAMgovPJo5ZpU346rjzwtUHC0ECh6KAYhD8mlS00hhv\njQo5ihaDBA5FAcUg+DXpzCpNqyxSInAtvQeHoYBiEPyaNGlS1ZprrlYjMPWLDkMBxSD4NWnt2Hj8\n4tXa2vH5cIsGmmLvZAjTAnMLek2qrWQbkcD57xbUo4MPTLqQsfooCazcbeBr0up1cgl/T7Fo5uqj\nJLByt4HpwluHSk3Z8aQL+auPRv+3kcAToAKLgUkXpq8+SgJnw61qgT3pwvTVRx8ZPPhGdPko0BRb\nZYkwLTC3ClYf9cGTLkxffXS3pi3Pl9VHGaD1g0ECe9KF/NVHUYQzQAIL3aqe+O7pNstYfXTh4xxw\nD2Tiscfgtgi3CFNFbh+LxO0vl8jgsR1BAnu6zWauPvrpWwTVeDvBFweMsJEsQoGABO7hIIF7OEQC\nJ0ZvtX/LjEUTCgR8gRNLimOOwJmx6ObtPLzHPZKB99+H2yLcvqfILdw0mj+srrpOBq0BAh8qLXUE\nZsWit/Jo/QP+t23bBrdFuH0bcXoRbhGm778dhdvZit+Dk29IJvixaAYo0FEogQ5X4PRYtFu6gqnw\nyEQUpSsM7DugxC3C9MC+KNxmT+D0WLRZfNaGnTzFQBRzshg40MU5AAz1yzPoOhAFg+wJ3FNi0cDT\nK8+Ae4tGMciSwJxYNAksQEEJzI5F543AmFNGAsNBAouQJYHZfYRJ4B4jMLuPMAnccwRm9hGORuBa\nbmM/7oFM9JBWhpn4rDUCtqJWhoCepNE0I8X07MxED2lGmolDvLODYiAQGNCTlG7RYlupP4xG0UmQ\nwDkV2M3+llktHIoZrQzlzgMJnFOB3exva0NDw45RVdTKEArlAgN6kmJqk0wsnKlTK0MolAssKiuE\nCuzN/hpbI9r0VCvD5u3bV5PAAhTELdrXylCfVqa7rQzNpe0+bTUBJNzKxr59nAMM7OXsxzCQY8tj\nwEBLUwRsW4MFZvYRhgrsW9qu8UT7+0utDCFQfgWL+ghDBfZmf/XZZpaQWhlCoX6QJegjDBXYW5uk\nn/u08YNaGUKh/goW9BEGC+ypTWqM7TZ2UCtDKLIxyOL2EQYLLAQJLEJ2RtHiZjskMJZt3gnM6SMc\njcDU6U6ELM2qVDmjg5a2E0F5PjiJLMzoABLmOOkht2iEWxSDnM7oIIFDUUAxyO2MDhI4DAUUA9Uz\nOtx8sJMIlupVyQAJLGSQvXywkwiW6lXJAAksZKC8T1YqH2wnguV6VTJAAgsZiAUOkh+TD7YTwak9\nNTNnztvfaQJIuJONgwc4BxjYz9mPYSDHlscA4xbFQCxhUMdoTD7YTgSn9pDAYSmgGCgW2JcPthLB\n3j10iw5FAcVALPBEWYE9+WA7EcyoD5Y7ZSSwkIHqUKWbD3YSwZn1wXKnjAQWMlAdqnTzwU4iOLM+\nWO6UkcBCBhSqxDKQY5tPAlOokslAjm0+Cay++IzShaEooBjktPiMEv6hKKAY5EN1IZAwxwndooUM\nSGAsAzm2+SRwFMVnnuRg990Del+xTWeUj8qdMhJYyEBx8ZknObjw9LdbbhmWoPJRaQooBtlLF064\nR9ebYjupfFSaAoqB4uIzT7pwV4uuL+/Tniof1UngkBRQDBQXn/nKR7vK+q2k8lF5CigGiovPvMnB\nzSMv2ZLca5WPtjc1VdIVHIYCioHi4jNPcnBz/wXmJ5WPSlNAMVBdfOamC8ffbAyfGzqpfFSaAoqB\n6uIzN114mtVGaSuVj0pTQDHIafEZCRyKAoqBQOC6kjna7dPnL6omgb0M5Njmk8ATx6zSJnx13HkK\n64MPRNDes4c0I0W4RTEQCFy00nhHGiXZwkEIygeHooBiIBD4zCpNqyxSKXBPv0XngVvRLXrSpKo1\n11xNAqcxQNjmgVuRwGvHxuMXr9bWjieBvQwQtnngVvyaVFspOgoS2FMsmrm8LF5goCn2PCBMC8xt\nFstHGcvLksDK3WYxH8xaXpYEVu1WtcCefHD68rLu4pRAwghT7ORDhGmBuVU9q9KTD6blZZGmBbG8\nrCcfnL68rMUjSfcAABJfSURBVHuLZiCPF6eUc9vTFoj25INZy8tufouD13gHMrHl73BbhNvX31Ti\nFmH65utRuFW9xLunnXDm8rKbHyaoxh9frpNBS5DAnnbCmcvLEgoEYSNZhAIBCdzDQQL3cIgEToze\nav+WGYsmFAj4AieWFMccgTNj0ZtfXMPBC7wDmaiogNsi3CJMFbl9iXt2MG7ve/Y5CSxtCBD4UGmp\nIzA/Fs0ABToKJNChJ9+QTKTHondr2nISWIBCEzg9Fv3I4ME37j3AQRPvQCZa9sNtEW73tXEOAEP9\n8gzamqNgkD2B8zkWDTxlaFupP6wgFqfUXYFZsWgSWICCEpgdiyaBRSgogdmx6J4+6Q7BgIFCEVgA\nElgEEjgJEljIgDrdKZw8hWBQqHOyAgRuPWQCSPgQGx3tnAMMtHD2YxjIseUxwLhFMcjp2oV0iw5F\nAcWA1i7EMpBjm4cCyzUEd5ODZVYLh2JqZShNAcVAdUNwNznY2tDQsGNUFbUylKaAYqC4Ibg3OWhg\n4UydWhlKU0AxyF7pirk1ok2nVobSFFAMsle6YmBame62MnRLV4CEObUZUZSuYBjIseXWmCDcohhk\nr3RF1xtPtL+/VivDUMVnDFCgQ8hAcUNwb3JQn20mkaiVoTQFFAPlDcHd0hX93KeNH9TKUJoCioHq\nUKWndKUxttvYQa0MpSmgGChuCC4GCRyKAoqB4obgYpDAoSigGChuCC4GCRyKAoqB4obgYlArw1AU\nUAxUNwQXgpqRhqKAYqC6IbhYYLpFh6GAYkANwbEM5Njmn8CSMzrcfLCTCJZqZcgACSxkoHpGh5sP\ndhLBUq0MGSCBhQwUz+jw5IPtRLBcK0MGSGAhA8UzOjz5YDsRnNpDC2OFpYBioHhGhycfbCeCU3to\nabuwFFAMspkPthLBjPJRIGHO/4KuYCGD7LUytBPBjPJRuVNGAgsZiAUOkh+TD3YSwZnlo3KnjAQW\nMhBLKAhiAQV288FOIpjKR2UpoBioFlgIEjgUBRQDscATSeAMBnJs80xgTW3xGZWPhqKAYpDT4jPK\nB4eigGKguvhMCLpFh6KAYqC6+EwIEjgUBRQDxaFKMUjgUBRQDJTPi05lf7vvHtD7im061QdLU0Ax\nyN7SdgtPf7vllmEJqg+WpoBikL2l7Sbco+tNsZ1UHyxNAcVAcfGZJx+8q0XXl/dpT9UH18ycOW9/\npwkg4U42Dh7gHGBgP2c/hoEcWx4DjFsUA8XFZ7764K6yfivd+mASOCwFFIMs5oM3j7xkS3KvVR+s\n0y06JAUUA4HA9X+46S9azWNVMgJ7sr+b+y8wP6k+WJoCioFA4J8Pv37EnLPOHvywhMCefPD4m43h\nc0Mn1QdLU0AxEAh83l+0R+Jztd9cKCOwmw8+zeqTtZXqg6UpoBgIBP5SpVYZX6PVDJIRWAgSOBQF\nFAOBwJf8pG76gBmv3HUpCexlIMc2nwRefE78vCVfiZ+zRJnAlC4MRQHFQPSatOmleq2+YpPAghL+\nWLb5l/AXgm7RQts8cFtoLf2BptjzgDAtMLdZTBcyVh8lgZW7zWK6kLH6KAms3G0W04X81Uej/9tI\n4AlZEtiTLkxffZQEzobbLLYTTl999IG+fUvQ5aNAU2yVJcK0wNxmsXw0ffXR5u3bV+fL2oUM0PrB\nIIE96cJ8Xn2UARIYJLC3nTBj9dH7f8/B9b/jHcnAHTPApvNKwKa/v3UW2PTeG+FuS+eATe++He72\nuvm8I79YIoPHdwYJ7GknnLn6aOvHPPTeyD2UjltuApu+djTY9ONvzQebLj0H7vaMFWDTX18Nd3vE\nm7wju+C3AT7CRrL4OO4jsOmdM8CmO3vBGYz/C9i0chTc7bBXwKYPXQN3e8SncNsQiF7gyZ+ATf/2\nJNh07yQ4g3lVYNM374C7vf0dsGnF/XC3/9ECtw2B6AUm5BVI4B4OEriHQ4XA+xX4LDRgzoHS86VA\n4OoxUMtuTOwAbpsHbuHnAGeLhwKB207eBDNcfErswg+AThG2eeAWfg6QtniouEXPuxpk9np83b5r\nT4W9JCBs88CtDj4HaFs0ohb4z+Pe0Nu++CbE9O5f6hv6v7AAFDhG2ObeLeIcoGxDIWqBP/3Jsf+5\nfR5vlp0Pvyk1zpje+90As6o7Es1QW11vhJuibDFsoecAYxsakQq87IyBf9M//sGxU3q9B7D+sO/J\nL+jbTmgNMNvS75tXQW31LcdUQk1Rtji2wHOAsQ2NKAVeH39qqfkN/7DkiO8H2ZrfhfI+P39k8B8D\n3f409oOEDrTV7+tVCTVF2ELZVv41yRZyDjC2EohS4KnzjR8flhk/tp7QIDZNfhdeu+Ga54PdPlPb\n77YEyNa8DgzVQG4xtmC2y0reS7LVg88BxlYCUQp803TjR1Vfk/K0BWJT97sQBO1Xbxj3stsSQXbG\nBbH9KDN5cVevyihtTcDZbjDcOmyDzsF2hG14RCew9quHj1mv66+cajD+7PLVYmPPd0GMx/pcaIiw\npd+1/94sNlx25JN6ufGf3n56r+0BTjG2KLbrhi898kmbbeA5sChAbUMjMoFNJa48ZtafBz9ibCRe\nFp2Mup++vj71XRDjnV5b9IsNhd8Y8+cg03JbtX/9x/pAshhbBFu947RVpluLrfgcWONRoK0UohI4\nqcTPiv/fs4Gmq0d8s1flfc53QYwH+xx866tzez26CkDBUu0Llw9YEqkthq2u/+mi5K0hCPZ4FGQr\nh6gEhivx3tQOc3RTFfhdSOxJ6G9ObL9gXeKMgYHPP/NRbZ6u6htXBDLA2ELZGo/1P133oXEJrzdc\nbwkyTT3Vg2zlEY3ATy0CK7H9qNEJa/waZLjxjNiFnxofA/XtFwYSSD6qYRcExhbM1hgTb7u21w8b\nH7pa198KNHWf6gG28ohE4Peueg2qhHFb+vyDunnOAkY3lX0fem30reZkrOvP+muQT+dRDbkgMLZg\ntskx8bZre9/aZ2uA4brXDFPwU10aUQi8/aiiTqASZsotee0EjG4q+76g62tLDPPqqcvFph2eB0TA\nBWGYgm0RbK1oqmX47uQjHhP7PHjaK3sMU/BTXRYRCNxofM3v0wFKOCk3wN2x6lgzpnDH0V+4qzPI\n9LWLwY9q0xT+WIezTcYck4aNAbYtJ3/zW6Zp8BgkGsgLvOUY6AMtlXILvjtaAYCHLvh42xmXbgww\n/eSYNTrwAZE0hT5MoGxTMcfgs2COr+60Iq/qh882IriC7wMNWcy/LZVyC747Ggo/9JVm44K4/BdB\npjMu1YEPiKQpxBbD1o05Bn5xzfHVc5at+uGzDVmB7ZAu4G8zvgKI7Jyh8MlW7Cp4HPKvXuZFDnhA\nOKYAWxRbN+YY+MW1IqSWrfLhsw1Jgd2QbiBh4yJHZOd0UAC6zprs8uPxAHcIUxMIth8sfzQVc4T4\nhf1pUUH2CkaEdA1TcCbPRPBp6L7x3/5vZUJvPObtQGcI0yTAbB/v3f+BVMxRDCvrmFT4BSANacgI\nnIqnAkK6dvQIkp2r++nryV8Ap6Fh+nHnP33otikAtghTDNt3jn/zkL7zcXh80lJ4N4RFJJAQ2Imn\nQkK6iOiRFfxN/go5Dc1zTz3z3qP/CbCEm2JiXX+8VK+75HNDHwUMmuz4ZNbGVxYkBHbiqYCQLiJ6\nZAd/ETw6Hh0W+1uUpqhY19+PHNL/wT2DlgEGTU58MlvjKwsSAoOzpPq/4NEjcPC30mPRXQ+2DTLV\nUWzNp8nmlZ16wwmBlYfdbXr24pMehBXYeP7eDOW7bvgb4OgROPj7B8RVjrHFsLWfJj8u7fdfQaZW\nVCxr8UkPQgpsPX+hfDtOWwWMHrXuAIaqddBVHsoWztZ5mmh3vRxkakfFshWf9CCkwMnn7/eAfP90\nESzStOT4z496Ezi4qbjqWLBqGFs4W/DTRMfNxo8WIQWGP3+dNDgg0rT3hDdbfnzsZtDgpuaLL797\nE1A1hC2CLSKVCJ42rwBhb9Hg8UIqDS6GGfx967jPdP2Os0FhvEm/NH7cDlMNYQtkqyMSn8ZTvfTH\nqzAxvEiBFrjyy0+aGTzg89c8DaA0uBmG7/7Sb3X9QC9QK4wpPzB+7D4eNCkSbAtmi0glGkOxC8qu\n7zUOE8OLEmiBvzds4P96wDgRoPGCXXIZnAZPRrVrjpzf9c/jgrvOGELUHL1O19+4MngshrBFsIUn\nPvX3rthjfH+PehQ0yT96oAXe0K+1/IKTfrkHYuuWXIrT4KkZEWtPPbnvQ4Fuk68cvX5838DgSZEI\nWyhbE/DE5/ajTjbb3hXfGOhTDfDP4Ise1Tf0/fIxtwFCftCxozsjovPVHYFebSFevmZc8KRIjC2M\n7YdTR09vBw6aLNvyz//a+PUXCgsIhcAL/Ny5xt+WeP78e4IMn1oEHjtCZ0RYXuGvHDhbENtXTvnZ\nsl93w2oNbdvyI2e3vXs6ZGa3CuAF7h52vJnkCRxAv3fVa+D8L3RGhOUV/LVB2kLY7o6/ZH6Aqh1d\n28/3Ou+JQAaKABc4FdBdMDz47SiRnGoJy6iaaUfwjIiiTujXBm4LZ/vbycaP5/+913c6g1OJrm32\nZmBlAi5wKqDbcWpQEOD1r/QuabOmWkIyqsm0I2RGhHmRG14hXxvzOwi0xbCdPlXf/v+HPLz+rIUB\nVP22OVQYcYtOBeXmjhBfwpUnzK0aOQP8V9lpx8AZEZYQltdAIZKmMFsU283/dvbRdxmj4pL5ONvy\nI18CeFcBuMBuQLd5rtDQmrP+1NfApVXAsKctBMSrYwqyxbF9/XHzBfHdk95H2q5tB/wDFQALDA7o\nJues3zhTBw2awGnSlBDBXl3TYFscWxNtiw5Wng6MSWFsFQEsMDiga89ZP2D+HjhoAqdJPUIEefWa\nBjJAsbXw6hdj/aBPVIytIoAFhgd/nTnrAMDTpF4hojPVUWxtdLzfpcRWDWACY4K/7pz1YCDSpAgh\ncJrB2RYkQAJjgr8mIBO7cf3NdJQQOM2yOQ09+4AIjAnoJgE4Z6j+ZlCnIUzR1gUGiMAh5psA5qwj\n+pu1OU6VKZy1QoOsAyIw/EaaKkoAzFmH9zfbMvBD+5dgIT5Y3gI19SB7hQZZR6DAy84YOAd6I/UU\nJQTBnDwBvMwqJ570pQ+TvwYKYRYKAU0PEwQJbAWKgTVjiKKE5OQJUEXeulNqPviBo3AAkoVCiHXl\nej6CBE4GiqdC5psgppE6kycAFXn61DnGjx/BFLYLhdSuRFRYCBIYPj8WM40UM2z74W3Gjz19hzYB\nbJ1CIYjfwwRCgTFd/DBFCfBhm/GoftVk8I+J08dBGAALhQ4jiATGdPEDFyV8et/MF7ug77/Wo/r+\nY2b+cdAzB44LWBEjyQBUKHQ4QSAwuOegjihKWNv/2zec8o2PYcM2+1Fdfe13ntHbjv0QxABQKHRY\ngS8wvJATUZRQ2fcZ415aPLg5+P23y/OoTjy97rLrBLYJDwOCD4IrGDpmQhQlJDN5+qHLg9cS/ccZ\n77qP6h1fO+0O/mrh1vwNTFnE4QSewE5HVsiYCV6U4IQ2th0ZuJxb940D3oU9qu35G9CyiMMMHIHB\nXSqQRQm2woljIX1xBoAiLKn5GzAGhxvYAsO7VOCKEhyF1/UBRJtMhYMf1Z75GzAGhxnYAiM6siKK\nEiyYCu8YKF6EveqLl816fq+lMMgffP7G4YdMgc1BMbwjK6IowcaWfjcM+UWQyRm3fvnIs0r++xsw\nhZFzbg4rZAq8zLoYgR1ZEUUJKWzp94tgk9sSB+vmfefUGCBo0cPn3EiCcYu2brew7q3wogQvAJk8\nZzAWXMip9/AZGbLIENh4P7IUBnWpgBYl4IESjRTmI11g6/0IPGTC5JqQQCrcc+fcSCJNYPv9CKSw\n4t5tKNFo/gYPaQI770dTggfFynu3kWhRIE3g1PtR4KA4d73bCBi4An/88EvmrRa6/FHuercRMEgJ\nXHXSRX3MthuA9yN0UQIhd3AEruz7rP5O3Pwt+P0IX5RAyBlsga2QfeegGYtBA2LUouuEnMIW2Hzr\nTNx0+ayTZkH+EbwogZBrOLdoQ+HbJnbpa48JXEoOU5RAyDlSg6wt/YZ26frWYw8F/QtEUQIh93Bf\nk6xE7ZA/BP0DTFECIffwBDoAiVqd3n8LDd5IVlCi1pqzTu+/hQVfqFIc/U3OWV9H778FBUSvSnvO\n+iP0/ltIAAvszlmn999CAlhgxJx1Qh4BfotGzFkn5A8Q3WYRc9YJeQNMx3fInHVCngHV0h8WCiHk\nE3BrNgDmrBPyC8hFOWgiXKFBYoFoQiGABO7hIIF7OEjgHg4SuIeDBO7h+B8ycBLpkTeMUAAAAABJ\nRU5ErkJggg==\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -h 550\n",
    "ggplot(df, aes(variables, values)) +\n",
    "    geom_bar(stat='identity') +\n",
    "    labs(y='Value') +\n",
    "    facet_grid(BD_window ~ .) +\n",
    "    theme_bw() +\n",
    "    theme(\n",
    "        text = element_text(size=16),\n",
    "        axis.title.x = element_blank(),\n",
    "        axis.text.x = element_text(angle=45, hjust=1)\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAiYAAAD6CAIAAAAazxtsAAAgAElEQVR4nO3daUAT5/Yw8DNZgACJ\nCYvKIuAOKiAi0rp1cRfFpSgK2sVWr/ZWq5ba6t9etVqrleK92rrXWuu+lGrr8mLVtiAoinWtWMEF\nkU0IIUCALDPvh2lTyhKTMMxk4Pw+hckkcwjDnMwzZ85DUBQFCCGEUPMTcB0AQgih1gJTDkIIIZZg\nykEIIcQSTDkIIYRYIuI6gLr0er1Op2NzixRFEQTB5hYZxOvggefxY/AcYj9+kUgkFovZ3GKLZHMp\n5+LFi3q9XiRiLzCtVmtnZ8fa5pil0+lEIhFPjx0URen1ev7+G/N6z+F18CRJkiTJ5lHi0aNHMpls\n7NixrG2xpbK5lAMA4eHhEomEtc1VVlY6OTmxtjlmaTQaiUTC05RDkmRNTQ2bf2tm8XrP4XXwBoNB\np9M5ODiwudHS0lI2N9dS4bUchBBCLGH+LIeiqLCwsD179vj7+9d/VqVSxcbGXrhwYcCAAXv37pXL\n5YwHYL48FdzKhcoqsbMEeniBl4LDWCxWWQNXH4KyXGwnBm8X6OkNAv6c6hhIuJkLeaWETidyawN9\nfEHCqzGeXCX8ngeVVWKpIwR2gHYyrgOyREU1ZDyE0nKxvT34uUGAJ9cBWcJAwvUcyCslSFLUVg4h\nvmBviyM1qFFM/rkoijp06FBiYmJGRkZj68TFxUml0nv37s2dOzcuLm7Hjh0MBmCRazlw9AooHAEo\nIUHAr3/A+D4Q6sdVOJYpLof40+DiBHYURYgg4xE8KIaIYBDy4azVQMKP1+DWE3CyA4NBkFkIP/wG\nH4wBhSPXkZnnUjacuAEKR6BIIRDwcyZEh0OgN9dhmaegDP6XBC7OIAShQABXHkCoH4wKAl4MzeoM\n8OM1uJMHEjEBILiTD99nwNJIcGZ1gA01CZOHKJIkz58/b+LEhSTJw4cPL1iwwN3dfeHChUePHuWq\n3U6VFg6ng5vzn8dogQDcnOG7K1BRw0k4FkvLgiDq0cCcE1Mur5166ZOBhT9lZ6vv5HEdlnlu5sKt\nJ+BsDwBAEOAgBncZpN3jOizzlFXBD9fA1enPc0qhANycYV8a1LBaZWm9i9nQVgZ2QgAAggCZBK48\nhKxCrsMyz2+PIDMfnOz/TJAOduDmDKlZXIeFLMHkWY5QKNyyZQsAbN26tcEVVCqVWq2mB9y6deum\nUqnKysroFJWenn7+/HkAcHR07N27t8FgYDCw+h6XEA5igs53FEXRmc9RDDnFVPf2tt50jqQg71p2\nZO5RtYO8WOIiIAhvZWbPvLSrPgsDPHjwfe/RU8Lxrw8fACiKEhKQfBdG9CI5jcssOcWEpN6eIxFD\nrpLyc7P1PadKB5cfCNydgaJq7fZ28OAp1cnd1oMHgCfKPz986i8iIZy/QwwJaN7DBQCQJA92Tl5g\ndRyULvmg62ScnZ0BoKSkhE45KpXq/v37AODj46PT6YRCYbNGUl0jAEpMkhQAUBRF708UENU1Oq3W\n1vctnQE6ld0tFDtW6coJICgACihS5Ch/fEsbGMR1dM+mN4hIUkB/Uf37w6eIqmqt7Q8M1miFQInq\n7DnAkz2nWksAaVc3eAq0OlKr1XMcnBkMBhFJ/rmL/L3nkIRWq23+TTd7VmslWE05dHbRaDQymayi\nogIAFIo/L9kPHz58+PDhAJCSkuLg4NDchbPe7qA1gOyvvEZnOK0BOrjb237JrkSv71F+OVcoExLG\nIzRRSlKOuicSSTiXkZmnvQKyisFB+OdRQygUUhSEdwFnJ5v/6AG8XEFLgjNP9xwJGCgQ/jP46hrw\nchVKJDy4O8pFCg+UYCf88yxHIBCQJPTvBizU2fP37jFbw+q3SoVCIZPJsrKyACArK0smkxlTDstc\nnOAFf1BX//kjRUF5NQzsBu5STsKxDCUgNGSZgLIDIIyLhJRYZZ/LaVzm6u0DpZWg/etbI0lBcTmE\n+HIak9k85PB8Z6ioBuPAoLoKhvQAmc3nG9prA6G4Asi/gq/Wgb8H9ORJ7UNoR1BWgO6v8zE9CcUV\nvCn5QTQ2Us6RI0fUajUACASCSZMmbdq0qbq6evPmzZMnT+bwHsaXe8C4ECgog6JyQaEaIoJhSA+u\nYrFMFaU77ldBOTymhFWEwZ4g7UlxucTuwX0+DMcDgJM9fBQJIT5QqCYK1YIu7eDfQ8HPjeuwzDY8\nEEYGQaEaisoFPq4wPhRebOB2ABsV4AkzX4BO7lCoFhSWQXhnGB38ZzWB7WsjgSVjIcgHCsqIQjXh\n7wHzR/Ds3gbExsDapEmT7ty5I5PJACA+Pj4mJsbT07N///579uxhYeuNERAQ1AGCOkBlpYZft2FL\nhPY/eGuicpxKHZ9WSgtFIJDVQLI3pfPhzR0WTvYwtCe8HMDL7gMCAkL9INSPf3sOzccVfFxhaHde\nBi91gOG9YEgAB90HECOaJeXUKX2u/aNcLj958mRzbLT1IICY0n3yLKeklwol7csJrUiQ5QKnnQv3\nu4RyHRpCCJmCd+7y0uueESp9+QE44+QhAYIq0ZV/1vXfPZ07cR0XQgiZgimHlxwEdh/6vTqx7YtZ\nZY+lDk4Bzn5uYi5bByGEkDkw5fBYN0cfb3Djbydp/qLy88isu4Sm0uAsFXTvQbi5cx0RQvyAKQch\nyxiuZxjOnyFkbQQURQIYLl0QRYwXdOdJvSNCnLL5u70RsiVUudrw02lCrgCBAABAKCQULvofjoKW\nJ+35EOIUphyELFGQTzj+s+U1QYCDhCws4CgghPgEUw5CFqBIQwM33BIEgW0fETIDphyELEC4uBL1\nm0hWVxOu/GmfgBB3MOUgZAHCvZ2gd1+qsuLvJmsV5cIBL4AzH9rzIcQ1rFhDXNDWGK5lUBXlgox0\nQ3h/QVAfQs6bVlnCQS8Rrm76U8eAIonefQW+nQTdArgOCiF+wJSD2EbVVJPnz5A5DyixHeXiSt67\na0hPFb/2L8K9LdehmUcgEPQKtusVXFlZ6cDDNmV8RxXkkbk5hE5HenoLfPz4MYc2+gumHMQ2KvN3\nw8NswkHy59iUUEi4uBmuZ4iGjuI6NGTTKNJAXvjFcDUdHBwBKP3li8IevYSDh4CdHdehIXPZXMqh\nKEqv1+v17M1RSJIkm5tjFh08v7oPkIX5YO9A1QIEQV67AoOH/HmzC0+0gD2H6ygsQ925Rd34DWRy\nerch7OwN2X+Qjk5Ev/7NvWmciJopNpdyAEAgEAhYPPQQBMHm5phFB8+vlEMRBH3lnSCIvyOnKIFQ\nyK9Bkhaw53AdhWUMBXkgcQQAgiD+7E9vZ09d+EX43MDm3jS//sVsmc2lHPo/AVOOmfiYcqBte33W\nXULiSB81CIKgSIOwd1+BkCczhQHcqMg6q7yiripXOLYZ5fp8V8cOXEdkMT7u9hRBkAIBAFAUZfy+\nQhHAwi/Cs38xG2ZzKQe1eERAT0FBHvUgC+wdAAD0OigtFUZGcR2XuQ4UnPn04Tee9u4ECWQltTX3\n+8+7zhvp9hzXcbUC9g6UQU8Iax21KErYuy93ASGL8exrDmoBCDt70ZCRwucHC7w6gIeXICBQPGc+\n4caPcrWCmpKVD3b6OLQXEUIAEAtEnSSe7/6xvsJQxXVoLZ+gZxCoSoH667IKRUGZStCVP9OAIzzL\nQdwQiQTBfYAkqZoaIa8mor5Zma0Qy+osbCN0vlP5IEyGzaSbF+HiKp76huHmVer2TSCA0ulFr0wl\nfDtyHReyAKYchCygI/X1B/UFBKGjDBxE0/oQnl4iTy/DwJf1Wq2dgje3DyMjHFhDyAKdJF71x9Aq\nDJWdJV6cxNNKOThQvDo5RkaYchCygL+Tb0z7ESXaMvpHCqhCrfKdDpPb2blwGxhCvIADawhZZoHP\nlM4Srw+zviRJUiAQbOj23nC3flwHhRA/tOqUQ6lKybu/C0qKDQoXQfcehIsr1xEhHhARwoltX5zY\n9sXKykon7LGGkCVab8oh/7ij++GooI2cAIJ88tiQliwcOVbYI5DruBBCqMVqrddyqqt1x44IXNxA\nJAaCALGYcHHVnzxGaSq5jgwhhFqsVppyqMJ8wtGxTkcvQuJIFeRxFRJCCLV4rTXlGPQNdJAUAGHA\nfrEIIdRcGE45KpUqIiJCLpdHRESoVKr6Kzx8+HDIkCEymWzYsGGFhYXMbt18hIsb1NTUXVpdA65Y\nQcAeA0VqSR3XUSCE2MNwyomLi5NKpffu3ZNKpXFxcfVXmDZtWmho6JMnT/z9/efNm8fs1s1HyBXC\nsOf/ceWmSiMMDSdc3LgKqVXJqyn+7OGeXhdj+117a9n97RnqTK4jQgix4a9pJ5hAkqRCoUhKSgoP\nD09PTx8xYoRSqazd9FupVLq6uhYXF7u6uhYWFvr4+FRWVopE/6iaS0lJCQ0NlbBwa7HBQN66rk86\nQQEQBAiHjBQG9QH+9M+naTQaiUTCr87qxTrV8+kzO0o87QgRSZKEkHhUlb8tYEl/Oc/KBXldJM3r\n4A0Gg06nc3BwYG2LKSkppaWlY8eOZW2Lz5SZmRkQENDYAZwgiDt37vj721zPUyaLpFUqlVqtpn/J\nbt26qVSqsrIyuVxuXEGr1QJAWVmZq6urWCzWarXFxcXt27cHgOrq6qqqKgCoqT/e1UyEQkFwH7vg\nPrz+3+OjE8Wp3g5t7QVi+r9FAAJfB49TJWm8SzmIE/k1xdfV96p1NUGKrp1aYp8h07mE75hMOaWl\npQBAH76dnZ0BoKSkpHbKadeuXUBAwPbt2xcvXrxq1SoAqK6upp/auHHjokWLACAmJiYkJITNj1ur\n1fLrLKE2rVZLT1fFdSAWyFI/loCdwWAAAIqi6AeHCn56v/1UekYAvuD7nsO74CmgDjz9KT53Xxuh\nM0EQ5Q81r7Yd+S+P8WKi2e8v1OnYu+golUqjo6NZ2xzLmPxT0dlFo9HIZLKKigoAUPyz1StBEPv3\n758xY8aWLVvmzJkDAG3b/jlLyvvvv//+++8DQEpKikQiYWNg7S8URTk6OrK2OcbxbmDNXmwnEAoI\nICiKIklSSA9mEoTUyZkAPv0ivN5z+Bj8OeWVzYWJXZw6UBRFUZSbvfx46QUPJ/fpHqOae9Nisbi5\nN2Hk5eV14MAB1jbHMibLBxQKhUwmy8rKAoCsrCyZTKao111cLpenpaWVlpZGRUX5+vrybqdHTddH\n1r1EV1Z7SQ2pfdNzLL/yDWJfqupmO/E/akoVYunKBzu5iqdBkydPDgsLM/5YWlpqb2+/YcMGALh2\n7dqoUaPc3NwcHBwCAwMPHz5sXI0giPPnz7/00kvh4eH0j5mZf9bUmHgVANy4ceOFF16QyWQ9e/bc\nsWNHg+NDJElu27YtODjY0dGxS5cu//3vf42r3bp1a/To0S4uLm3atBk+fLhxo82HyZQjEAgmTZq0\nadOm6urqzZs3T548mf72feTIEbVaTa8zZcqUjz766OnTp6tXr37ttdcY3Drii2Gu/SLcBhRqlXrK\nQAJZpq94UJU3zWMk13EhW2cAUlDvhJ6wsZsLp02bduXKlfv379M/Hj58mCCIadOmkSQ5evTohw8f\nfvLJJwcPHnzuuediYmLKy8uNL3z33Xdfe+21ffv21X63Z74qIiJi1KhRe/bseemll2bOnHnw4MH6\nIa1bt+69996bOnXq999/P2/evNWrV69fvx4ADAbDiBEjFArF5s2bt27dKhKJ2DgmU4wqLS0dNWqU\nQqGIiIgoLS2lFwLAnTt36MfXr18PCgqSy+UzZsygr0PUkZycrNFomI3KtIqKCjY3x6zKykqSJLmO\nwmIkRZ4rufLZg2//k7n1UMFZjaGa64iswes9h4/Br33w7ciM+WN+ey/i6sLRGQvG/PbemN/e+0/2\nNhY2nZycfPz4cXPWrKmpcXV1Xb16Nf3j4MGDY2NjKYpSqVQffPBBeno6vZz+Fm48MALA9u3bjW9i\nfMrEq+7cuQMAX3/9tfFV8+fP79evX513MBgMzs7OW7ZsMa52/Pjx559/nqKox48fA8C9e/fo5UVF\nRTt37rToY7ECw5fd5HL5yZMn62c14+OgoKDr168zu1HEOwQQL7mEviAPqampYfO6HeK1Ea7h3+af\n8pN4GJcU1JQ8J+vFYUj12dnZRUdHHzx4cPHixTk5Ob/++uvPP/8MAG3atFmzZk1WVtaRI0euXbuW\nlJRU54V9+vSp/27PfFXtuu2JEyfu3Fl3mPHx48cVFRWzZ8+ePXu2caG7uzsAeHp6RkdHh4WFjRs3\n7qWXXpowYcIbb7xh/W9uHts6J0UIocYES7smdHv3D01OgbakUKu8p8ld6Dt1pNtzXMdV1/Tp069f\nv3737t19+/Z169Zt8ODB9PK4uLhBgwadOnWqS5cuO3bsqPOqxi5sm35VbRRF2dvb11kolUoB4NCh\nQ/m13LhxAwAEAsGBAweuXr3apUuXvXv3+vn5xcfHW/crm6/1Tl6AEOKdYa79fgvf/UdlTpW2uoe8\nUxuRM9cRNSA8PLxz584HDx48fPjwW2+9RV/SViqVCQkJubm5np6eAPDo0SNz3uqZrzp+/Ljx1OTo\n0aNBQUF1VnBxcenQoUNGRsakSZOML0lPT1+1alVpaenHH3+ckJCwdOlSADhx4sSUKVMa7BrDIEw5\nCCE+cRQ6BDp11tnpHETsdR+wCEEQ06dP37hxY1lZmfGCvL29vVgs/vrrr4cMGZKTk/Ppp58KBIJz\n5875+fmZaKNg4lX0CitXriwqKgoICEhKStq0adOZM2fqv8ny5cv/9a9/iUSiAQMGXL9+fe3atZs2\nbQIAmUy2Z88etVo9cuRIAPjmm2969OjB+KdRV3NfLLIUlg9YhKflAzSDwcDy35pZvN5zeB28Xq+v\nqqpic4vmlw/Q7t27BwBRUVG1Fx44cKBjx45SqfSFF15ITU398MMPZTLZw4cPqX/WWNX5sbFX0eUD\nt2/ffvHFF6VSaXBwcGJiYoPvQJLk7t27g4ODJRJJ9+7da9cpJCcnh4eHOzk5yeXyMWPGZGVlWf7Z\nWIbJHmuMYK/H2l943fCGjz3WjEiS5HX5AK/3HF4Hjz3W+AvLBxBCCLHErJSTmZnZr1+/r7/+GgDW\nrVsnlUrDw8Ozs7ObOTaEEEItilnlA3PnznVzcxszZoxarV69evWBAwc2b968YMGC48ePN3d8CCHE\nF2VlZXR7SYvIZDK6lLk1MOss59KlS6+++qq7u/t3330XFhYWERHx6quvpqSkNHdwCCHEI0+fPqUn\nYTGfRqMpLi6mHxN/EYlEQUFBZ8+eNfHCzMxMdi7i1m741nRmneUIhUKBQAAA58+ff+GFF+iFbHbz\nRgghXlAoFK6WzGdfVFRUWfn39MRXrlzp3r17VVXVzp07o6KiCgsL7ezsmiFMzph1lvPcc88dP348\nKyvr+PHjMTExFEWdOnUqMBAn1EIIISY5OTk5Ozu7u7u/8847KpWKnoQMAA4fPtyjRw8HBwdPT891\n69bVeVX9Z+lzoL1790ZERLRr1+6jjz4CAJ1Ot3jxYi8vLxcXl9mzZ+t0Ooqi/vvf/3bu3Fkmk0VH\nRyuVSgCorq5+991327Vr5+fnt2XLFmZ/QbNSzmeffXbu3LmuXbuOGDGiY8eOCxYs+PHHH+v/2ggh\nhJquurp6165dL774Yrt27QCgsrIyNjZ20KBBBQUF9GyWRUVFxpVNPJubm3vixIkjR46sWrXqyZMn\n8fHxSUlJaWlpqamp58+f37Zt28GDB7dt23by5MkHDx4AwOuvvw4Aq1atOnPmzLlz51JTU7/77jtm\nfzWzBtYCAwMfP36cn59PN11YtmzZ559//ufMWkyjb1wiSbI53tzEFlnbHLPo4Pl7X04L+PC5jsJK\nvA6e/T2HtfsXhwwZIhaLdTpdQUGB8VqOSCS6evWqn5+fk5OTl5cXANCnI898dsaMGQAwcOBAACgv\nL9+1a9eaNWt8fHwAYPfu3RUVFatXr162bFn37t0BYMOGDb6+viRJ7tu377PPPuvZsycAxMfHBwcH\nM/gLmtvwRigUent704/rT7zGLOPds+xgeXPMagHB8z1+rqOwUgsInuWjBDsbOnv2rL+/PwDs27dv\n4sSJxcXFAoHA3t6+pKTkgw8+yM7Opp+tzcSzbm5uAGD8SpqTk9OxY0f6MT0X3FtvvTVlypQpU6YY\nX1JUVJSXl9elSxf6x65duzL7C5qVciiKSkhIOHr0aFZW1o0bN7744otevXrVjpJBBEEIhcJmOoVq\nkEAgYHNzzKKD5+lZDkEQLeDD5zoKK/E6eAD4ewpzVtD1U2waMWJEaWlpXl6et7d3dXX1qFGjvvzy\ny+nTpxsMhjptF0w8W+fI4OHhkZub27t3bwC4fPlyTk6Ou7v7xo0bx4wZAwAGg+Hp06ft2rXz9vbO\nysqiVzPONccUsz7H9evXf/bZZ0uWLHn69CkA9OvXb/bs2fUnZkAIIVSXXk/l5pB/3KGePIZnDQZW\nVlZWVFSUl5fv37/fw8PDw8MDANRqdVVVVUBAQHl5+apVqwCg9t0/pp+tLSYm5uOPP378+PH9+/dn\nzZqVnZ0dFRW1atWqR48eKZXK+fPnR0VFEQQRGxv7n//85/fffy8sLPzwww8Z+xwAwMyznC+++GLp\n0qV0JgSAyMjIDz74YN26dfRAIUIIoQZRRYWG39LJe3dBKCIMeqJrd2FYf8Kl0Srqvn370g8CAgIO\nHjxIn8m1bdt25cqVo0aNkslkb7/99owZMyIjI8+dO0evafrZ2pYuXVpeXt63b1+9Xj958uT58+cT\nBKFUKvv3769Wq1988cX9+/cDwOLFi5VK5Ysvvujo6LhmzZoff/yRwQ/ErLaeEonk+PHjw4YNIwgi\nPz+/ffv2J0+enDRpUu1ycqaw2dYzp7owqeRSfuVTDyf3IS59O0o8Wdgos7CtJ4d43RmT18HbbFvP\nrKysf9yXo9dr168m3NyB+Gs8iSQJrw7CYaMIwZ+jgvR9OcZLLC2eWQNrXbt2/e2332ovSUlJqX8V\ni19+Ul4e+dv8g4VnUstvHSz8acy1906XXOQ6KIRQy0E+yQFHp7/zDQAIBOTd36miQu6C4phZA2tv\nv/32+++/36ZNGwBITk6+cuVKQkLCrl27mje05lSu1/w7M76bYwcAMBgMDgK7ThKv+XfXX+z3ldwm\n5xlECPGPRkMI6x1jhULQaLiIxiaYlXJmzZqlUqnoCUonT57coUOHbdu2xcbGNnNszehWZbZCVLeP\nnkIsu1F+b7AihJOQEEItjZMzpdcR8I8BQMJgIJxb7/das1KOQCD48MMPFy1a9PjxY5lM1tz35bCg\n2qAVEHUHFYUEUU1qOYkHIdTyCLw6QJWGkjgSxnpug17g35Nwb8dpXFyyoNhcIBD4+vq2gHwDAB0d\nPSv1dc9tK/RVfKwgQAjZKKFQPGue0LcjVVZKVVZQqlLCr7Og/2AwWe8zbdo0giBMt5HmL7POchrr\nXM3fCgI/B4/pnqMSi35xEcvoJaV69dT2w7s6duA2MIRQS0K0kQuHjBT0CYOKCpDKCIWL6fXVanVi\nYuLQoUO//fbbIUOGsBMkm8w6ywloRHMH16zmdZg8z2fyXU1OdtWTu5qcOV6vLPBtln4KCNmcykri\nyWOqqIAy6LkOpRUgCMLFjfDxe2a+AYCjR4+GhoZ++umnR48eNd6FUr8DdP0ldabPoSe5oRceOnSo\nXbt2mZmZ9RtO13+f2bNn05ftASAjI8Pd3Z3ZeWrMbXhjfFxRUZGcnLx69erdu3czGAf77ATi6HZD\no9sNLSkvdZW2hNFChJ7NYDCkpxrSkgUikQ4AaqrFMW8QXnhybyu++eabWbNmhYaG+vr6fv/993SV\nlrEDtEajGTdu3LZt29RqdZ0lJk6Jzpw5c+vWLUdHx6CgoDfeeCM1NfXs2bNRUVGvvfbaV199Ved9\nxo8fP3fu3HXr1tG5Kjo6WiwWM/gLmtvW08jZ2XnUqFFlZWVvvfVWyxhtdBC0qBmQEDLBcDmNvJ5B\nuLiCwUAIhUDJ9Pu/Ef9rHkhlXIeG4OHDhzdv3nzllVcIgnj99de//fZbOuXU7wD99ttv11li4m2X\nL1/u7u5eU1NTv+F0/XceNGhQUVHRnTt3AgICDh06tHfvXmZ/R4tTDs3b2zs9Pb3+cpVKFRsbe+HC\nhQEDBuzdu1cul9dZ4fTp04sWLcrOzg4ODt64cWNoaKh1ASCELGYw6JPPC9zb/r2EIEAmM9zLFPbp\nx11Y6E/ffvutUql0dHSkfxQIBPn5+R4eHvU7QNdfUvuKe52eMnSjtgYbTtd/HwCIiIg4duyYRqOh\nKOr5559n9nc061pO5j9duXJlxYoVfn5+9deMi4uTSqX37t2TSqXGAUGjoqKiqKio+fPnP3nyZMKE\nCePHj+fvjB0I8Q5VXUWI6n7LpEQiqlzNSTyoNoqidu/efezYMePUDP369du3bx/81QGaXu3y5ctH\njx6tv4R+rNfrAcD4FI1ug003nI6Kirp169bBgwfppxp8nwkTJhw7duzQoUMxMTGMN9OypnwgLCws\nOzv7yy+/rLMaSZKHDx9esGCBu7v7woULjx49WifZZmZmSqXSGTNmyOXy2bNn5+bmFhQUMParIIRM\nIuztgTTUXWggCYkjJ/Gg2lJTUwsKCoYPH25cMnHixG+//RYa6gBdf4mLiwsA7Nmzp6am5rPPPqv/\n/g02nK7/PgAwcuTIGzdu7N69e+rUqYz/mhaXD5igUqnUajV9ytatWzeVSlVWVlZ7bC0oKEiv1+/c\nuXP06NHbtm0LCAigz/gQQmwQiYVhz5N3fwfx39cvKXWZoHM3DoNqSbRa7ePHj588eWL+S0iSpDvb\n7t69e+zYsbV7lb7yyiuLFi26efNm/Q7QJEnWWWJnZ7dmzZr33ntv5cqVa9eu/eKLL+psqMGG0/fv\n36/zPgAglUqHDBny6NGjwMBAJj6VfzCrk7SZsrOzu3TpotPpRCKRXq8Xi8VZWVmdO3euvc727dtn\nzZoFAARBXLlypU+fPvTy48eP79mzBwC6dwcnnqsAACAASURBVO++YMECNnvE1tTU2Nvbs7Y5Zmm1\nWrFYzNNO0hRF0fsJ14FYiZd7Tk01cekCkXmLFIkFAESVhhwZSfEt5ZAkaTAY2NxzLl26VFFR8cxO\n0nfv3rW3t3e2pJ9NeXm5Xq9nfPLNJpoxY0a3bt0YnywHTJzlNHb7Z211bgWlT2g0Go1MJqMrKOq0\nKjhz5syqVatOnz7du3fvLVu2TJ48+d69e/Th0tfXd+jQofRq9vb2bKYckiTZ3ByzKIpycHDgacoh\nSZIgCP5++LzccxwcYHgEFdynJj9PLJOBhxcfR9UMBoNer2cz34vqXQNrkFAolEqlf09eYAaSJJtj\nFhir6XS6/Pz8U6dOrVixojnev9HP0Zw7PeucISkUCplMlpWV1adPn6ysrPrd2E6cOBERETFixAgA\nWLhw4fLly/Pz8z09PQEgODg4ODgYAFJSUgQCAZvTvtLTIbO2OWbRwfM05UCL+PC5jsIqHl6UTC7k\n7Xw5FEWx/OHz91/MUmlpaQsWLEhISOjQoVnu1mr0b0aZoe57CQSTJk3atGlTdXX15s2bJ0+eTP+d\njhw5olarAaB///7Hjx8/c+ZMUVFRQkKCl5dX+/btm+O3QgghZIXBgwdnZGQ0R+EAzcqvCdXV1Y8e\nPaq/PD4+Pi8vz9PTs7CwkG6oAACTJk3Ky8ujH8TFxc2ePdvPz+/06dPff/89X78kIoSQefSU4UZ5\n1k/Ky7cr7lNg6tp5bm7u66+/7u3tLZFIunTpsnjxYtP3eJqjdiMcugtOnYXmrM8gc28FLS8vr12G\nce7cuSVLlqhUqjqryeXykydP1lloPB8iCGL+/Pl0UQRCCLV4WZrcXfknThdftBeKqww1EW79/+U9\nwdu+bf01KYqKiIh47rnnUlNT3d3dMzMz58+fP2fOHLpO2mqenp6HDx82Z+Ezn2KEWScZhw4dUigU\ntW/NmTt37ttvv918YSGEEN9pKd3Iawsy1Hc97F1dRDIve/d09e9fPj5CQgO3wOfl5d24cWPFihU+\nPj4SiSQkJOTLL780GAwAQFHUf//7386dO8tksujoaKVSCX+djuzduzciIqJdu3YfffQR/T7fffdd\njx49JBJJQEDA6dOn8/LyJk2aBABhYWEAMGTIkIsXLxoXRkRErFy5kn5hZmamg4PD7du366/PYK9P\ns1LO8uXLZ8yYoVarw8LCbty48fDhw6CgoGfWCyKEUGuWoc50FUlFtWaDtCPE/6/k0p2Kh/VXdnNz\n8/HxmT59+o8//lheXg4AvXr1orsPHDx4cNu2bSdPnnzw4AEAvP7668ZX5ebmnjhx4siRI6tWrXry\n5IlarY6Ojv6///u/wsLCadOmzZw507jm5cuXAeDs2bPPPfecceHkyZMTExPpx4cPH46MjDTWfNVe\nf/z48XRbBABoYq9Ps1JOdnZ2RESEVCodPnz41atXfX19P/zwwyVLlli3SYQQag1KtGp7Yd2uwQ6E\nXbGurP7K9vb2N2/eHDp06Jo1a9q3bx8aGrp161b6ZGL79u3Lli3r3r27q6vrhg0bkpKSjK3CZsyY\nAQADBw4EgPLy8pqaGpIkVSqVWCz+8MMP79y5YzrCcePG3b59++HDhwBw+PDhV199tcHVXn75ZbrX\nJ0VRdCMciz6H2sxKOU5OTkVFRQAQHBycnJwMAD4+PhkZGVZvFSGEWjxXO1mNoe7c9tWU1k3cpv7K\nJEna2dktWrQoJSWlpKTkP//5z5YtW1555RUAuH///pQpUwiCIAiiffv2NTU19AEZANzc3KBWDbe7\nu/uPP/546NAhFxcX+sZH0xHK5fLhw4d///33mZmZBQUF9B0s9dnZ2dG9PjMyMprY69OslBMWFrZ+\n/frLly+HhIScOHEiPz//7Nmz9K+KEEKoQaEy/xJ9uZ76+8qNjtSPcA0PcParv7JarZZIJHSTTQcH\nh3HjxiUkJJw/fx4A3N3df/jhB/rWFL1en5+f365dO/pVdQrPqqur27dv/8svvzx8+NDX13f06NHP\nDDI6OjoxMfHw4cOxsbEmhsuY6vVpVspZu3atSqU6evRoly5dYmNjvby8Vq5c+emnn1q9VYQQavHs\nCPHp3utDZd3za0qUevWTmqf92vT4d4coQUMHXrlcPmzYsOnTp6enp5eUlNy+fXvDhg10T5aoqKhV\nq1Y9evRIqVTOnz8/KiqqsYO+Tqd7/vnnz5496+Tk5OHhYWdXd1ivftV1ZGTklStXvv766wZH1Yzr\nM9br05xbPimKMhgMKpWKflxSUlJRUWHmCy2VnJxMz9PAmub7XVhQWVlJkiTXUVjJYDCw/LdmFq/3\nHF4Hr9frq6qq2NxicnLy8ePHn7navXv3iouL6yzUkfrr6ntnStJvlWeTVN3/1sLCwvv379OPnz59\nOnv2bG9vbzs7O19f37lz5yqVSoqitFrtBx984Onp6ezsPGbMmJycHIqi6Os0xvcBAPpay86dO318\nfOzt7QMDA8+fP197tdjYWJlMduHChTqvHTduXM+ePekjSYPr0z+OGTMmMDDQzE+sMWa19dy4ceO0\nadPqdK9pJikpKaGhoXRrVXZUVlY68bbzh0ajkUgkPO3GQZJkTU0Nm39rZvF6z+F18AaDQafTsdng\nLiUlpbS09JllullZWQqFwqIea0VFRZWVlcZ50mwZI70+zRpYi4uL8/DwmD59+q+//mpOikIIIdRi\n6HS6nJycU6dO0RNjN4VZKSc/P3/Dhg05OTkvvPCCv79/fHy8sV4CIYRQy5aWljZhwgRGen2a1fDG\nxcVl1qxZs2bNysnJ2bdv3zfffLNkyZLx48cfOnSoiZtHCKEWw2AwFBcX0zdymqm6utrMmRE4RPf6\nZOStLPtVfXx8xo8fr9VqN27c2Kx9eBBCiHcMBoNAIDDep2kOiqL0en3zhWRrzE05WVlZhw4dOnTo\n0PXr17t06fLOO+80fVAPIYRaEjs7O+vKB5ovJFtjVsrp27dvRkaGq6trdHT0li1bwsPDm69EiqIo\neirrZnr/+ujqF9Y2xyw6eP5WrLWAD5/rKKzE6+BJktTr9WzGT7fXRE1n1pG9U6dO//nPf0aOHFn/\nxiLGEQQhFovZnNVcq9WyuTlm6XQ6sVjM35RDkiR/P3xe7zm8Dt5gMFAUxWb8QqGQtW21bOZOXhAZ\nGclCvkEIoRapxozrNc09Rdubb77p5uaWnJzMgynaEEIIWapGD6n34MxtEBJgIGF0MIR3BnFDp0xU\n80/RtnPnzitXroSGhqrVapueog0hhJClKAqSbkL6fWgnAzcptGsDyX/Aud8bXtnEFG30mcfmzZvD\nw8O7du06b9686upqqqF523Q63eLFi728vFxcXGbPnq3T6YyzsdHT5ERGRqakpBgXmlifyynaEEII\nWSq/DDIegkOtS06OdpByD0obqlAzMUUbLT09PS0t7fr169euXVu7dm2D87bFx8cnJSWlpaWlpqae\nP39+27ZtxpdfvHgRAM6ePUtPrkMzsX4zTdGGA2sIIdQsisvBvt6R2UEET8tBUa+/HT1F2+bNm9es\nWRMdHe3v7z9r1qwZM2YYD+7/93//JxAIHB0d33///ffee69Dhw70vG0AsGHDBl9fX5Ikd+3atWbN\nGh8fHwDYvXv3My8Fmbm+cYq2gICAQ4cO7d2719KPwgjPchBCqFmIhVD/rlCSavhajokp2mje3t7G\nB48fP25w3racnBxjh9Dw8PAhQ4aYjtDM9dmeog0hhJClOrhCZQ3U6YRcWQOe8gZWNjFFGy07O5t+\nkJmZ6enp2eC8bR4eHvQ7AMDly5ePHj1qOkLz12d1ijaEEEKWcraHmOfhaTnU6IEkoUYHhWp4c3AD\no21gcoo2WlxcXEFBwe3bt5cvXx4TE9PgvG0xMTEff/wxfQ40a9YsY5ZqzDPXZ3yKNryWgxBCzaWX\nN7w/Gn7PA3UVtJFAoDc4Nz7Lz759+z766KNXXnmlqKjIw8MjMjJyxYoVxmcjIyP79etXVVUVHR29\nZMkSkUikVCr79++vVqtffPHF/fv3A8DSpUvLy8v79u2r1+snT548f/78+/fvmwjP9PqxsbFDhgw5\ndepU//79pVLpkCFDHj16FBgY2JQPxKwp2tiEU7RZBKdo4xCv9xxeB9/apmjLzMwMCAjg/FjN3hRt\nCCGEWi22p2hDCCHUarE9RRtCCCFz6PX6mpoai9Z/5jr+/v7cjqpxNkXbM6lUqtjY2AsXLgwYMGDv\n3r1y+T+KAbds2TJnzpzaS6Kjow8cOMBsDAghxAmxWJyfn5+fn2/Rq9zc3JopHhvEcMqJi4uTSqX3\n7t2bO3duXFzcjh07aj87bdq0MWPG0I8piho3btysWbOYDQAhhLji6+vr6+vLdRQ2jclrOSRJHj58\neMGCBe7u7gsXLjx69Gidk0FnZ2fvv5w9e3b48OEvv/wygwEghBCyZUye5ahUKrVa7e/vDwDdunVT\nqVRlZWV1xtZoSqVy/fr1qampxiV37969ceMGAFRVVYWGhjIYFUIIIRvBZMopLS0FALrY39nZGQBK\nSkoaTDmLFy+eM2dO7dsCbty4QTcxDQkJGTNmDJvXyrRaLU/vawEArVZLURRP46cbdXB+t4HV+L7n\n8Dd4egpzsn7/smbD30m7bQ2TKYfOLhqNRiaT0W0SFApF/dXy8/OPHDmSkJBQe+GkSZPoSRpSUlIk\nEgmbtwdSFOXo6Mja5hiHt4Jyhdd7Dq+DZ/9WUP5O2m1rmLyWo1AoZDJZVlYWAGRlZclksgZTzldf\nfTVx4kT+3vmMEELIOkymHIFAMGnSpE2bNlVXV2/evHny5Mn0t+8jR46o1WrjakeOHBk1ahSD20UI\nIcQLDHcfiI+Pz8vL8/T0LCwsXLduHb1w0qRJeXl59OP8/Pzr168PGDCA2e0ihBCyfQzflyOXy0+e\nPFlnYe3rwx4eHvy9XIwQQqgpsMcaQgghlmDKQQghxBJMOQghhFiCKQchhBBLMOUghBBiCaYchBBC\nLMGUgxBCiCWYchBCCLEEUw5CCCGWYMpBCCHEEkw5CCGEWIIpByGEEEsYbuvZdBRFGQwGg8HA2hbp\nGQZZ2xyz6OD5O0VbC/jwuY7CSrwOnp4SlOWjBGvbatlsLuUAAEEQbB5DWd4cs4i/cB2INXgdPLSI\nPYfrKKzE/p7D38/K1thcyiEIQiAQCATsjfjRW2Rtc8yig+fv/0ML+PC5jsJKvA6eoiiW4+fvv5it\n4es+hxBCiHcw5SCEEGIJphyEEEIswZSDEEKIJZhyEEIIsQRTDkIIIZbYXJE0QqhZPa4uPFx07qlG\naS+2D5F1i3AbICKEXAeFWgtMOQi1IlfUd6bdXO4r8RBShFAoTFJe+k39x9JOb2DWQezAgTWEWpGj\nRee7OHnbCf78rukqbnO6JO18aQa3UaHWA1MOQq1Fqa78h6fJBPzjRnq5WHa74j5XIaHWBlMOQq1K\n3cYtBFCcxIFaJ0w5CLUWCrFUS+mof+YYlb6yp3MnrkJCrQ3DKUelUkVERMjl8oiICJVKVX+Fqqqq\n6dOnKxSKsLCwP/74g9mtI4RM29NrxX1Nnp76s+1/mb5iqEvflxSh3EaFWg+GU05cXJxUKr13755U\nKo2Li6u/wtKlSzUaTWZmZlhY2DvvvMPs1hFCpvWT9UgMXjNY0fte1eNQmf8sr/H/1xHL1RB7CIpi\nbCSXJEmFQpGUlBQeHp6enj5ixAilUlm76TdFUW3btk1KSgoJCVGpVJcuXRoxYkSdN0lJSQkNDZVI\nJExF9UyVlZVOTk6sbY5ZGo1GIpHwtLM6SZI1NTVs/q2Zxes9h9fBGwwGnU7n4ODA2hZTUlJKS0vH\njh3L2hZbKibPclQqlVqt9vf3B4Bu3bqpVKqysrLaK5SVlRUXFx88eNDFxWXYsGEeHh4Mbh0hhJCN\nY/JW0NLSUgCgvzo5OzsDQElJiVwuN65QUlICAHq9/sGDB+vWrYuJibl58yb9DX3jxo3Lli0DgPHj\nx5MkKRaLGQzMNL1eLxLx9ZZYvV4vFAp5epZDURRJkkIhX0d1+L7n8Dd49vecnJwcqVTK2uZaMCYH\n1kpKStzc3MrKymQymUqlUigUJSUlLi4uxhWePn3atm1bpVKpUCiUSqWrq2t+fn779u0BQKVSKZVK\nep2cnBymQnomkiTffffddevWsXmSzqCtW7eGhIT069eP60CskZaWdvv27bfeeovrQKxRWVm5ePHi\nDRs2cB2IldavXz906NDAwECuA7HG2bNnCwoKYmNj2dyov78/Tz8um8Lk1xyFQiGTybKysvr06ZOV\nlSWTyRQKRe0VXFxcnJycdDodAJAkCQDGA71cLqfPhzp16hQeHs5gVKbp9fopU6aMGzdOJpOxtlEG\n7d69OzAwcNKkSVwHYo2ysrJHjx7xNPji4uI333yTp8EDwIYNG/r27RsZGcl1INZ49OiRTqfj74ff\nmjF5LUcgEEyaNGnTpk3V1dWbN2+ePHkyPeBz5MgRtVoNAEKhcOLEiR9//LFKpVq9evXAgQNrD7sh\nhBBq2ZgcWAMAlUoVExNz8eLF/v3779mzh84oBEHcuXOHLisoKSmJiYlJS0vr27fvV1991bFjRwa3\nbgWKoh48eODn5ycQ8PKu2IKCAmdnZ/rKGe+Ul5drNJp27dpxHYg1DAbDo0ePOnXi602UeXl5crnc\n0dGR60CsUVZWptVq3d3duQ4EWYzhlIMQQgg1hpdf7RFCCPERphyEWp2qqqqbN29yHQVqjTDlINTq\nxMfHz5w5k6eD6unp6QMHDuzateuaNWt4+iu0Zny9FwwhZJ3c3Ny1a9cmJSURBKHVatPS0qRSaUhI\nCC9uKH769GlkZOTy5cs9PDzmzZsXEBAwbtw4roNCFsDyAYRal2nTpgHAnj17njx58tJLL5WUlFRW\nVvbr1+/o0aO2XwO2ZcuWX3/9dd++fQCwcePGPXv2VFVV+fj4JCQkdOvWjevo0LPhwFoDnj59ynUI\nTcLr+G0/eBMXQmw/+MePH+/fv3/gwIEAMGfOnAkTJhQVFeXn57u6us6bN4/r6J7NYDDk5+cbDAYA\nSEtLUygUGzdu9PDwGDVqlF6v5zo6ZAaqtbp161ZkZGRwcPCyZcu0Wq1xuVKplEql8+bNq73Q1qSm\npoaGhvr5+a1evZokydpP2X78CQkJCxYsKC0trf+U7QdPUdTHH38cHh5e52OneBJ8UVHR6dOnXV1d\nlyxZ4uXlpdFo6OW5ubkuLi7cxmaO4uLiTp069e7de9CgQQCQlZVFUZRer3d2ds7NzeU6OvRsrfQs\np6CgYPjw4WFhYe+///7XX3+9cuVK41MKhSI1NVWlUjU4348tKCoqeuWVV+bMmbN+/fqdO3cePHiw\n9rO2H3/Hjh3Xr1/ftWvXbdu20V9X6R5IwIfg6QshCQkJ9IWQX3755erVqxRFQSPB29R5j0aj6du3\n761bty5fvnzq1KnCwkJj/9y8vDy626GNc3V1zcjIeP/991evXu3l5dWhQwcAuHXrllQq5ek9xa0O\n1zmPGytWrJg7dy79OCkpqW/fvnVWIEnywYMHbIdlnoSEBLrciKKorVu39unTx9vbe/jw4dnZ2cZ1\nbDl+g8Hg7++/YsUKLy+voKCg8+fPjx49mj5w02w5+NjY2NjYWIqicnNzu3bt6uLiYm9vP2jQoKKi\nInqF2sFXVlb6+PjEx8dzFW0dK1as6N27d48ePWpqajQazd69eymKunv37qlTpzp16rR161auA7TM\niBEjRo8evW7dOm9v73379nEdDjJLKz3LuX//vvFKaZ8+fYxTYt+/f3/atGkkSRIE4efnx1l8JpWX\nl5eXl9OPf/755/bt23/33XcdOnSIiIigTxoAwJbjFwgECxcuzM/Pv3v3bmRk5KhRoy5dulS7A6wx\n+NTU1LFjx06YMKG6upqzcGsx50JI7U8+Pj7excVl586dWq2Wq5iN6POzL774Yvz48du3b5dIJDEx\nMQaDYcaMGXPmzJk5c+bMmTO5jtEye/bs6dy5c3p6+rZt26ZOncp1OMg8XOc8ttHfRhMTE+VyeVpa\nGkVRxcXFxlHsCRMmzJo1i8v4nqWoqCgnJ0ehUGzfvp2iqL179yqVSoqidDqds7Oz8bu2bTKGV1VV\n5evrW1BQUFNT07NnzzFjxjg4OBw7dqz2yhUVFa6urqtWrUpMTCwtLU1MTLx06VL9KyisoYM3/0LI\n48ePHR0dU1JSlixZ8sUXX7Adbj0xMTExMTEURRUVFfn5+eXn53MdEWqNWlfKMQ50kCT55ZdfJicn\n0wvt7e0NBsNPP/0kk8kKCwvNeStODu7G+LOysh4+fFj7qevXr3t7e+v1euMSjUZz48aNxt6K/fjr\njDJ98sknS5YsSUhIePfddymK+vnnn+uE9Msvv4SFhVEUdffuXXd398DAQGdn54EDB3JyrKwd/P37\n90NCQkQikU6no59NT0/v0aNHnZfY1CH+woULTk5Ojx8/pn/cvHkzXeLFYUiodWpdKaf2WLZxIUmS\nHh4e6enpvXr1+vzzz2uv31hhGFe1SQ3G//nnn8fHx/v4+Ozatav2yo0VVlEcxV8neKVS6e3t3blz\nZ/osrbEg6WvFq1atoiiqrKwsKipq6NChrMVsVCf4Z14IsbVDvFKpTEpKMv5oMBiGDRu2aNEiruKx\nSGOf2/379y9evMhyMKiJWlHKMTHQsWrVKqlU2q1bt9qH8sLCQg8Pjx07diQmJnbp0mX//v21X3Lz\n5s1XX3113rx5LEXfSPzV1dVxcXFRUVEHDhyos7KTk9OFCxcoiqqpqfn5558zMjJqpx+W428w+O++\n+27Pnj0Nrk+f8Xz99ddt2rQZOnSo8eJweXm5SCSqrKxkJ2xaY3uOXq8fMGCAn5/fp59+WvuzLSoq\nsv1D/NOnTzt06LBs2TIOxyrNodPpevXqtWPHjvqJZ9euXYGBgQaDgZPAkHVaUcoxMdCh1+vfe++9\nH3/8sfb6zywMY7mwyqKBmmcWVlH/jL+5x9ksCr72KNYPP/zg7u7u4+NDp88LFy54enqyfJSxOvja\nbPAQf+fOnXbt2v3www9cB/IMCQkJABASEvLrr7/Wear2d0TEC60l5Vgx0LFixYopU6bQj6dOnTp6\n9Oj09PQ333zT39+f/RESi+LPyckRCASbN2+mKGrs2LGLFi3S6/VKpXL8+PHG36i25q7ltfTDrz8E\nt3LlSm9vb3q28tpnDyxoYvC12eAhPjMz0/aP2mvXrg0PD1+yZIm9vX10dHSdq5iIX1pLyrFioMOm\nCsMsjd+iO8xNHCUZYVHwJsY/i4uL2U/2TAVP48Uh3tYsXbqUvmaTnZ09ceJEBweHZcuWVVRUcB0X\nskZrSTn1mRjoMGYUcwrDuPLM+M0srOKkltdE8DZV6NUgXgffApw9ezYwMLBnz554FYePWm/KoRoZ\n6GhslKmxwjATmvtk6Jnxm3OHOVdHyQaDt7VCr8bwIngTm+ZjrVdVVZXxsU6nu379OofBIKu16pRD\nNTTQ0eAoU2OFYVTjTSrZKUQ2J/7GCqsoro+S9YO3/UIvIxsP3kShF8XDWq/CwsIOHTo0Vk+PeKS1\np5w6rBhlSkxMBAA3N7etW7fS/97GHGMjhdQm3L1713aOkg2ywUIv83EbvIlCL4pvtV5vvfUWfcsw\n4rsWnnIsvYnMilEm000qm1JIbcXASNPLeZk6SjI4qsNJoRdTtx9yWKXG00Kv+p+8RqMZOHAgnuK0\nDC055Vh6E5nVo0zbtm2bPXt2RUXF0qVLHRwcXF1dG0wzpjvQmB88U/E3VqjW9KMk46M6LBd6MXv7\nIVdVanws9DK956AWoCWnHMrCm8gsGouvXRpgTpNKymQHGkuDb3r8pofgmn6U5PuoTsu7/ZAvhV6m\n9xzEd8Lly5dz1sW6+SUnJ5eXlw8fPnzOnDm3bt0KCwuTy+X0U0KhsM7KEomkc+fOxh8Jghg1atTc\nuXPLyspeeOEFgiCMT2k0muDgYJIk+/fvDwAikUij0aSkpNy+fdvV1XX37t2DBg3q06ePk5OT8SW5\nublTp07ds2ePj4+PVqtNSUlRKpXt27ev/bbmB9/E+AFgzpw5/v7+8+fPDwoKeuONN6ZMmeLs7Gx8\n1s3Nrf77W8TS4G2NRXuOLauurhaJRADQsWPHmTNnDh482ManYjO95yDe4zrnNa+mjy00OMpUf0jq\nmU0qzelAw3jwjcXPQqGa+cHbZi1v0+O3hUJkfhV60f8LfBwPROZr4SmnDuvGFuqMMjU2JGWiSaWl\nHWgYDL5+/BQX5byNBc+XWl4r4reF4HlU6NXY/XB8GQ9EZmpdKYdi4iYyS6vaLJ3ay4TmuwOOhXLe\nxoLny1UfK+JnOfg6aY9fhV4mui7hjZ8tSatIObXvW67PooERS4ekrJjai5HgG4vfhOYo5zUdPM2W\na3l5FD+va71sbQZV1HxafsoxPZxt6cCIpUNSlk7txVTwjcVvGrPlvGZeSLDZsXvexc/fWi/sTdd6\ntPyU88zh7CYOjJgYkrJ0ai9mgzcz/uZj3YUE2xm75138NnK+ZSlb602HmlWLSjm3bt165513/v3v\nf+/fv5/uOmPOcHbT/1EbG5Ky9LtbnfjZCZ5iqLbKiuBr47Zpo3V7Tm0cxm8serSd8y2L2FRvOtTc\nWk7KKSkp8fDwiIuLW7RoUceOHX18fL799ltzLoYz8o9af0jK0u9u1sXPSC1y02urrP7wadzW8jYx\neIrT+E1Mr2c754uW4nVjPWRay0k5x48fHzx4MP1Yp9Pt2LHDxcVl6tSpljZyZuof1dLvbozEb10t\nMtXkIbgmBs9tLW/TP3kO4zc9vR5/a71scAZVxIiWk3IePnzo6OiYnp5uXFJQUBAWFmZFI+dm+kc1\n/d2Nqfitq0VuIouCv3XrVmRkZHBw8LJly6wYgmNcU4KnOK1FbtmFXjiDaovUQlIOPZz95ZdfSqXS\nr776ynhMz8/Pl8vl5nxdZbCQ2oQGyvmRLgAAB3lJREFUv7sZx+Ktjp/DWl5Lg8/Pz/f09Fy5ciXd\n++ejjz5iJAyrWbTn2FrwWOiFeKclpJzaw9nHjh1zd3cPCwtLTEzUaDTXrl3z8fF55ogws4XUptX5\n7lZnLN6K+Dms5bUi+BUrVsydO5d+nJSU1Ldv36YE0ESW7jk2FTwWeiE+agkpp85wdmlp6erVq/38\n/AiCcHV1rX1BpTHNXUhtfvCU5fFzWMtrRfCvvfbaxx9/TD8uLi6WyWT04+zs7NjYWJYvdFu659hU\n8Lwu9LLNxnqIBbxPOSaGs1UqVf2jgHUTQJk/KmXREJzpsfgG42/ihRAGa3ktDZ4ewkpMTJTL5Wlp\naRRFFRcXG/v9TJgwYdasWVYHYwVL95yioiLbCb5BfCn04ktjPdQceJ9yLBrOtropiJmjUpYOwVk6\nFt/EawnM1vJaNwMpSZJffvllcnIyvdDe3t5gMPz0008ymaywsJCRwMxkRfzr1q2zkeAbw5dCL1u+\nhRk1K36nHCuGsxkp3DIxKmX+EJwVwTfxWgKDtbyMzEBKkqSHh0d6enqvXr0+//xzRgIzU9Pj5zB4\n03hR6MXTRgmo6fidcqwYzm7Kvm7OqJT5729F8BZdS0hISFiwYEFpaSn9I7O1vEzNQLpq1SqpVNqt\nWzeWj5KMxM9V8C0ATxsloKbjd8qpz8RwdhMngGKhMOyZwVt0LSExMREA3Nzctm7dSn9/t/SuWIuY\nCN7EEJZer3/vvfd+/PHH5gvMTFbEbzvB8x1/GyUgS7W0lEM1Mpzd9Amg2CkMMx28RRdCDAaDv7//\nihUrvLy8goKCzp8/P3r06KtXr1r6K5iPqxlImcKL+G15EtKm4G+jBGSRFphyqIaGs5s4AVRTRqUs\n/V8yM3hzriVs27Zt9uzZFRUVS5cudXBwcHV1ffDggRW/QlOC51ctr43Hb+OTkCL0TC0z5dRhaV+Q\n+k1NLGVOOwAzWXohxNgOgA7D19e3oKCgpqamZ8+eY8aMcXBwOHbsGFOxWYcvtbyN4TZ+25mEFCEr\n8C/lWHETmUXlsE1vamL6qo+lAyMWXQipP374ySefLFmyJCEhgR4V/Pnnn2vnJEs1a+Of5mbjM6ia\nCWu9EK/xLOVYcROZFeWwTWxqYuKqj6UDI02v5VUqld7e3p07d256rVqzNv5pbjY+g6r5sNYL8RrP\nUg5l+U1klo7FW1SInJqaGhoa6ufnt3r1anqY5ZlXfSwaGGGklve7777bs2dPY/FYhNejOi3y9kOs\n9UL8wr+U0/SBBdO1yOYXIhcWFnp4eOzYsSMxMbFLly779+9nIX7rapGZwutRHV4HbwLWeiEe4V/K\nYWRgwUQtsvlNTRISEmbOnEk/3rp1a58+fby9vYcPH56dnd2s8XNYy9v0SUg5rOVlZAZVvtciI8Qt\n/qWcOqweWDCnFtl0IfKKFSumTJlCP546dero0aPT09PffPNNf39/84/11sVvI7W8VkxCaju1vNbN\noGo78SPER7xPORRDAwuWNjUpKirKyclRKBTbt2+nKGrv3r309RudTufs7GxRVRgnk5AyxYpJSG3n\nqol1M6jaTvwI8U5LSDn1MVhI3WBTE2MtclZWVp1LAtevX/f29m7iiBava5FpvL5wwuvgEbJlLTDl\nsFNIXb8dwOeffx4fH+/j47Nr165mih9rednB6+ARsmUtMOVQzVxI3eAQXHV1dVxcXFRU1IEDB5o1\nfp6O6vC6lpfXwSNkU1pmymnWQmqsRbYOr2t5eR08QrZDAC1ReXn5//73v08++eT333/X6XT+/v7L\nly+vrKw0/x3c3NySkpK2bNly4sSJ2stTU1OPHTu2du1aAHB3d//ggw9mzJhhMBi4ir+xTT948ODS\npUvMRtVEIpEoKCiI6yisxOvgEbIdBEVRXMfQ7M6dOzd//nySJG/cuCEQWJBl796927FjRzs7O+OS\n0tLSK1euDBs2jP6RJMmRI0eGhITQSaiZNBa/Xq8PCQmZP3/+66+/LhQKa7/km2+++fzzz69du2bR\n74sQQs2qVaQcANDr9b///ntzfFEtLi7u06fPjBkzli1bRhAE4+9Payz+9evXL1y4MCQk5H//+9+g\nQYNqP6XVamsnS4QQ4lxr+Qpcf2DExGiYRaNSjQ3BMauxgR2dThceHj5q1Khhw4ZNmTLl0aNHxqcw\n3yCEbE1rSTl16PX63r17f/XVVw0mnl9//XXmzJkkSZr5bv7+/r/88svw4cMZjdEsTb9qhRBCrGkt\nA2v1mRiSAt6OSll91QohhFjQeo9KJoakwOxRKVsrDHv55ZevXr26b98+zDcIIRvUeg9MTS9EBsuH\n4FiA5bwIIZvVegfW6rCiEJnG0yE4hBBiX+s9y6mjsSEpkUg0Y8aMt956KywsLDk5uf4LMd8ghJCZ\nMOX8zYpCZIQQQubDlPNsWIiMEEKMwGs5FsNCZIQQsg6mHGs0X/schBBqwTDlIIQQYgmOCyGEEGIJ\nphyEEEIswZSDEEKIJZhyEEIIsQRTDkIIIZZgykEIIcQSTDkIIYRYgikHIYQQSzDlIIQQYgmmHIQQ\nQiz5/+U4T3ZBBi8fAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -h 250 -w 550\n",
    "df.f = df %>%\n",
    "    filter(variables %in% c('Sensitivity', 'Specificity', 'Balanced Accuracy')) %>%\n",
    "    mutate(variables = gsub(' ', '\\n', variables))\n",
    "\n",
    "ggplot(df.f, aes(BD_window, values, color=variables)) +\n",
    "    geom_point(size=3, alpha=0.7) +\n",
    "    theme_bw() +\n",
    "    theme(\n",
    "        text = element_text(size=16),\n",
    "        axis.title.x = element_blank(),\n",
    "        axis.text.x = element_text(angle=60, hjust=1)\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%pushnote validation MW-HR-SIP complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HR-SIP with multiple windows (post-occur)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def HRSIP_multi_window_post(physeq, BD_shift, BDs, outname, padj=0.1, log2=0.25):\n",
    "    # HR-SIP for each window\n",
    "    occurs = ','.join([str(x/100.0) for x in range(0,55,5)])\n",
    "    outname2 = outname + '_DS2.txt'\n",
    "    !SIPSimR phyloseq_DESeq2 --occur_heavy $occurs -w $BDs $physeq > $outname2\n",
    "    # making confusion matrix\n",
    "    !SIPSimR DESeq2_confuseMtx --padj $padj --log2 $log2 $BD_shift -o $outname $outname2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.70-1.74,1.72-1.76"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1070\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 75 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.76\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 19\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 973\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 4 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 100\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1070\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 75 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.76\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 19\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 973\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 4 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 100\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1023\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 74 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.76\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 19\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 870\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 5 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 101\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 972\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 70 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.76\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 19\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 803\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 5 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 103\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 917\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 66 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.76\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 19\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 737\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 5 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 100\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 863\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 57 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.76\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 19\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 677\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 5 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 94\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 18\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 809\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 44 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.76\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 19\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1090\n",
      "**OUTPUT MUTED**"
     ]
    }
   ],
   "source": [
    "BDs = '1.70-1.74,1.72-1.76'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window_post(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW1p')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.70-1.75,1.72-1.77"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1076\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 82 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 21\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 25\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1000\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 21 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 103\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1035\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 82 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 20\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 25\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 923\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 22 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 103\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 997\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 83 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 23\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 25\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 850\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 22 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 103\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 947\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 81 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 20\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 25\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 791\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 22 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 103\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 906\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 79 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 28\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 25\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 709\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 22 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 98\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 815\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 72 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 33\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 25\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 665\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 20 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 94\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 24\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 765\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 68 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 35\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 25\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 611\n",
      "# DESeq2 run\n",
      "**OUTPUT MUTED**"
     ]
    }
   ],
   "source": [
    "BDs = '1.70-1.75,1.72-1.77'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window_post(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW2p')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.70-1.73,1.72-1.75,1.74-1.77"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1064\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 13 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 15\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 944\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 2 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 79\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.74-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 16\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 945\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 39 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 92\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1064\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 13 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 15\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 944\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 2 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 79\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.74-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 16\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 945\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 39 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 92\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1008\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 13 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 15\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 835\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 2 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 80\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.74-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 16\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 832\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 40 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 93\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 948\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 12 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 15\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 747\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 2 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 83\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.74-1.77\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 16\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 722\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 38 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 93\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 14\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 948\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 12 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 15\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 674\n",
      "# DESeq2 run\n",
      "**OUTPUT MUTED**"
     ]
    }
   ],
   "source": [
    "BDs = '1.70-1.73,1.72-1.75,1.74-1.77'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window_post(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW3p')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.70-1.72,1.71-1.73,1.72-1.74,1.73-1.75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1040\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 979\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 816\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 29\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.73-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 896\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 92\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1040\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 979\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 816\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 29\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.73-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 896\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 92\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1040\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 880\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 816\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 29\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.73-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 759\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 92\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 971\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 880\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.72-1.74\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 659\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 33\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.73-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 759\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 92\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.7-1.72\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 9\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 971\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 0\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.73\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 10\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 771\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "**OUTPUT MUTED**"
     ]
    }
   ],
   "source": [
    "BDs = '1.70-1.72,1.71-1.73,1.72-1.74,1.73-1.75'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window_post(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW4p')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.71-1.75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1025\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 17 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 59\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 963\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 17 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 59\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 908\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 18 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 63\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 851\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 17 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 63\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 794\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 17 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 62\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 728\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 11 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 63\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 670\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 11 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 56\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0.35\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 611\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 9 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 49\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0.4\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 550\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 9 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 41\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0.45\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 499\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 7 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 32\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.75\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 20\n",
      "# occurance cutoff: 0.5\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 444\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 6 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 25\n",
      "\n",
      "# A file of all DESeq2 run results was written to: DESeq2_all_runs.txt\n",
      "\n",
      "#- Greatest number of rejected hypotheses for each BD window -#\n",
      "heavy_BD_min\theavy_BD_max\tmax_n_rej_hypo\n",
      "1.71\t1.75\t63\n",
      "\n",
      "# Global adjustment of p-values for all BW windows with \"best\" occurance cutoff\n",
      "\n",
      "#- Number of rejected hypotheses for each BD window (post global p-value adjustment-#\n",
      "heavy_BD_min\theavy_BD_max\tn_rej_hypo\n",
      "1.71\t1.75\t63\n",
      "\n",
      "Log2Fold cutoff: -Inf\n",
      "padj cutoff: 0.1\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW5p\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW5p_data.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW5p_table.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW5p_overall.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW5p_byClass.txt\n"
     ]
    }
   ],
   "source": [
    "BDs = '1.71-1.75'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window_post(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW5p')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.71-1.8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "replacing previous import ‘S4Vectors::Position’ by ‘ggplot2::Position’ when loading ‘DESeq2’ \n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 1053\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 69 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 88\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0.05\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 992\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 68 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 89\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0.1\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 915\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 67 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 92\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0.15\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 874\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 66 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 91\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0.2\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 790\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 64 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 90\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0.25\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 722\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 60 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 85\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0.3\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 688\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 57 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 81\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0.35\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 624\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 59 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 76\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0.4\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 580\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 57 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 67\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0.45\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 509\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 57 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 62\n",
      "\n",
      "# occurance cutoff: 0\n",
      " pre-filter: number of taxa: 1147\n",
      " post-filter: number of taxa: 1090\n",
      "# \"heavy\" BD window: 1.71-1.8\n",
      " pre-filter: number of samples: 56\n",
      " post-filter: number of samples: 32\n",
      "# occurance cutoff: 0.5\n",
      " pre-filter: number of taxa: 1090\n",
      " post-filter: number of taxa: 435\n",
      "# DESeq2 run\n",
      "converting counts to integer mode\n",
      "using pre-existing size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "-- replacing outliers and refitting for 55 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "estimating dispersions\n",
      "fitting model and testing\n",
      "# Number of rejected hypotheses: 54\n",
      "\n",
      "# A file of all DESeq2 run results was written to: DESeq2_all_runs.txt\n",
      "\n",
      "#- Greatest number of rejected hypotheses for each BD window -#\n",
      "heavy_BD_min\theavy_BD_max\tmax_n_rej_hypo\n",
      "1.71\t1.8\t92\n",
      "\n",
      "# Global adjustment of p-values for all BW windows with \"best\" occurance cutoff\n",
      "\n",
      "#- Number of rejected hypotheses for each BD window (post global p-value adjustment-#\n",
      "heavy_BD_min\theavy_BD_max\tn_rej_hypo\n",
      "1.71\t1.8\t92\n",
      "\n",
      "Log2Fold cutoff: -Inf\n",
      "padj cutoff: 0.1\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW6p\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW6p_data.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW6p_table.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW6p_overall.txt\n",
      "File written: OTU_n2_abs1e9_PCR_subNorm_MW6p_byClass.txt\n"
     ]
    }
   ],
   "source": [
    "BDs = '1.71-1.8'\n",
    "physeq = '../OTU_n2_abs1e9_PCR_subNorm.physeq'\n",
    "BD_shift = '../ampFrags_BD-shift.txt'\n",
    "HRSIP_multi_window_post(physeq, BD_shift, BDs, 'OTU_n2_abs1e9_PCR_subNorm_MW6p')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Plotting results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['OTU_n2_abs1e9_PCR_subNorm_MW1p_byClass.txt',\n",
       " 'OTU_n2_abs1e9_PCR_subNorm_MW2p_byClass.txt',\n",
       " 'OTU_n2_abs1e9_PCR_subNorm_MW3p_byClass.txt',\n",
       " 'OTU_n2_abs1e9_PCR_subNorm_MW4p_byClass.txt',\n",
       " 'OTU_n2_abs1e9_PCR_subNorm_MW5p_byClass.txt',\n",
       " 'OTU_n2_abs1e9_PCR_subNorm_MW6p_byClass.txt']"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files = ['OTU_n2_abs1e9_PCR_subNorm_MW{}p_byClass.txt'.format(i) for i in xrange(1,7)]\n",
    "files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "  library      variables    values file             BD_window\n",
       "1       2    Sensitivity 0.9082569  MW1 1.70-1.74,\\n1.72-1.76\n",
       "2       2    Specificity 1.0000000  MW1 1.70-1.74,\\n1.72-1.76\n",
       "3       2 Pos Pred Value 1.0000000  MW1 1.70-1.74,\\n1.72-1.76\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -i files\n",
    "\n",
    "renames = data.frame(file = c('MW1', 'MW2', 'MW3', 'MW4', 'MW5', 'MW6'),\n",
    "                     BD_window = c('1.70-1.74,\\n1.72-1.76', \n",
    "                                   '1.70-1.75,\\n1.72-1.77',\n",
    "                                   '1.70-1.73,\\n1.72-1.75,\\n1.74-1.77',\n",
    "                                   '1.70-1.72,\\n1.71-1.73,\\n1.72-1.74,\\n1.73-1.75',\n",
    "                                   '1.71-1.75', '1.71-1.8'))\n",
    "\n",
    "df = list()\n",
    "for(F in files){\n",
    "    tmp = read.delim(F, sep='\\t')\n",
    "    df[[F]] = tmp\n",
    "}\n",
    "df = do.call(rbind, df)\n",
    "df$file = gsub('.+(MW[0-9]+).+', '\\\\1', rownames(df))\n",
    "\n",
    "df = inner_join(df, renames, c('file' = 'file')) \n",
    "\n",
    "df %>% head(n=3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAImCAMAAAC8U6mTAAADAFBMVEUAAAABAQECAgIDAwMEBAQF\nBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcY\nGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKior\nKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+\nPj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBR\nUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2Nk\nZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3\nd3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmK\nioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJyd\nnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+w\nsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLD\nw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW\n1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp\n6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8\n/Pz9/f3+/v7////isF19AAAgAElEQVR4nO19C3wV1bX36eOqKKIoUj3oVx4RxU8QxNrafq3WT/G2\nCn70a28MXMFYIdr6ailSq/Qh0tLC1VZrc7UVfLQ8giJY0EYSzENAyChFrYiKWoKBKiSQBBITcubO\nzJk5M3PO3nvWmj37PML6/37mZGYWy3/mf2Zmz1p7rR3TCb0asVwTIKgFCdzLQQL3cpDAvRwigRNj\nt9m/tXzzhG+2pD5MdB/i4SD3iBQQbjEM8tntrnd3yOCTAIETS4tjjsDfLf6o+LupDxP1L9ZzUM07\nkImaGrgtwu26OiVuEaZ166Jw+4s6Gaz+Z4DAh8vKHIF7+r2sbzoxYX8kBT7E+3ftgntCGro+gdsi\n3B7qUeIWYdrDPTsYt/dqMqgOEtg8aAu8L7Zfb4m12B+6vmnevAfaDnPQyjuQic4OuC3CbXuXErcI\n0672KNxmT+B3Y916d+xd+0PXK6dPv7elk4P9vAOZaG+H2yLcth7iHCiBQZ5Bx4EoGGRP4L2xA8al\nu8/+sHblzS0aeMrQtlJ/GPcWjWKQPYF7+r2iN/RL2B/WLhJYBNUCb24I3gILvPyAMXz+bsf1N6Q+\nTJDAIqgWuPT+4C2wwOaPlm/0v7Il9WGCBBZBgcC+y7S0OHgLIrAAtsByp4wEFjLwCWxfmHEbFwdu\nkcBYtjkWOHlhrnSgBW1JClyXfA8GEua86EXxHoxhIMc2ivdgFANbYP+FCYaswO09JoCEe9jo7OAc\nYKCNsx/DQI4tjwED3QcjYNtjC+y/MLMlMN2iRSiU92ABSGAR1AkcfwC6FSywmxwsj5ko3mv+vDp5\nkAQORQHFgCXwyjroVrDAbnKwrbGxceeY6o3DjM+PkwdJ4FAUUAycQZbvwgQjUGBvctDAoln6k1e5\nR0ngUBRQDJxBVl2mevX3Bm4FCpxKDia3RrXrP/3fg/td9X5ymwQORQHFgHWLXpv8qIwHbgUKnEoO\nWpherut3TWrcc80XjI0H+vcv/bjNBJBwGxv793MOMLCPsx/DQI4tjwHGLYqBX+DkhRn/yqy1roii\nrUCBvclBvekk+/vbFPvIGH7t2LGGruAwFFAM7mVctGt/N2X49TWpa1awBXgGu8lBfY6ZRCrfoesf\nxVqtHXSLDkUBxcAZZPkuTE17ac6X5//N2eBvAUbRyeSgmS7Uz3va+HHD19766NorkwdJ4FAUUAyc\nK9h3YVoy3vxVd4O3BXgPTiYHzXRhU2yPsaN10gknT9mbPEgCh6KAYuA+g/2XqYFHi7WgLYpkYdnm\ndBTtXJijvr3cuKZL5m/2HHNTxQ01q2ucDRIYyzbHr0nJCzM+7ZzlWv2PRv3Ac8SZw1FbWhSPx4uu\nq41E4IMJE0DCCTY+6eQcYKCNsx/DQI4tjwHGLYqBLXDDg3ONn79ebW3EK+d9ZZOmrRjpFdiewzF+\nypLaTbWLiydEIXDdgS4TQMJdbBw6yDnAwAHOfgwDObY8Bhi3KAa2wDNHmgKPG3J/UuCGccb++mFJ\nuX2p4uEbk8oW0S06FNtc3aJHGk/dJxq0RRckBdaeOuvB2rIrrUP+VPH4qcvqNtUtKR5PAodimyuB\nB6/X6uJ12sYhtsDaQ0XxC5/VMlEzdZj5DJ5SQwKHYpsrgS+986XZg+auv8u6D680HsDahsoGhr4G\nNq9bXeOMrzH5YCcR7CkfJYFDUUAxsAWuOC8+4vHz4+f82VbOP7jmbmHywU4i2Fs+SgKHoYBi4Ki1\nuXKz1lCZevON+yTlbmHywXYimFE+KnfKSGAhA/dyvJcjopTAnnywnQhO7XmrouJxEjgMBRSDe9kq\nTtJAW5h8sJ0ITu2puOyymfsOmgASPshG6wHOAQaaOfsxDOTY8hhg3KIYcATWIio+8+WDrUQwo3wU\nSJjzv6ArWMjAFdh/mUZUfObJB9uJYEb5qNwpI4GFDFQXn7n5YCcRnFk+KnfKSGAhA9XFZ24+2EkE\nZ5aPyp0yEljIIKfFZyRwKAooBjktPiOBQ1FAMchp8RmVj4aigGKQ0+IzKh8NRQHFQHXxmRB0iw5F\nAcVAdfGZECRwKAooBoqLzzzJwZ67B/W9YrtO5aPSFFAMFBefeZKDi854s/WWEQkqH5WmgGKguPjM\nkxwsuUfXm2O7qHxUmgKKgeLiM0+6cHerrq/o10Hlo9IUUAwUF5/5yke7ywescstHnxg79jZKF4ah\ngGKguPjMmxzcMvqSrcm9VvnoB2vXLqErOAwFFAPFxWee5OCWgQvNTyoflaaAYqC6+MxNF0682Rg+\nN3ZR+ag0BRQDVvGZpn2ztD6111fVklaYhkkXnm61UdpG5aPSFFAMWMVnmjb69guXOPt8VS1phWkU\nycKyzZXA9aVztdtnLFhcYwusPXnBHfYcWl9VS1phGgmMZZsrgSeNW62VfGnC+XFHYK3u2sufs373\nVbWkFaaRwFi2uRK4aJXxjjTGHkU72f9kBaGvqiWtMI3KR7Fsc1U+ela1plUV2QKvXHmuJ/vvr2rx\nF6bRFYxlm6srePLk6rXXXJ16D57tHXn5qlr8hWkksNA2D9w6Aq8bH49fvEZbN1FDIesCA02x5wFh\nWmBu3dekuir7lzXrvblD0RYuH8xYfZQEVu6WkS68dbg3+y/awuWDWauPksCq3aru+O7JB/NXH43+\nbyOBS7IksCcfnL766B5NW0ECq3arWmBPPjh99dFHhg698WNs318GomgnzEBzqxK3CNPW5ijcKhbY\nkw/O59VHGaAFokECe/LBrNVHm5o52MU7kImP/gW3Rbht2qvELcJ0L/fsYNzOUSuwt51w5uqjG+cR\nVOOX1fUSWLc3SGBPO+HM1UcJBYKwkSxCgYAE7uUggXs5RAInxm6zf8uMRRMKBHyBE0uLY47AmbHo\nd15Yy8HfeAcy8QLXiZRbhKkqt5VRuF34fKUEnv1XgMCHy8ocgfmxaAYo0FEggQ49+YZkIj0WrZPA\nYhSawOmx6N/EYpMoFi1AQcSiPQKnx6I7mpurevmUHQSDQs0m6a7ArFg0CRyGAopBlgRmx6JJ4FAU\nUAwCBRY1nYULzI5Fk8ChKKAYBAosajpL02bRbPNBYP9VKmo6SwKj2eaDwM5VKmozG43AdbS0XRgK\nKAYsge2rVNRmNiKBW7tNAAl3s9FxiHOAgVbOfgwDObY8Bhi3KAb+pTiwPWfpFo1lm9tbNLrnLAmM\nZZsPz2AESGAs23wSWNRmFiywm/0tt3p0FFOvSmkKKAYCgUVtZsECu9nftsbGxp1jqqlXpTQFFAP/\nIAvbcxZTm2Ri0SydelVKU0Ax8A+yWFepqOkspjbJ3BrVrqd6VW6aN+8BaukfhgKKgeAWLWozCxXY\n16tSn16uu70qK6dPv7el0wSQcCcb7e2cAwzs5+zHMJBjy2OAcYtiwBLYvkpFbWahAvuWtms6yb5B\nWb0qdbpFh6SAYuATOO2aFbSZhQrszf7qc8wsIfWqlKaAYuAfZKVfpaKmsyCBvbVJ+nlPGz+oV6U0\nBRQD/xWceZWKms7ielUaN+Y9xg7qVSlNAcUg/RmceZWKms5SJAvLNucCe65Sf2NZTWuoWV3jn9BB\nAqPZ5oHAqavU31i2trTI7G54XW2EAtPSdqEooBikCextFO1vLDt+ypLaTbWLiydEKTAtbReGAopB\nmsDeRtH+xrLDNyYlLaJbtAzbXN+ivY2i/Y1lx09dVrepbknxeBJYhm3uBXYbRfsby9ZMHWY+g6fU\n4AR204VOnlCqlSEDJLCQAXPKTvI27G8sq2mb162u2axpOIHddKGTJ5RqZcgACSxkkDZl51z/lB3/\n9Z054sakC+08oVwrQwZIYCGDNNFm+zfjgi2QwJ50oZ0nZJSPyp0yEljIIP2q9G9LC+xJF9p5wtSe\nB/r3L02WjwIJc+ojoygfxTCQY8ut80S4RTFIF9gv4iTBFkhgX7rQyhOm9oRqRsoAXcFCBmKBNdni\nM0+60M4TUvmoNAUUg3SBM65SyeIzN13o5AmpfFSWAoqB6uIzN13o5AmpfFSWAopBTovPSOBQFFAM\nclp8RgKHooBikNviMyofDUMBxSCnxWe0tF0oCigGVHyGZSDHNh+ewQ4iKT4TggQORQHFQHXxmRAk\ncCgKKAaKi8882d+euwf1vWK7TuWj0hRQDBQXn3myv4vOeLP1lhEJKh+VpoBioLj4zJP9LblH15tj\nu6h8VJoCioHi4jNP9nd3q66v6NeRKh99q6LicRI4DAUUA8XFZ77y0e7yAavc8tGKyy6bue+gCSDh\ng2y0HuAcYKCZsx/DQI4tjwHGLYqB4uIzbz54y+hLtib3UvmoFAUUA8XFZ57s75aBC81PKh+VpoBi\noLr4zM0HT7zZGD43dlH5qDQFFANB8Vl96Vzt9hkLFidnQjc8ONf4+evVPltMPvh0q43SNioflaaA\nYiAoPps0brVW8qUJ5yflnjnSFHjcEN+UDopkYdnmhcD2a1LRKmO8NcZ5Io9crmlPNGiLLiCBZdjm\n02vSWdWaVlXkbA1er9XF67SNQyIUmMpHQ1FAMRC8Jk2eXL32mqudrUvvfGn2oLnr74q0XzQJHIYC\nioHgNWnd+Hj84jXauonWVsV58RGPnx8/5885vUUDTbF3MoRpgbkVvybVVXmObK7crDVU+qvPSOB8\ndyt6TQIAky5krD5KAit3y39NWrPemzz0byEE9hSLZq4+SgIrd8tPF9463Jv+92/BBfakC/mrj0b/\nt5HAJYECg4BJF/JXH43+byOBS7IksCddmL766BNjx96GLh8FmmKrLBGmBeZW9eqjnnRh+uqjH6xd\nu4TWDxagINYP9qQL+auPoghngAQWulU98d3TbZax+uiSCg64BzKxdCncFuF2yTI1buGmyyJx+4tn\nZbAM0202c/XR3VLfLgIEm9sPyaAnSGBCbwAJ3MtBAvdyiAROjN1m/5YZiyYUCPgCJ5YWxxyBM2PR\nHc087OUeycC+fXBbjFu4KcYtwhRDge9222tvyOBQgMCHy8ocgVmx6I28sd96+DBx08twW4TbjQ3B\nNiHcIkwbuGcH4/aeDTKoBOSDHYH5sWgGKNBRKIEOV+D0WLS78hlmgbBMRLHyGQOth5S4RZh2HIjC\nbfYETo9Fm2sXcjtWcA9kIooeHQy0dXIOAEP98gw626JgkD2B8zkWDTxlaFupP4x7i0YxyJLAnFg0\nCSxAdgXezBxUggVmx6Jp4rsI2RU4rQspQmABSGARVAss6klKAodlm0cCs3uSksBybPOi+Mx/zYr6\nG8qWrrR2mwAS7maj4xDnAAOtnP0YBnJseQwwblEMWALDe5LKCkzNSMNQQDFgCQzvSUq3aCzbPLpF\nQ0ACY9nmocCi/oaY2qRyq4VDMbUylKaAYhAsMLMoCSywm/1ta2xs3DmmmloZSlNAMWAOssA9STG1\nSSYWzdKplaE0BRQD5iAruCcpVGBv9tfYGtWup1oZ6iRwSAooBsG3aGYXUqjAvlaG+vRy3W1l+JtY\nbBK1MgxDAcVAIDCzJylSYN/Sdk0n2d9fq5VhR3NzFV3BYSigGLAEFvUkRQrszf7qc8wsIbUylKaA\nYsAcZIl7kmIE9tYm6ec9bfygVobSFFAMmFdwUE9ShMCe2qSm2B5jB7UylKaAYsB5Bgf1JAULLAQJ\nHIoCigF3kOW5ZhtqVtewZwmTwFi2+SNw6pqtLS2Kx+NF19UqEJiySWEooBiwBB717eXGg7hkfrLp\n2fgpS2o31S4unhC9wJQPDkUBxYA5ip52znKt/kejfmBtDU9WUFQX0S06ArZ5cYuOV877yiZNWzEy\neQVPXVa3qW5J8XgSOAK2eSJwwzhjf/0wa6tm6jDzGTylhgSOgG2eCKw9ddaDtWVX2tub162u2cyw\nw+WDnUSwVK9KBkhgIQOOwNpDRfELn3V2cIfamHywkwiW6lXJAAksZMBMFxoPYG1Dpfvqy+0+i8kH\n24lguV6VDJDAQgaci9O/O7zAnnywnQhO7TGrC6njexgKKAYcgf2STgotsCcfbCeCffXBzVYvJiBh\nTientlZ416cWzn4MAzm2PAYYtygGIIG10MVnvnywlQj27qFbdCgKKAYcgTOu2ZDFZ558sJ0IZtQH\ny50yEljIQHXxmZsPdhLBmfXBcqeMBBYyUF185uaDnUQw1QfLUkAxyGnxGQkcigKKQU6Lz0jgUBRQ\nDHJafEYrn4WigGKQ0+KzuvYeE0DCPWx0dnAOMNDG2Y9hIMeWxwDjFsVAdfGZEHSLDkUBxUB18ZkQ\nJHAoCigGiovPPMnBnrsH9b1iu07lo9IUUAwUF595koOLzniz9ZYRCSoflaaAYqC4+MyTHCy5R9eb\nY7uofFSaAoqB4uIzT7pwd6uur+jXQeWj0hRQDBQXn/nKR7vLB6xyy0cfGTr0RiofDUMBxUBx8Zk3\nObhl9CVbk3ut8tE9mraCruAwFFAMoi8+Syy46JTddy1JbniSg1sGLjQ/qXxUmgKKQfTFZ/818K+x\n3atOeDS55aYLJ95sDJ8bu6h8VJoCigGk+OybpfUcI5bAQx7QY7v1X56d3HLThadbbZS2UfmoNAUU\ng+DiM00bffuFS+ACH/OCKfCaYwHnmwQORQHFgC3w5hdX17kzOEZrT15wB3PmO0vgkb82Bb7zfMD5\nJoFDUUAxYAn84mSzWGXE9zakBNbqrr38OaDA5X3/O1Zxx2f/DDjfVD4aigKKAUvgq26oqrnp5mXf\nnmxtOel/aHVhz6/6xmJnLAToq9cdiGAdkyhWXcEwkGPLXXUF4RbFgCXwsFrjkh2uVZ1pba1ceS5y\nyk7PB80QeekWHZICigFL4HE3vVhz+8W1NzhTdmYzbEQCg4EXGGiKPQ8I0wJzyxR4zaXx+BefeuQb\nq7jCigTeZoMEzge3nFF0zVpnDL1mPTbhH7OR3PIUi2YuL0sCK3cbmC68dXiYKTttz/2f95K/eYpF\nM5eXJYGVu1XW8X3JpdaHJx/MWl6WBFbtVpnA9X2tD08+OH152bcqKh4ngVW7VSCwNcJquOxca8OT\nD05fXrbisstm7sMmbhmIIh/MwP52JW4Rpu0tUbiNXuDkEGtIrbXhyQenLy/r3qIZoAWiC2SBaE8+\nmLW87HsfcvAu70Am/vkB3Bbh9r1GJW4Rpo3cs4NxO0etwN52wpnLyzb8lof7uUekkHu3WWewYEOw\njHy8vN8v8DYPkns87YQzl5clFAhSAsc8yCUhQrQgMXs5GAJ3fJB9GgRV8Avcaj6BHzrB3kqMddIO\nmbFoQoHAJ/Cyz5hP4E/faW0klhbHHIEzY9HbXljLwd94BzJRWQm3RbhFmCpy+0Ikbv/wzCoJLGvK\nFHjEtNYvvPbB6A3WxuGyMkdgfiyaAQp05G+g46iV+l2P6UsvcbYdgdNj0S07dqwhgQXIV4H7P6JX\nfFffcLyz7QicHot+oH//0kKMRQND/fIMuLFoFIPoBR43YvM7pzbNGZIucD7HojH5GYyt1B/GvYJR\nDKIXeMtps/QZnzpqqbPtPoMzY9E06U6AfBVY79lvPHDdP8QSmB2LJoFFyK7AsG6zXQw6lsDsWDQJ\nHIoCigFUYFi32ZOufx5xukngcBRQDKLtNlt2auzE69aAzzgJHIoCikHE3WZ7Ns46K3bClL92gs43\ndboLRQHFQEG32W2/+mKs33+ugghMS9uFoYBioKbb7Iff/xQkhUi36FAUUAwUdJtNbL37rNjJZYDz\nTQKHooBiEPV7cGLLT86MnXDd86wXpgyQwKEooBhE2232lR8Pix1XstIzxHKzv+XWRJ5i6lUpTQHF\nINpus7Gjv7XsoI+3m/1ta2xs3DmmmnpVSlNAMYi22+wTB9Joe7O/BhbN0qlXpTQFFAPF3WY9tUnm\n1qh2PdWrsnbWrPktnSaAhDvZaG/nHGBgP2c/hoEcWx4DjFsUA8XdZn29KvXp5brbq9IUmHp0hKGA\nYqC426xvabumk+wblNWrUqdbdEgKKAaKu816s7/6HDNLSL0qpSmgGCjuNuutTdLPe9r4Qb0qpSmg\nGETfbdYPT21SU2yPsYN6VUpTQDGIvtssAiRwKAooBpBusw01q2uYEzpIYDTb/BE4dc3WlhaZjQyv\nq41eYEoXhqKAYsASeNS3lxsP4pL5yQaz46csqd1Uu7h4ggKBqRlpGAooBsxR9LRzlmv1Pxr1A2tr\n+MakltBmpAjQLToUBRQDpsCV876ySdNWjExewVOX1W2qW1I8ngSOgG2eCNwwzthfP8zaqplqdo8u\nmlITSmA3XejkCaVaGTJAAgsZsAXWnjrrwdqyK+3tzetW1zAbvqOWtnPyhFKtDBkggYUMOAJrDxXF\nL3zW2cEdamOWtrPzhHKtDBkggYUMmOlC4wGsbah0X31ZQSyYwJ50oZ0n9CYQSeBQFFAMOBenf3d4\ngT3pQjtPmNrzm1hsEi1tF4YCigFHYL+kk0IL7EsXWnnC1J6O5uYquoLDUEAxAAmsQYvPMuBJF9p5\nQioflaaAYsAROOOahRWfZcJNFzp5QioflaWAYhBt8Vkm3HShkyek8lFZCigGERef4UACh6KAYqCg\n+AwOEjgUBRQDNcVnQNDSdqEooBgoKD6Dg/LBoSigGCju+B5wBdMtOgwFFINoi8+QIIFDUUAxiLb4\nDAkSOBQFFINoi88y4WZ/e+4e1PeK7TqVj0pTQDFQXHzmyf4uOuPN1ltGJKh8VJoCioHi4jNP9rfk\nHl1vju2i8lFpCigGiovPPNnf3a26vqJfR6p89O8PP/xw22ETQMKH2ejs4BxgoJWzH8NAji2PAcYt\nioHi4jNf+Wh3+YBVbvnoqu98Z3bzIRNAwofYaGvlHGCghbMfw0COLY8Bxi2KgeLiM28+eMvoS7Ym\n91L5qBQFFAPFxWee7O+WgQvNTyoflaaAYqC6+MzNB0+82Rg+N3ZR+ag0BRSD4OKz+tK52u0zFiwO\nOS86lQ8+3WqjtI3KR6UpoBgEF59NGrdaK/nShPPD1QcLQQKHooBiEPyaVLTKGG+NCTmKFoMEDkUB\nxSD4Nemsak2rKlIicB29B4ehgGIQ/Jo0eXL12muuViMw9YsOQwHFIPg1ad34ePziNdq6iflwiwaa\nYu9kCNMCcwt6TaqrYhuRwPnvFtSjgw9MupCx+igJrNxt4GvSmvVyCX9PsWjm6qMksHK3genCW4dL\nTdnxpAv5q49G/7eRwCVQgcXApAvTVx8lgbPhVrXAnnRh+uqjjwwdeiO6fBRoiq2yRJgWmFsFq4/6\n4EkXpq8+ukfTVuTL6qMM0PrBIIE96UL+6qMowhkggYVuVU9893SbZaw+uuhxDrgHMvHYY3BbhFuE\nqSK3j0Xi9udLZfDYziCBPd1mM1cf/fgNgmq8meCLA0bYSBahQEAC93KQwL0cIoETY7fZv2XGogkF\nAr7AiaXFMUfgzFh0yw4e3uEeycC778JtEW7fUeQWbhrNH1ZfUy+DtgCBD5eVOQKzYtHbeLT+Af/b\ntm+H2yLcvok4vQi3CNN334zC7RzF78HJNyQT/Fg0AxToKJRAhytweizaLV3BVHhkIorSFQb2H1Ti\nFmF6cH8UbrMncHos2iw+a8dOnmIgijlZDBzs5hwAhvrlGXQfjIJB9gTuLbFo4OmVZ8C9RaMYZElg\nTiyaBBagoARmx6LzRmDMKSOB4SCBRciSwOw+wiRwrxGY3UeYBO49AjP7CEcjcB23sR/3QCZ6SSvD\nTHzSFgFbUStDQE/SaJqRYnp2ZqKXNCPNxGHe2UExEAgM6ElKt2ixrdQfRqPoJEjgnArsZn/LrRYO\nxYxWhnLngQTOqcBu9retsbFx55hqRitDufNAAocVGNCTFFObZGLRLJ1aGUKhXGBRWSFUYG/219ga\n1a6nWhm27NixhgQWoCBu0b5Whvr0ct1tZWgubfdxmwkg4TY29u/nHGBgH2c/hoEcWx4DBlqbI2Db\nFiwws48wVGDf0nZNJ9nfX2plCIHyK1jURxgqsDf7q88xs4TUyhAK9YMsQR9hqMDe2iT9vKeNH9TK\nEAr1V7CgjzBYYE9tUlNsj7GDWhlCkY1BFrePMFhgIUhgEbIzihY32yGBsWzzTmBOH+FoBKZOdyJk\naValyhkdtLSdCMrzwUlkYUYHkDDHCd2ihQxyOqODBA5FAcUgtzM6SOAwFFAMVM/ocPPBTiJYqlcl\nAySwkEH28sFOIliqVyUDJLCQgfI+Wal8sJ0IlutVyQAJLGQgFjhIfkw+2E4Ep/bUzpo1/0CXCSDh\nLjYOHeQcYOAAZz+GgRxbHgOMWxQDsYRBHaMx+WA7EZzaQwKHpYBioFhgXz7YSgR799AtOhQFFAOx\nwJNkBfbkg+1EMKM+WO6UkcBCBqpDlW4+2EkEZ9YHy50yEljIQHWo0s0HO4ngzPpguVNGAgsZUKgS\ny0CObT4JTKFKJgM5tvkksPriM0oXhqKAYpDT4jNK+IeigGKQD9WFQMIcJ3SLFjIggbEM5Njmk8BR\nFJ95koM9dw/qe8V2nVE+KnfKSGAhA8XFZ57k4KIz3my9ZUSCUT4qd8pIYCGD7KULS+7R9ebYLiof\nlaaAYqC4+MyTLtzdqusr+nWkykd1EjgkBRQDxcVnvvLR7vIBq6h8VJ4CioHi4jNvcnDL6Eu2Jvda\n5aMdzc1VdAWHoYBioLj4zJMc3DJwoflJ5aPSFFAMVBefuenCiTcbw+fGLioflaaAYqC6+MxNF55u\ntVHaRuWj0hRQDHJafEYCh6KAYiAQuL50rnb7jAWLa0hgLwM5tvkk8KRxq7WSL004X2F98MEI2nv2\nkmakCLcoBgKBi1YZ70hjJFs4CEH54FAUUAwEAp9VrWlVRSoF7u236DxwK7pFT55cvfaaq0ngNAYI\n2zxwKxJ43fh4/OI12rqJJLCXAcI2D9yKX5PqqkRHQQJ7ikUzl5fFCww0xZ4HhGmBuc1i+ShjeVkS\nWLnbLOaDWcvLksCq3aoW2JMPTl9e1l2cEkgYYYqdfIgwLTC3qmdVevLBtLws0rQglpf15IPTl5d1\nb9EM5PHilAEDoJQAABJoSURBVHJue9sC0Z58MGt52S1vcPAK70Amtv4dbotw++rrStwiTF9/NQq3\nqpd497QTzlxedsvDBNX4/Yv1MmgNEtjTTjhzeVlCgSBsJItQICCBezlI4F4OkcCJsdvs3zJj0YQC\nAV/gxNLimCNwZix6y9/WcvA870AmKivhtgi3CFNFbl/gnh2M2/ue/asEljUGCHy4rMwRmB+LZoAC\nHQUS6NCTb0gm0mPRezRtBQksQKEJnB6LfmTo0Bv3HeSgmXcgE60H4LYIt/vbOQeAoX55Bu0tUTDI\nnsD5HIsGnjK0rdQfVhCLU+quwKxYNAksQEEJzI5Fk8AiFJTA7Fg0CSxCoQgsQG+fVYlgwAAJnAQJ\nLGRAne4UTp5CMGAgS0vbKRW47bAJIOHDbHR2cA4w0MrZj2Egx5bHgIGu9gjYHs7p2oV0iw5FAcWA\n1i7EMpBjm4cCyzUEd5OD5VYLh2JqZShNAcVAdUNwNznY1tjYuHNMNbUylKaAYqC4Ibg3OWhg0Syd\nWhlKU0AxyF7pirk1ql2nVobSFFAMsle6YmB6ue62MnRLV4CEObUZUZSuYBjIseXWmCDcohhkr3RF\n15tOsr+/VivDUMVnDPSSQAfCLYqB4obg3uSgPsdMIlErQ2kKKAbKG4K7pSv6eU8bP6iVoTQFFAPV\noUpP6UpTbI+xg1oZSlNAMVDcEFwMEjgUBRQDxQ3BxSCBQ1FAMVDcEFwMEjgUBRQDxQ3BxaBWhqEo\noBiobgguBDUjDUUBxUB1Q3CxwHSLDkMBxYAagmMZyLHNP4ElZ3S4+WAnESzVypABEljIQPWMDjcf\n7CSCpVoZMkACCxkontHhyQfbiWC5VoYMkMBCBopndHjywXYiOLWHFsYKSwHFQPGMDk8+2E4Ep/bQ\n0nZhKaAYZDMfbCWCGeWjQMKc/wVdwUIG2WtlaCeCGeWjcqeMBBYyEAscJD8mH+wkgjPLR+VOGQks\nZCCWUBDEAgrs5oOdRHBm+ajcKSOBhQxUCywECRyKAoqBWOBJJHAGAzm2eSawprb4jMpHQ1FAMchp\n8Rnlg0NRQDFQXXwmBN2iQ1FAMVBdfCYECRyKAoqB4lClGCRwKAooBsrnRaeyvz13D+p7xXad6oOl\nKaAYZG9pu0VnvNl6y4gE1QdLU0AxyN7SdiX36HpzbBfVB0tTQDFQXHzmyQfvbtX1Ff06UvXBtbNm\nzT/QZQJIuIuNQwc5Bxg4wNmPYSDHlscA4xbFQHHxma8+uLt8wCq3PpgEDksBxSCL+eAtoy/Zmtxr\n1QfrdIsOSQHFQCBww+9u+pNW+1i1jMCe7O+WgQvNT6oPlqaAYiAQ+Kcjrx819+xzhj4sIbAnHzzx\nZmP43NhF9cHSFFAMBAKf/yftkfg87VcXyQjs5oNPt/pkbaP6YGkKKAYCgT9fpVXF12q1Q2QEFoIE\nDkUBxUAg8CU/qp8xaOZLd11KAnsZyLHNJ4GXnBs/f+kX4+cuVSYwpQtDUUAxEL0mbX6hQWuo3Cyw\noIS/0DYP3Oa2XzT+Fg00xd7JEKYF5rbQWvoDTUngkmwJ7CkWzVx9lARW7jaL6ULG6qMksHK3WUwX\n8lcfjf5vI4FLsiSwJ12YvvooCZwNt1lsJ5y++ugD/fuXostHgabYKkuEaYG5zWL5aPrqoy07dqzJ\nl6XtGKD1g0ECe9KF+bz6KAMkMEhgbzthxuqj9/+Wg+t/wzuSgTtmgk3nl4JNf3vrbLDpvTfC3ZbN\nBZvefTvc7XULeEd+tlQGj+8KEtjTTjhz9dG2D3nou4l7KB233AQ2feUYsOmH31gANl12LtztmSvB\npr+8Gu7206/zjuyG3wb4CBvJ4uP4D8Cmd84Em+7qA2cw8U9g06oxcLcjXgKbPnQN3O2nP4bbhkD0\nAk/5CGz6lyfBpvsmwxnMrwabvn4H3O3tb4FNK++Hu/2PVrhtCEQvMCGvQAL3cpDAvRwqBD6gwGeh\nAXMOlJ4vBQLXjINa9mBiB3DbPHALPwc4WzwUCNx+ymaY4ZJTYxe9B3SKsM0Dt/BzgLTFQ8Utev7V\nILNX4+v3X3sa7CUBYZsHbnXwOUDbohG1wH+c8Jre/rnXIaZ3/1zfOPD5haDAMcI2924R5wBlGwpR\nC/zxj477zx3zeVOLffhVmXHG9L5vB5hV35FogdrqehPcFGWLYQs9Bxjb0IhU4OVnDv6L/uH3jpva\n5x2A9fv9T3le335iW4DZ1gFfvwpqq289tgpqirLFsQWeA4xtaEQp8Ib4U8vMb/j7pZ/+bpCt+V2o\n6PfTR4b+PtDtj2PfS+hAW/2+PlVQU4QtlG3Vn5NsIecAYyuBKAWetsD48X658WPbiY1i0+R34ZUb\nrnku2O0zdQNuS4BszevAUA3kFmMLZru89J0kWz34HGBsJRClwDfNMH5U9zcpT18oNnW/C0HQfvGa\ncS+7LRFkZ1wQO442kxd39amK0tYEnO1Gw63DNugc7EDYhkd0Amu/ePjYDbr+0mkG408uXyM29nwX\nxHis30WGCFsHXPvvLWLD5Uc9qVcY/+kdZ/TZEeAUY4tiu37ksqOetNkGngOLAtQ2NCIT2FTiymNn\n/3HoI8ZG4kXRyaj/8asbUt8FMd7qs1W/2FD4tXF/DDKtsFX7139sCCSLsUWw1TtPX226tdiKz4E1\nHgXaSiEqgZNK/KT4/z0baLpm1Nf7VN3nfBfEeLDfoTe+NK/Po6sBFCzVPnv5oKWR2mLY6vofvpq8\nNQTBHo+CbOUQlcBwJd6Z1mmObqoDvwuJvQn99UkdF65PnDk48PlnPqrN01Vz48pABhhbKFvjsf6H\n6943LuENhuutQaapp3qQrTyiEfipxWAldhw9NmGNX4MMN50Zu+hj42OwvuOiQALJRzXsgsDYgtka\nY+Lt1/b5ftNDV+v6G4Gm7lM9wFYekQj8zlWvQJUwbkufeVA3z1nA6Kaq/0OvjL3VnIx1/dl/DvLp\nPKohFwTGFsw2OSbefm3fW/ttCzBc/4phCn6qSyMKgXccXdQFVMJMuSWvnYDRTVX/53V9XalhXjNt\nhdi00/OACLggDFOwLYKtFU21DN+e8unHxD4Pnf7SXsMU/FSXRQQCNxlf8/t0gBJOyg1wd6w+zowp\n3HHMZ+/qCjJ95WLwo9o0hT/W4WyTMcekYVOAbespX/+GaRo8BokG8gJvPRb6QEul3ILvjlYA4KEL\nP9x+5qWbAkw/OnatDnxAJE2hDxMo21TMMfgsmOOrO63Iq/rhs40IruD7QEMW829LpdyC746Gwg99\nscW4IC7/WZDpzEt14AMiaQqxxbB1Y46BX1xzfPVXy1b98NmGrMB2SBfwtxlfAUR2zlD4FCt2FTwO\n+Vcf8yIHPCAcU4Atiq0bcwz84loRUstW+fDZhqTAbkg3kLBxkSOyczooAF1vTXb54USAO4SpCQTb\n91Y8moo5QvzC/rSoIHsFI0K6hik4k2ci+DT03Phv/7cqoTcd+2agM4RpEmC2j/cd+EAq5iiGlXVM\nKvw8kIY0ZAROxVMBIV07egTJztX/+NXkL4DT0Djj+AuePnzbVABbhCmG7VsnvH5Y3/U4PD5pKbwH\nwiISSAjsxFMhIV1E9MgK/iZ/hZyGlnmnnXXvMf8EWMJNMbGu31+q11/yqeGPAgZNdnwya+MrCxIC\nO/FUQEgXET2yg78IHp2Pjoj9JUpTVKzr70cNG/jg3iHLAYMmJz6ZrfGVBQmBwVlS/V/w6BE4+Fvl\nsehpANsGmeootubTZMuqLr3xxMDKw552PXvxSQ/CCmw8f2+G8l0/8jVw9Agc/P0d4irH2GLY2k+T\nH5YN+K8gUysqlrX4pAchBbaev1C+naevBkaP2nYCQ9U66CoPZQtn6zxNtLteDDK1o2LZik96EFLg\n5PP3O0C+f/gqLNK09ITPjHkdOLipvOo4sGoYWzhb8NNEx83GjxYhBYY/f500OCDStO/E11t/eNwW\n0OCm9nMvvn0TUDWELYItIpUInjavAGFv0eDxQioNLoYZ/H3j+E90/Y5zQGG8yT83ftwOUw1hC2Sr\nIxKfxlO97IerMTG8SIEWuOoLT5oZPODz1zwNoDS4GYbv+fyvdf1gH1ArjKnfM37sOQE0KRJsC2aL\nSCUaQ7ELy6/vMwETw4sSaIG/M2Lw/3rAOBGg8YJdchmcBk9GtWuPWtD9z+ODu84YQtQes17XX7sy\neCyGsEWwhSc+9Xeu2Gt8f49+FDTJP3qgBd44oK3iwpN/vhdi65ZcitPgqRkR6047pf9DgW6Trxx9\nfnjf4OBJkQhbKFsT8MTnjqNPMdveFd8Y6FMN8M/grz6qb+z/hWNvA4T8oGNHd0ZE18s7A73aQrx4\nzYTgSZEYWxjb96eNndEBHDRZthWf+aXx688UFhAKgRf4r+cZf1viuQvuCTJ8ajF47AidEWF5hb9y\n4GxBbF869SfLf9kDqzW0bSuOmtP+9hmQmd0qgBe4Z8QJZpIncAD9zlWvgPO/0BkRllfw1wZpC2G7\nJ/6C+QGqdnRtP9Pn/CcCGSgCXOBUQHfhyOC3o0RyqiUso2qmHcEzIoq6oF8buC2c7a+nGD+e+/c+\n3+oKTiW6ttmbgZUJuMCpgG7naUFBgFe/2Le03ZpqCcmoJtOOkBkR5kVueIV8bczvINAWw3bGNH3H\n/x/28IazFwVQ9dvmUGHELToVlJs3SnwJV504r3r0TPBfZacdA2dEWEJYXgOFSJrCbFFst/zbOcfc\nZYyKSxfgbCuOegHgXQXgArsB3ZZ5QkNrzvpTXwaXVgHDnrYQEK+OKcgWx/bVx80XxLdPfhdpu64D\n8A9UACwwOKCbnLN+4ywdNGgCp0lTQgR7dU2DbXFsTbQvPlR1BjAmhbFVBLDA4ICuPWf9oPl74KAJ\nnCb1CBHk1WsayADF1sLLn4sNgD5RMbaKABYYHvx15qwDAE+TeoWIzlRHsbXR+W63Els1gAmMCf66\nc9aDgUiTIoTAaQZnW5AACYwJ/pqATOzG9TfTUULgNMvmNPTsAyIwJqCbBOCcofqbQZ2GMEVbFxgg\nAoeYbwKYs47ob9buOFWmcNYKDbIOiMDwG2mqKAEwZx3e32zr4PftX4KFeG9FK9TUg+wVGmQdgQIv\nP3PwXOiN1FOUEARz8gTwMquadPLn30/+GiiEWSgEND1CECSwFSgG1owhihKSkydAFXnrT61973uO\nwgFIFgoh1pXr/QgSOBkongaZb4KYRupMngBU5OnT5ho/fgBT2C4UUrsSUWEhSGD4/FjMNFLMsO37\ntxk/9vYf3gywdQqFIH6PEAgFxnTxwxQlwIdtxqP6ZZPBPybNmABhACwUOoIgEhjTxQ9clPDxfbP+\n1g19/7Ue1fcfO+v3Q545eHzAihhJBqBCoSMJAoHBPQd1RFHCuoHfvOHUr30IG7bZj+qaa7/1jN5+\n3PsgBoBCoSMKfIHhhZyIooSq/s8Y99LioS3B77/dnkd14un1l10nsE14GBB8EFzB0DEToighmcnT\nD18evJboP858231U7/zy6XfwVwu35m9gyiKOJPAEdjqyQsZM8KIEJ7Sx/ajA5dx6bhz0NuxRbc/f\ngJZFHGHgCAzuUoEsSrAVThwH6YszCBRhSc3fgDE40sAWGN6lAleU4Ci8vh8g2mQqHPyo9szfgDE4\nwsAWGNGRFVGUYMFUeOdg8SLs1Z+7bPZz+yyFQf7g8zeOPGQKbA6K4R1ZEUUJNrYOuGHYz4JMzrz1\nC0edXfrfX4MpjJxzc0QhU+Dl1sUI7MiKKEpIYeuAnwWb3JY4VD//W6fFAEGLXj7nRhKMW7R1u4V1\nb4UXJXgByOQ5g7HgQk69l8/IkEWGwMb7kaUwqEsFtCgBD5RopDAf6QJb70fgIRMm14QEUuHeO+dG\nEmkC2+9HIIUV925DiUbzN3hIE9h5P5oaPChW3ruNRIsCaQKn3o8CB8W5691GwMAV+MOHXzBvtdDl\nj3LXu42AQUrg6pO/2s9suwF4P0IXJRByB0fgqv7P6m/Fzd+C34/wRQmEnMEW2ArZdw2ZuQQ0IEYt\nuk7IKWyBzbfOxE2Xzz55NuQfwYsSCLmGc4s2FL5tUre+7tjApeQwRQmEnCM1yNo6YHi3rm877nDQ\nv0AUJRByD/c1yUrUDvtd0D/AFCUQcg9PoAOQqNXp/bfQ4I1kBSVqrTnr9P5bWPCFKsXR3+Sc9fX0\n/ltQQPSqtOesP0Lvv4UEsMDunHV6/y0kgAVGzFkn5BHgt2jEnHVC/gDRbRYxZ52QN8B0fIfMWSfk\nGVAt/WGhEEI+AbdmA2DOOiG/gFyUgybCFRokFogmFAJI4F4OEriXgwTu5SCBezlI4F6O/wE2XMS3\nDfj7ngAAAABJRU5ErkJggg==\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -h 550\n",
    "ggplot(df, aes(variables, values)) +\n",
    "    geom_bar(stat='identity') +\n",
    "    labs(y='Value') +\n",
    "    facet_grid(BD_window ~ .) +\n",
    "    theme_bw() +\n",
    "    theme(\n",
    "        text = element_text(size=16),\n",
    "        axis.title.x = element_blank(),\n",
    "        axis.text.x = element_text(angle=45, hjust=1)\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAiYAAAD6CAIAAAAazxtsAAAgAElEQVR4nO3daUAT59YA4DMJWyCJ\nCYvI7orghqhIa7WbiguVqhU37K3XqtXeatVS23rtVau1Wil+V1v3qrW4L1Sr1tqqtaIoLte1YAEX\nQPYlBAiQZGa+H9OmKUtMwjCTgfP8CpNJ5hCGOZl3zpyXoGkaEEIIoeYn4jsAhBBCrQWmHIQQQhzB\nlIMQQogjmHIQQghxxI7vAOrS6/U6nY7LLdI0TRAEl1tkkaCDB4HHj8HziPv47ezs7O3tudxii2Rz\nKefy5ct6vd7OjrvAtFqtg4MDZ5tjl06ns7OzE+ixg6ZpvV4v3H9jQe85gg6eoiiKorg8Sjx+/Fgu\nl48aNYqzLbZUNpdyACA8PFwikXC2uaqqKhcXF842xy6NRiORSASaciiKqq2t5fJvzS5B7zmCDp4k\nSZ1O5+TkxOVGy8rKuNxcS4XXchBCCHGE/bMcmqbDwsISEhKCgoLqP6tSqWJiYi5evPjcc8/t3r1b\noVCwHoD5clVwNweqqu2lEujmAz5KHmOxWG5x0fUz56CyCkRiBx/Pl18ebM/hOEMTkRTcyYHcMkKn\ns3NvA30CQCKoMZ6cUvgtF6qq7WXO0NMPPOV8B2SJyhq4/gjKKuwdHaG9OwR78x2QJUgKbmVBbhlB\nUXZtFRAaAI6C2esRALsph6bpAwcOJCYmXr9+vbF1YmNjZTJZenr6nDlzYmNjt23bxmIAFrmZBYev\ngdIZgBYTBPz6O4zuA33b8xWOZe5l/i77JsHPQaolCIIgpFlPfnz8ZMjUKU72AjhykxQcvwl3n4CL\nA5CkKK0Avv8ffPAKKJ35jsw8VzLhxG1QOgNNiYGAX9JgQjj09OU7LPPkl8N/T4OrFMQgFong2kPo\n2x5G9AJBDM3qSDh+E1JzQWJPAIhS8+C767A4CqScDrChJmFzYI2iqHPnzpk4caEo6uDBg/Pnz/fw\n8FiwYMHhw4f5ardTrYWDKeAuBbEIAEAkAncpHLkGlbW8hGOxh2eTihzlWrEIRARNQIWDfcfc/PPn\nf+E7LrPcyYG7T0DqCABAEOBkDx5ySE7nOyzzlFfD9zfBzQVEBACAWATuUtiTDLWcVlla73ImtJWD\ngxgAgCBALoFrjyCjgO+wzPO/x5CWBy6OfyRIJwdwl8KlDL7DQpZg8yxHLBZv2rQJADZv3tzgCiqV\nSq1WMwNugYGBKpWqvLycSVEpKSnnzp0DAGdn5969e5MkyWJg9WWXEE72BJPvaJpmMp+zPWQV013b\n2XrTOa1e1zM7q0AiNV6osXfQZuU19+fGisdFhPOfHz4A0DQtJuDCfRjWg+I1LrNkFROSenuOxB5y\nSun27ra+51Tr4OpDkYcUaNpot3eAh0V0Rw9bDx4AnpT+8eHTf7ITw7lUYnBws+/2FCWAnVMQOB0H\nZUo+mDoZqVQKACUlJUzKUalUDx48AAB/f3+dTicWi5s1kppaEdD2FEUDAE3TzP5EA1FTq9NqbX3f\nqqqu+uvwQAMQAAAU0ARJarVa3sIym560oygR80X1rw+fJqprtGKbL2ep1YqBtquz54BA9pwaLQGU\nQ93gadDqKK1Wz3NwZiBJO4r6Yxf5a8+hCA52e0F8mRMETlMOk100Go1cLq+srAQApfKPS/YRERER\nEREAkJSU5OTk1NyFs74eoCVB/mdeYzKclgQ/D0fbL9mVSCRXvX3dy8ppoOHPIXgJSdLuSkEUHLdT\nQkYxOIn/OGqIxWKahvDOIHURQPA+bqClQCrUPQdIGsR/D76mFnzcxBKJAO6OcpXBw1JwEP9xliMS\niSgKBgQCB7u9cO8eszWcfqtUKpVyuTwjIwMAMjIy5HK5IeVwzNUFXggCdc0fP9I0VNTAwEDwkPES\njsWcenV1q6kU/Xmy40BSCl31My+9yGdMZuvtD2VVoP3zWyNFQ3EFhAbwGpPZvBTwbCeorAHDwKC6\nGgZ3A7nN5xvGGwOhuBKoP4Ov0UGQF3QXSO1D3w5QWgm6P8/H9BQUVwqm5AcxuEg5hw4dUqvVACAS\niaKjozds2FBTU7Nx48bx48fzeA/jy93g1VDIL4fCClGBGiJDYHA3vmKx2EsDBhW/Nirfw81TU9Gu\nujK7g5947jseCmFUebs4wsdREOoPBWqiQC3q7An/GgLt3fkOy2wRPWF4LyhQQ2GFyN8NRveFFxu4\nHcBGBXvDjBegowcUqEUF5RDeCUaG/FFNYPvaSGDRKOjlD/nlRIGaCPKCecMEdm8DIpqjZowgiNTU\nVMN9OcY/qlSqyZMnX758ecCAAQkJCfXL25KSkvr27YvdB8yE3Qd4JOg9R9DBc999ICkpqaysDBve\nNF2zXMupk8aMf1QoFCdPnmyOjSKEELJxNl8hhBBCqKXAZhEIWYzWVNEPM4niIsqjLdGpC+GIt78j\nZBZMOQhZhn6YqT20RySTiwDI32j6RKJ9zDTCWyBVXwjxCgfWELIAXa3RHdkr8mgLTk5gbw8SCeHe\nVr97B+gFcCslQrzDlIOQBejcHJD8vf8oQYCTE10okD5lCPEKUw5CFqBra4Go918jEtPamoZWRwj9\nDV7LQXzQ66l7t6mSIkJPkjKZqGdvQiqMxg8ipSup08Lf7yWidVpC4cpXSAgJCKYcxDm9jjx7mnqQ\nTjs6AUVRNEUmnbN/81+EqxvfkT0d4eUj7taTepgJfzbdomtrxL37EQJp/YAQv3BgDXGNSvuNepAO\nTn+eKNjZg5s7efMar0FZQPzci6Kg7nRJEaEup0uKxD1DRQMG8R0UQsKAZzmIa1RBHu3kZNyihxCJ\nqRtX4aUIYUxOKZGIB70kHvC8pqTY2c0dmnmiDYRaEjzLQVwjAAiom1pooIGnKWKtJBbTLlLMNwhZ\nxObOcmia1ul0dnbcBca0CORsc+xighdYW882SjrjPjg4GuZ2BJomeobqSBIENRFWC9hz+I7CShRF\n6fV6LuPHKdrYYnMphyAIe3t7LidE0mq1wp1/SafT2dvbCyzl9AzRnTsNbdqAvQNBEARFUWUlDiNf\nJYTzV6ildNfVaVkVuR3Br4+sqx0hvHMdQe/2JEnSNM1l/M09T3HrYXMpB7V8jk727y4kb14nz58B\nihI9M9AupI+AKr7uVj4Yc+sDV3u5HS3W5emHuvZ/02dUR4kP33EhJACYchAfHBzF/QcQ/Z6pra0V\nC2q+nGqqdsythYEu/iIQkSQpFotvVNynnlArOs0S179FFDWH6mrIzyW0WvDxBYHczoUMMOUgZIHr\n6jQ3+zYio7obO0J8uuRKTLthPaSdeAyslaDu3tKdOkY4SQggtNpa8XMviMOeBREme8HAlIOQBcr0\nFQ5E3UsIDiKHMn0FL/G0KtSjTP2ZUyL3tkzViUgqJa9fISTOol6hfIeGzIXfDhCyQFsHZQ2trbOw\nhqz1dMCGN82OfphJyOXGSwhnF/2Px/mKB1kBUw5CFugj61qqr9BRf01VUENpIz2e6+Lsx2NUrQVJ\nNtBTVVDVmghTDkIWsCfszvX5MlzRPbumoEhXllWd/4Iy9F2/CfVvbkXss7MDmvrbEpoW9erDUzTI\nGq34Wg5FUb+n0k+y7W5cJfuEQTtvUXB3QoTV9+gp/Jw8l3ac/obXyMfluZ0U/v5OnnxH1FqIOnel\nbt2ANgrDErqqUoTzsQpK6005+uRfqf9dI6Qy2tWNynpEp92D0hLxoJf4jgsJAAFER4mPJ6VwcXLh\nO5ZWhPD1Fw8ZofvhKFOxRmtr7Qa9JOoRwndcyAKtNeVUqKkrl/7qlk8QhIuUvJos6hFCKPE6cLOr\nobR78n5M12QTFHg5e0zwHNLWQTC3giIeibr3cuzYhSwu1Gu1jp5eIJXyHRGyTCtNOVRRITg61l3q\nJIHiIsCU08yqyOqVD7/5VXWzjdiFoqhrVfe/yj50rPeaQGd/vkNDQiCRgLcvrdOBkxPfoSCLsVw+\noFKpIiMjFQpFZGSkSqWqv8KjR48GDx4sl8uHDh1aUMDfdPGiBi/20jTeU9b8jhdfvFB2U2H3x/dT\nR5F9R4n3vvyf+I0KIcQBlo+wsbGxMpksPT1dJpPFxsbWX2HKlCl9+/Z98uRJUFDQ3Llz2d26+URt\nvehqTZ2FdLVG5OnFSzytyv2qx3L7v10CERPi3fk/UkA19hKEUMvAZsqhKOrgwYPz58/38PBYsGDB\n4cOH6b/PgFJaWnrx4sUPPvhAJpMtXrz4u+++0+v1jb1b83J2th8eRZeV/tEtn6LoshL7ISNxaJgv\nBBDCmi4HIWQFNq/lqFQqtVodFBQEAIGBgSqVqry8XKH4q6JRq9UCQHl5uZubm729vVarLS4ubteu\nHYsxmE/UI8S+rSeV9pu+opyQyuyCuhN4isOJri4BP5dcbWP/V3YnaXJyuwhsi4lQi8dmyikrKwMA\nFxcXAJBKpQBQUlJinHI8PT2Dg4O3bt360UcfrVixAgBqamqYp44dO5aQkAAAXbt27datG83NN16p\nHPo9U1tbSzOlBJq6Q222T6vV0jQtrPlyXnYOvS5Lvay+JxVLaJqu1tdk1RR+FjBbI7TPX6vVCuuT\nNybo4CmKIkmSorgbiRXudHa2hs2Uw2QXjUYjl8srKysBQKn8W+UrQRB79+6dNm3apk2bZs+eDQBt\n27ZlngoICBgyZAjz2NHR0YnDWhSKorjcHLtomnZychLWscMJnBZ3mra/4OeC2tJv83+Y7hMV3Xaw\nEG+oFPSeI+jgSZLU6/WO9YtOmw2X8xSbKS0tLTg4uLFv5wRBpKamMmNONoXNz1GpVMrl8oyMjD59\n+mRkZMjl8jopBwAUCkVycrKDg8ONGzf27Nnj7OzMLA8JCQkJCQGApKQkkUgk4rByjCAILjfHLiZ4\nYaUcAJCLXGb4vkpR1HveEyWCmi/HWAvYc/iOwkrMmT3HRwnOtmU6lwgdm38zkUgUHR29YcOGmpqa\njRs3jh8/nvk7HTp0SK1WM+tMnDjx448/LioqWrly5RtvvMHi1hFCqAWQyWQTJkzgO4rmwvLXhLi4\nuNzcXG9v74KCgjVr1jALo6Ojc3NzmcebN28+depUYGBgmzZtFi9ezO7WEUJI6Hx8fPbt28d3FM2F\n5ZSjUChOnjxZWlp6/PhxQ+EATdOGIcVevXrdunWrrKzs66+/trevO9UVQggJ2vjx48PCwgw/lpWV\nOTo6rlu3DgBu3rw5YsQId3d3Jyennj17Hjx40LAaQRDnzp176aWXwsPDmR/T0tKYp0y8CgBu3779\nwgsvyOXy7t27b9u2rcHhOIqitmzZEhIS4uzs3Llz5//7v/8zrHb37t2RI0e6urq2adMmIiLCsNHm\nI9TBXIQQskFTpky5du3agwcPmB8PHjxIEMSUKVMoiho5cuSjR48+/fTT/fv3P/PMM5MnT66o+Gsy\n2XffffeNN97Ys2eP8bs99VWRkZEjRoxISEh46aWXZsyYsX///vohrVmz5r333ps0adJ33303d+7c\nlStXrl27FgBIkhw2bJhSqdy4cePmzZvt7Oy4uNhB25gLFy5oNBout1hZWcnl5thVVVVFURTfUViJ\nJEmO/9bsEvSeI+jg9Xp9dXU1l1u8cOHCsWPHzFmztrbWzc1t5cqVzI/PP/98TEwMTdMqleqDDz5I\nSUlhljOXt1NTU5kfAWDr1q2GNzE8ZeJVqampALBjxw7Dq+bNm9e/f/8670CSpFQq3bRpk2G1Y8eO\nPfvsszRNZ2dnA0B6ejqzvLCwcPv27RZ9LFbAsxyEEGKNg4PDhAkTmLONrKysX3/9dcaMGQDQpk2b\nVatWKZXKQ4cOLV68ePDgwXVe2KdPA3PNPfVVo0aNMjweO3Zs/ZGx7OzsysrKWbNmEX+KiorKyMgA\nAG9v7wkTJoSFhU2dOvWbb75xdHT85z//2eQP4Ckw5SCEEJtef/31W7du3b9/f8+ePYGBgc8//zyz\nPDY2dtCgQT/88EPnzp23bdtW51WGO0bqMP0qYzRN179XSSaTAcCBAwfyjNy+fRsARCLRvn37bty4\n0blz5927d7dv3z4uLs66X9l8Nnd/E0IImaanSR2tt9kbWcPDwzt16rR///6DBw9Onz6duVektLQ0\nPj4+JyfH29sbAB4/fmzOWz31VceOHTOcmhw+fLhXr151VnB1dfXz87t+/Xp0dLThJSkpKStWrCgr\nK/vkk0/i4+OZ4uETJ05MnDixwXbMLMKUgxASjMc1+XvyftyZexKAnuw17LW2L/WQduQ7qLoIgnj9\n9dfXr19fXl5uuCDv6Ohob2+/Y8eOwYMHZ2VlffbZZyKR6OzZs+3btzfRBsLEq5gVli9fXlhYGBwc\nfPr06Q0bNvz0UwOTgCxduvStt96ys7N77rnnbt26tXr16g0bNgCAXC5PSEhQq9XDhw8HgG+++aZb\nt26sfxp1NffFIkth+YBFsHyAR4Lec4QYfE5NYZeL0RE35kbeWDDy+vwRN+YFX5zwP/V9DjZtfvkA\nIz09HQDGjRtnvHDfvn0dOnSQyWQvvPDCpUuXPvzwQ7lc/ujRI9roaj/D+MfGXsWUD9y7d+/FF1+U\nyWQhISGJiYkNvgNFUbt27QoJCZFIJF27djWuU7hw4UJ4eLiLi4tCoXjllVcyMjIs/2wsQ9A21lYh\nKSmpb9++XDZBqaqqYlqRCpFGo5FIJIJreMOgKKq2tla4DW8EvecIMfivsg99V3TeSeTIHLxEIhFJ\nk+Fteizp+GZzbzopKamsrMz4Wj2yDpYPIISEoVhX7iT62+VxMSHek/cjX/EgK5iVctLS0vr3779j\nxw4AWLNmjUwmCw8Pz8zMbObYEELoaQR5ht96mVU+MGfOHHd391deeUWtVq9cuXLfvn0bN26cP3/+\nsWPHmjs+hBBidHPp8FPpFaWd3LBEQ9X+y28cjyHVUV5ezszbYhG5XM6UMrcGZqWcK1eubNmyxcPD\nY+fOnWFhYZGRkVVVVbNmzWqOgJiLXVxOvsRskbPNsYsJXrjXclrAh893FFYSYvCRbgPuVmaeKkl2\ntZPTNK3SVhboSid6DOHgFzHzmndRUREAWHR5UqPRaLVaJuUY/pHFYnG3bt3Wrl1b/95PA87mOGB3\n6h2zUo5YLGampjh37twLL7zALGy+afIw5ZhP6CmH4781u1rAnsN3FJZxALt/B7zxrKzH/yp+r9Vr\ne8g7jXR/1oGwt52UAwBKpdLNzc38dy4sLKyqqjL8eO3ata5du1ZXV2/fvn3cuHEFBQUODg6WxWrj\nzClrGz58eExMTHp6ukKhePDgAUVRU6dOZbr0sA6LpC2CRdI8EvSeI+jgbbbHWnp6enFxsUXvXFBQ\n8ODBA+YxGFU2MwN0+fn5zI8HDhwIDg52dHT08vL6/PPP6T97rJl+NiEhYeTIkW3btl28eDFN01qt\n9sMPP/T29lYqlW+99ZZWq6Uoau3atR07dpTJZOPHjy8pKaFpurq6eu7cuW3btg0ICNi4cSP8vYC7\nicwqH/j888/Pnj3bpUuXYcOGdejQYf78+cePHzdMh4MQQohFNTU1O3fufPHFFz09PQGgqqoqJiZm\n0KBB+fn569evX7hwYWFhoWFlE8/m5OScOHHi0KFDK1asePLkSVxc3OnTp5OTky9dunTu3LktW7bs\n379/y5YtJ0+efPjwIQBMnToVAFasWPHTTz+dPXv20qVLR44cYfl3MzM16fX67OxskiRpmi4tLdXr\n9WwlvTo4PsvRkDX3SjKq9Jx+Y2IRnuXwSNAnCoIOvgWf5Xh7ewcEBHh7e4tEonPnzjHLa2pq7ty5\nU1FRQVFUcnIyAKSmphrOckw8W1hYSP85gpqamhoYGHjkyBHmPS9fvvzzzz+//PLL+/btY5bk5+c7\nOjqSJNmhQ4eDBw8yC2/dugWsnuWY2/BGLBb7+voyj5VKJct5jw8VpGbHk+Mbco44EGId6N/yGfNP\n71fa2En5jgsh1HqdOXOGuVC/Z8+esWPHFhcXi0QiR0fHkpKSDz74IDMzs/5lfBPPuru7g1FVQlZW\nVocOHZjHzFxw06dPnzhx4sSJEw0vKSwszM3N7dy5M/Njly5d2P0FzRpYo2n6iy++GDBgQNu2bfPz\n8xcvXtwC5kmNf7w3seh8F2dfX4e2nSV+x4qS/i9rPw221YsBIdQ6DRs2rKysLDc3FwBqampGjBgx\nbty4u3fv1p+EzcSzdQqLvLy8cnJymMdXr149fPiwh4fH999/z5x/6PX6vLw8T09PX19fZnYDADDM\nNccWs1LO2rVrP//880WLFjElgP379581a9b27dvZDYVLv2uyvis6LxX/VcvoInY6WvTrb5UPeYwK\nIdQC6fV0Thb1eyr9JBueVlxXVVVVWVlZUVGxd+9eLy8vLy8vAFCr1dXV1cHBwRUVFStWrAAA47t/\nTD9rbPLkyZ988kl2dvaDBw9mzpyZmZk5bty4FStWPH78uLS0dN68eePGjSMIIiYm5j//+c9vv/1W\nUFDw4YcfsvY5AICZRdJffvnl4sWLX3nlFebHqKioDz74YM2aNdOmTWM3Gs48qS1yFtWdWMJZ7JRd\nU9jd9hrTIoQEii4sIP+XQqXfB7EdQeqJLl3FYQMI10arqPv168c8CA4O3r9/v1gsBoC2bdsuX758\nxIgRcrn87bffnjZtWlRU1NmzZ5k1TT9rbPHixRUVFf369dPr9ePHj583bx5BEKWlpQMGDFCr1S++\n+OLevXsB4KOPPiotLX3xxRednZ1XrVp1/PhxFj8Qs9p6SiSSY8eODR06lCCIvLy8du3anTx5Mjo6\n2ricnC3ctPW8qv7tX6lxno6uAECSJPN3LdKq4gPfHaDo2aybZhe29eSREDtjGgg6eJIkdTqdiZ7/\nrDOzrWdGRsbf7svR67VrVxLuHkD8OZ5EUYSPn3joCEIkZhYw9+UYLrG0eGYNrHXp0uV///uf8ZKk\npCS2bkblRU9pZxVZaXzlhga6VF8eIuvMY1QIoZaEepIFzi5/5RsAEImo+7/RhQX8BcUzswbW3n77\n7ffff79NmzYAcOHChWvXrsXHx+/cubN5Q2tOTiKHvT0+mXz3P54Orna0WA9kQU1pQs9lLmKhfuNG\nCNkcjYYQ1zvGisWg0fARjU0wK+XMnDlTpVIxE5SOHz/ez89vy5YtMTExzRxb8+orD0rpv/2XshuP\n1Ln+Mq8XlX2U9q2lsx5CiAsuUlqvI+BvA4AESRLS1nszhlkpRyQSffjhhwsXLszOzpbL5S3jvhwA\naGMnfdXj+SpnAQ9qI4RslsjHD6o1tMSZEP9x5QZIvSioO+HhyWtcfLJgijaRSBQQEGA636hUqsjI\nSIVCERkZqVKp6q9w6tSpXr16ubi4DBgw4Pr16xbHixBCQiEW28+cKw7oQJeX0VWVtKqMaN9JNOB5\nMFnvM2XKFIIgzpw5w1mYXDLrLCctLa3B5fUrCGJjY2UyWXp6+pw5c2JjY7dt22b8bGFh4bhx49at\nWzd27NitW7eOHj368ePHTI9qhBBqeYg2CvHg4aI+YVBZCTI5oXQ1vb5arU5MTBwyZMi3335rYuYC\nATOnK46ZryVJUi6XX758mabpK1euKBSKOu2/zp8/365dO+axWq0GgCdPntR5E+wkbRHsscYjAe85\nlZVV91OpgjxKr+M7FGu01B5rNE1v37590KBBV69elUqlhh2sfgfo+kuMG0vTfzalZhbu37+/bdu2\nqamp9RtO13+ft95667333mPe5Nq1a+7u7lqt1qLfyDRzG94YVFRUnDx5cuDAgfUbIahUKrVazZz6\nBAYGqlSq8vJy4xV69eql1+u3b9+en5+/du3a4OBg5t5ahBBHSJJMvqDd/F/RqWO6fbt08SvpJ9l8\nx4T+8s0338ycObNv374BAQHfffcds7B+B+j6S0y8508//XT37l0/P7/6Dafrv8/o0aOPHj3KJK0D\nBw5MmDDB3t6exV/Q3LaeBlKpdMSIEeXl5dOnT68z2lhWVgYAzKV4qVQKACUlJQqFwrCCQqFYuXLl\nm2++CQAEQVy7ds1wD+P69euXLFkCAKNHj+7WrRvd/FPdGWi1WoHeSgkAzBcQgcZP0zTTkpzvQKwk\nxD2HuHZZdPcm3UZBURSIRODsot2zg3rjLVoqpHJNiqJIkuRyirnmm5HS2KNHj+7cufPaa68RBDF1\n6tRvv/2WKQzeuXPnqlWr/P39AWDXrl2VlZVvv/12nSUm3nbp0qUeHh61tbU3btxo3769i4uLj48P\nAJSWltZ/50GDBhUWFqampgYHBx84cGD37t3s/o4WpxyGr69vSkpKnYVMdtFoNHK5nPkI6tQa/PTT\nTytWrDh16lTv3r03bdo0fvz49PR05p/29ddfj4yMBIC0tDSJRMLlHek0TTs7O3O2OdZh9wG+CG/P\nIcnaKxdFHm2Zn5imG9BGYZfzWNynP5+BWYj77gPsftNvzLfffltaWmrYqUQiUV5enpeXV/0O0PWX\nGF9xr/M1jhlMarDhdP33AYDIyMijR48yg97PPvssu7+jWQNraX937dq1ZcuWtW/fvs5qSqVSLpcz\nLUgzMjLql1OfOHEiMjJy2LBhnp6eCxYsyMzMzMvLY55SKBQdO3bs2LGjXC5n4ddCCNVD11QTdnW/\nZdJ2dnSFmpd4kDGapnft2sUMajH69++/Z88eaKgDdP0lzGO9Xg8AhqcYTIlWgw2nG3yfMWPGHD16\n9MCBA5MnT2b966xZKSf478LCwjIzM7/66qu67yUSRUdHb9iwoaamZuPGjePHj2fCPXToEFMsMGDA\ngGPHjv3000+FhYXx8fE+Pj7t2rVj9/dBCDWGcHQEiqy7kKQIiaDO1VqoS5cu5efnR0REGJaMHTv2\n22+/hYY6QNdf4urqCgAJCQm1tbWff/55/fdvsOF0/fcBgOHDh9++fXvXrl2TJk1i/dc0a2DN/NH2\nuLi4yZMne3t7DxgwICEhgVkYHR2dmpoql8ujo6Nzc3NnzZqVl5cXEhLy3XffYYU0QtyxsxeHPUvd\n/w3sHQzLaHW5qFMgj0G1JFqtNjs7+8mTJ4aX76MAACAASURBVOa/hKIoZmx5165do0aNMh4tfO21\n1xYuXHjnzp36HaApiqqzxMHBYdWqVe+9997y5ctXr1795Zdf1tlQgw2nHzx4UOd9AEAmkw0ePPjx\n48c9e7Lf49isTtJc4qaTtDFBt9TFTtI8EuKeQ9dUkxfPU/duUWJ7EQGg0dhFvSYKDOY7LsvYbCfp\n+/fvOzo6Si3pZ1NRUaHX61mffLOJpk2bFhgYyPpkOWDiLKex2z+NCbqZNEKtEOEksRs8nO4RUv0k\nx6FNG8LbF0fVWCQWi2Uy2V+TF5iBoqjmmAXGajqdLi8v74cffli2bFlzvH+jKSc4+OlffGztDAkh\nZA7C04uWykVCO0VDHEhOTp4/f358fLyfn19zvH+jKQfTCUIItTbPP/98s3a/tPLqfU1NzePHj9kN\nBSGEWh49Td6uyPi59Oq9ygfG00LWl5OTM3XqVF9fX4lE0rlz548++sj0PZ7mSEtLM1zrJQiCuWJi\nvNCc9Vlk7q2gFRUVxmUYZ8+eXbRoUYO9ohFCCDEyNDk7806cKr7sKLavJmsj3Qe85TvG17Ft/TVp\nmo6MjHzmmWcuXbrk4eGRlpY2b9682bNnM3XSVvP29j548KA5C5/6FCvMOss5cOCAUqk0vjVnzpw5\nb7/9dvOFhRBCQqeldcNvzr+uvu/l6OZqJ/dx9EhR//ZV9iEKGmjVk5ube/v27WXLlvn7+0skktDQ\n0K+++ookSQCgafr//u//OnXqJJfLJ0yYUFpaCn+ejuzevTsyMtLT0/Pjjz9m3ufIkSPdunWTSCTB\nwcGnTp3Kzc2Njo4GgLCwMAAYPHjw5cuXDQsjIyOXL1/OvDAtLc3JyenevXv11581axYzRScAXL9+\n3cPDw+oOQGalnKVLl06bNk2tVoeFhd2+ffvRo0e9evV6ar0gQgi1ZtfVaW52Mjvir8OsA2H/Y8mV\n1MpH9Vd2d3f39/d//fXXjx8/XlFRAQA9evRgug/s379/y5YtJ0+efPjwIQBMnTrV8KqcnJwTJ04c\nOnRoxYoVT548UavVEyZM+Pe//11QUDBlypQZM2YY1rx69SoAnDlz5plnnjEsHD9+fGJiIvP44MGD\nUVFRhpYxxuuz2OvTrJSTmZkZGRkpk8kiIiJu3LgREBDw4YcfLlq0yLpNIoRQa1CiVTuKHeosdCIc\ninXl9Vd2dHS8c+fOkCFDVq1a1a5du759+27evJk5mdi6deuSJUu6du3q5ua2bt2606dPG1qaTps2\nDQAGDhwIABUVFbW1tRRFqVQqe3v7Dz/8kJm8wIRXX3313r17jx49AoCDBw/+4x//aHC1l19+men1\nSdM00wjHos/BmFkpx8XFpbCwEABCQkIuXLgAAP7+/jinJ0IImeDmIK8ltXUW1tBad/s29VemKMrB\nwWHhwoVJSUklJSX/+c9/Nm3a9NprrwHAgwcPJk6cSBAEQRDt2rWrra1lDsgA4O7uDgCGC/4eHh7H\njx8/cOCAq6sr0zfZdIQKhSIiIuK7775LS0vLz88fNmxYg6s5ODgwvT6vX7/exF6fZqWcsLCwtWvX\nXr16NTQ09MSJE3l5eWfOnGF+VYQQQg3qKw8q0Vfo6b+u3Ogo/TC38GBp+/orq9VqiUTCNNl0cnJ6\n9dVX4+Pjz507BwAeHh7ff/890+tTr9fn5eV5enoyr6pTeFZTU9OuXbvz588/evQoICBg5MiRTw1y\nwoQJiYmJBw8ejImJMTFcxlavT7NSzurVq1Uq1eHDhzt37hwTE+Pj47N8+fLPPvvM6q0ihFCL50DY\nn+q9tq+8a15tSale/aS2qH+bbv/yGydq6MCrUCiGDh36+uuvp6SklJSU3Lt3b926dUOGDAGAcePG\nrVix4vHjx6WlpfPmzRs3blxjB32dTvfss8+eOXPGxcXFy8vLwaHusF79quuoqKhr167t2LGjwVE1\nw/ps9fo0q0i6d+/eOTk5zBWtuLi4RYsWOTo6NlN3KZqmSZJk6jS4wUz3xNnm2MUEL9weay3gw+c7\nCisJOnhmfjaOjxLWvbCzs+/SjtPHtx1cqCvzcnDrJu1AQKP/rXv27Pn4449fe+21wsJCLy+vqKgo\npuvM/PnzS0tLBwwYoFarX3zxxb179zb2DjKZbOPGjdOmTSsoKAgMDPzmm2+Mn42JiRk8ePAPP/zA\ntJ1myOXyoUOHZmRk9O7du867GdYfMGAAW70+zWrruX79+ilTptSZ/KaZXLhwITQ0lMtWjxqNRmAT\nbRmprq52cnISbsrRarVcNmdkl6D3HEEHT5KkXq93dHTkbIsXL14sLy9/apluRkaGUqm0qMdaYWFh\nVVWVYZ40W8ZKr0+zznJiY2Pff//96OjoGTNmDBo0qFkPcARBiMXiP+Yr5IRIJOJyc+xighdoyiEI\nogV8+HxHYSVBBw8AFEVxfJTgbFs2iMVen2Z9jnl5eevWrcvKynrhhReCgoLi4uIM9RIIIYRatuTk\n5DFjxrDS69OssxxXV9eZM2fOnDkzKytrz54933zzzaJFi0aPHn3gwIEmbh4hhFoMkiSLi4uZy95m\nqqmpsas3O7itYbHXp2W/qr+//+jRo7Va7fr165u1Dw9CCAkOSZIikciiWgOm7rn5QrI15qacjIyM\nAwcOHDhw4NatW507d37nnXdiYmKaNTKEEBIWBwcH68oHmi8kW2NWyunXr9/169fd3NwmTJiwadOm\n8PBwgV6vRgghxCOzUk7Hjh3/85//DB8+vP6NRQghhJCZzJ28ICoqCvMNQghZp9aM6zXNPUXbm2++\n6e7ufuHCBQFM0YYQQshStXq4lA4/3QMxASQFI0MgvBPYN3RDEQdTtG3fvv3atWt9+/ZVq9U2PUUb\nQgghS9E0nL4DKQ/AUw7uMvBsAxd+h7O/NbyyiSnamDOPjRs3hoeHd+nSZe7cuTU1NQ3O26bT6T76\n6CMfHx9XV9dZs2bpdDrDbGzMNDlRUVFJSUmGhSbW53OKNoQQQpbKK4frj8DJqDuzswMkpUNZQxVq\nJqZoY6SkpCQnJ9+6devmzZurV69ucN62uLi406dPJycnX7p06dy5c1u2bDG8/PLlywBw5swZZnId\nhon1m2mKNhxYQwihZlFcAY71jsxOdlBUAcp6XZGZKdo2bty4atWqCRMmBAUFzZw5c9q0aYaD+7//\n/W+RSOTs7Pz++++/9957fn5+zLxtALBu3bqAgACKonbu3Llq1Sp/f38A2LVr11MvBZm5vmGKtuDg\n4AMHDuzevdvSj8KA5bMclUoVGRmpUCgiIyNVKlWdZzdt2kT83cSJE9kNACGEbIS9GOrfFUrRDV/L\nMTFFG8PX19fwIDs7u8F527KysgwdQsPDwwcPHmw6QjPX53qKNvPFxsbKZLL09HSZTGYY+zOYMmVK\n9p+ysrJCQ0NnzpzJbgAIIWQj/NygqhbqNOuvqgVvRQMrm5iijZGZmck8SEtL8/b2bnDeNi8vL+Yd\nAODq1auHDx82HaH563M6RZuZKIo6ePDg/PnzPTw8FixYcPjw4TozI0ilUt8/nTlzJiIi4uWXX2Yx\nAIQQsh1SR5j8LBRVQK0eKApqdVCghjefb2C0DUxO0caIjY3Nz8+/d+/e0qVLJ0+e3OC8bZMnT/7k\nk0+Yc6CZM2caslRjnro+P1O0mUmlUqnV6qCgIAAIDAxUqVTl5eUKRQMJvbS0dO3atZcuXTIsKSgo\nYJJtUVERiyEhhBCPevjC+yPht1xQV0MbCfT0BWnj80M1NkUbIyoqqn///tXV1RMmTFi0aJGdnV39\nedsWL15cUVHRr18/vV4/fvz4efPmPXjwwER4ptfnbYo2M2VmZnbu3Fmn09nZ2en1ent7+4yMjE6d\nOtVf86233goNDZ01a5ZhydatW1etWgUAERERn376KZfTdtXW1nI51xO7tFqtvb29QPsPMQMCVpe+\n8E7Qe46gg2emBOVyz7ly5UplZSVfU7SlpaUFBwezeKy2DndTtJmJOaHRaDRyuZw5HWtwItG8vLxD\nhw7Fx8cbL5wxY8aMGTMAICkpSSKRcDkrKE3Twp0eEQAkEolAUw5FUbW1tVz+rdkl6D1H0MGTJKnT\n6bj8YircL0as4HqKNjMplUq5XJ6RkQEAGRkZcrm8wZTz9ddfjx071sWlXpEgQggh28PiFG1sphyR\nSBQdHb1hw4aampqNGzeOHz+e+fZ96NAhtVptWO3QoUMjRoxgcbsIIWQj9Hp9rSXMmSwnKCiI31E1\nZoq2JhYOMFi+FTQuLm7y5Mne3t4DBgxISEhgFkZHR6empsrlcgDIy8u7devWc889x+52EUKId/b2\n9nl5eXl5eRa9yt3dvZnisUEspxyFQnHy5Mk6C43zs5eXF+8XwRBCqDkEBAQEBATwHYVNwx5rCCGE\nOIIpByGEEEcw5SCEEOIIphyEEEIcwZSDEEKII5hyEEIIcQRTDkIIIY5gykEIIcQRTDkIIYQ4gikH\nIYQQRzDlIIQQ4gjLPdaajpm2y5zuqmyhKIrLzbGLCV648+WQJCn0D5/vKKwk9OA53nMoiuJsWy2b\nzaUcABCLxWKxmLPNEQTB5ebYxQQv0JRDEIRIJBL6h893FFYSdPAAwPGeI9B/MRtkcymH+BPHW+Rs\nc+zi/uNikaCDhxax5/AdhZV4OUpwtq2WDa/lIIQQ4gimHIQQQhzBlIMQQogjmHIQQghxBFMOQggh\njmDKQQghxBFMOQghhDiCKQchhBBHMOUghBDiCKYchBBCHMGUgxBCiCMspxyVShUZGalQKCIjI1Uq\nVf0VqqurX3/9daVSGRYW9vvvv7O7dYQQQraM5ZQTGxsrk8nS09NlMllsbGz9FRYvXqzRaNLS0sLC\nwt555x12t44QQsiWsdlJmqKogwcPnj592sPDY8GCBcOGDdu6datxB1aapnft2nX69GlPT8+VK1de\nuXKFxa0jhBCycWymHJVKpVarg4KCACAwMFClUpWXlysUCsMK5eXlxcXF+/fvHzx4cKdOnb7++mvD\nUykpKefOnQMAZ2fn3r17kyTJYmCmMdM9cbY5djHBC7SzOkVRLeDD5zsKKwk6eJIkOY4fp2hjC5sp\np6ysDABcXFwAQCqVAkBJSYlxyikpKQEAvV7/8OHDNWvWTJ48+c6dO8zhMj8///r16wAQFBSk0+m4\nnHyJJEmtVsvZ5til1+u1Wq1AUw4zA6xwP3yB7jk5tYVHSs6X1Kgc7BxCpJ2HK5+xIwQ2Vxv337SE\nm55tDUHTNFvvVVJS4u7uXl5eLpfLVSqVUqksKSlxdXU1rFBUVNS2bdvS0lKlUllaWurm5paXl9eu\nXTvjN0lKSurbt69EImErqqeqqqpi0qQQaTQaiUQi0JRDUVRtbS2Xf2t2CXHPuaZOnXJnaYDES0wT\nYrG4RFce4Rq+uOM/hZV1SJLU6XROTk6cbTEpKamsrGzUqFGcbbGlYrN8QKlUyuXyjIwMAMjIyJDL\n5Uql0ngFV1dXFxcXnU4Hf56ocrnTIIQOF57r7OLrIPpjeMPNvs2pkuRzZdf5jQq1HmymHJFIFB0d\nvWHDhpqamo0bN44fP5759n3o0CG1Wg0AYrF47Nixn3zyiUqlWrly5cCBA42H3RBCzapMV/F90QUC\n/nZOrLCX36t8wFdIqLVhuUg6Li4uNzfX29u7oKBgzZo1zMLo6Ojc3Fzm8dq1a9PT0/39/W/evLlr\n1y52t44Qepq6Y7AEsDa0jtBTsVk+AAAKheLkyZN1FhpfLnJzc/vxxx/Z3ShCyBxKe5mW1tFAG5/o\nqPRV3aUdeYwKtSrY8AahViShx7IHmlw9/Uf9Vbm+cohrv5eUffmNCrUemHIQakX6y7slhqx6Xtk7\nvTq7rzxops/of3cQWLkaEjSWB9YQQjauq0vARy5vzPUYJ7gKb9QC4FkOQgghjtjiWc6tW7fs7e05\n21xtba2joyNnm2OXVqu1t7cX6K2gTPcBLv/W7BL0niPo4JnuA1zuOU+ePHF2duZscy2YzaUcd3f3\nO3fucLY5iqLefffdNWvWCPSm1M2bN4eGhvbv35/vQKyRnJx879696dOn8x2INaqqqj766KN169bx\nHYiV1q5dO2TIkJ49e/IdiDXOnDmTn58fExPD2RZFIlH79u0521wLxmbDGyFivmUzTXr4jsUao0aN\nio6O/sc//sF3INbYtm3bDz/8cPjwYb4DsUZxcbGnp6dwW28NGjTo/fffj4qK4jsQa8TFxaWmphr3\nBUZCgddyEEIIcaS1n+XQNP3w4cP27duLRILMvvn5+VKplOnbLTgVFRUajcbT05PvQKxBkuTjx487\ndhTqTZS5ubkKhUKg1yfKy8u1Wq2HhwffgSCLtfaUgxBCiDOC/GqPEEJIiDDlINTqVFdXc1kXipAB\nphyEWp24uLgZM2YIdFA9JSVl4MCBXbp0WbVqlUB/hdbM5u7LQQg1q5ycnNWrV58+fZogCK1Wm5yc\nLJPJQkNDBXFDcVFRUVRU1NKlS728vObOnRscHPzqq6/yHRSyAJYPINS6TJkyBQASEhKePHny0ksv\nlZSUVFVV9e/f//Dhw7ZfA7Zp06Zff/11z549ALB+/fqEhITq6mp/f//4+PjAwEC+o0NPhwNrDSgq\nKuI7hCYRdPy2H7yJCyG2H3x2dvbevXsHDhwIALNnzx4zZkxhYWFeXp6bm9vcuXP5ju7pSJLMy8tj\n7sBNTk5WKpXr16/38vIaMWKEXq/nOzpkBrq1unv3blRUVEhIyJIlS7RarWF5aWmpTCabO3eu8UJb\nc+nSpb59+7Zv337lypUURRk/Zfvxx8fHz58/v6ysrP5Tth88TdOffPJJeHh4nY+dFkjwhYWFp06d\ncnNzW7RokY+Pj0ajYZbn5OS4urryG5s5iouLO3bs2Lt370GDBgFARkYGTdN6vV4qlebk5PAdHXq6\nVnqWk5+fHxERERYW9v777+/YsWP58uWGp5RK5aVLl1QqVWxsLI8RmlBYWPjaa6/Nnj177dq127dv\n379/v/Gzth9/hw4d1q5d26VLly1btjBfV3U6HfOU7QfPXAiJj49nLoScP3/+xo0bNE1DI8Hb1HmP\nRqPp16/f3bt3r169+sMPPxQUFBg6Y+bm5rZr147f8Mzh5uZ2/fr1999/f+XKlT4+Pn5+fgBw9+5d\nmUwm0HuKWx2+cx4/li1bNmfOHObx6dOn+/XrV2cFiqIePnzIdVjmiY+PZ8qNaJrevHlznz59fH19\nIyIiMjMzDevYcvwkSQYFBS1btszHx6dXr17nzp0bOXIkc+Bm2HLwMTExMTExNE3n5OR06dLF1dXV\n0dFx0KBBhYWFzArGwVdVVfn7+8fFxfEVbR3Lli3r3bt3t27damtrNRrN7t27aZq+f//+Dz/80LFj\nx82bN/MdoGWGDRs2cuTINWvW+Pr67tmzh+9wkFla6VnOgwcPDFdK+/Tp8/vvvxuWT5kyhaIogiBs\ntnFsRUVFRUUF8/iXX35p167dkSNH/Pz8IiMjDV0mbTl+kUi0YMGCvLy8+/fvR0VFjRgx4sqVK0ql\n0rCCIfhLly6NGjVqzJgxNTU1vIVrxJwLIcaffFxcnKur6/bt27VaLV8xGzDnZ19++eXo0aO3bt0q\nkUgmT55MkuS0adNmz549Y8aMGTNm8B2jZRISEjp16pSSkrJly5ZJkybxHQ4yD985j2vMt9HExESF\nQpGcnEzTdHFxsWEUe8yYMTNnzuQzvqcpLCzMyspSKpVbt26laXr37t2lpaU0Tet0OqlUaviubZsM\n4VVXVwcEBOTn59fW1nbv3v2VV15xcnI6evSo8cqVlZVubm4rVqxITEwsKytLTEy8cuVK/SsonGGC\nN/9CSHZ2trOzc1JS0qJFi7788kuuw61n8uTJkydPpmm6sLCwffv2eXl5fEeEWqPWlXIMAx0URX31\n1VcXLlxgFjo6OpIk+fPPP8vl8oKCAnPeipeDuyH+jIyMR48eGT9169YtX19fvV5vWKLRaG7fvt3Y\nW3Eff51Rpk8//XTRokXx8fHvvvsuTdO//PJLnZDOnz8fFhZG0/T9+/c9PDx69uwplUoHDhzIy7HS\nOPgHDx6Ehoba2dnpdDrm2ZSUlG7dutV5iU0d4i9evOji4pKdnc38uHHjRqbEi8eQUOvUulKO8Vi2\nYSFFUV5eXikpKT169Pjiiy+M12+sMIyv2qQG4//iiy/i4uL8/f137txpvHJjhVU0T/HXCb60tNTX\n17dTp07MWVpjQTLXilesWEHTdHl5+bhx44YMGcJZzAZ1gn/qhRBbO8SXlpaePn3a8CNJkkOHDl24\ncCFf8Viksc/twYMHly9f5jgY1EStKOWYGOhYsWKFTCYLDAw0PpQXFBR4eXlt27YtMTGxc+fOe/fu\nNX7JnTt3/vGPf8ydO5ej6BuJv6amJjY2dty4cfv27auzsouLy8WLF2marq2t/eWXX65fv26cfjiO\nv8Hgjxw5kpCQ0OD6zBnPjh072rRpM2TIEMPF4YqKCjs7u6qqKm7CZjS25+j1+ueee659+/afffaZ\n8WdbWFho+4f4oqIiPz+/JUuW8DhWaQ6dTtejR49t27bVTzw7d+7s2bMnSZK8BIas04pSjomBDr1e\n/9577x0/ftx4/acWhnFcWGXRQM1TC6vov8ff3ONsFgVvPIr1/fffe3h4+Pv7M+nz4sWL3t7eHB9l\nrA7emA0e4lNTUz09Pb///nu+A3mK+Ph4AAgNDf3111/rPGX8HREJQmtJOVYMdCxbtmzixInM40mT\nJo0cOTIlJeXNN98MCgrifoTEovizsrJEItHGjRtpmh41atTChQv1en1paeno0aMNv5Gx5q7ltfTD\nrz8Et3z5cl9fX4VCoVAojM8eONDE4I3Z4CE+LS3N9o/aq1evDg8PX7RokaOj44QJE+pcxUTC0lpS\njhUDHTZVGGZp/BbdYW7iKMkKi4I3Mf5ZXFzMfbJnK3iGIA7xtmbx4sXMNZvMzMyxY8c6OTktWbKk\nsrKS77iQNVpLyqnPxECHIaOYUxjGl6fGb2ZhFS+1vCaCt6lCrwYJOvgW4MyZMz179uzevTtexRGi\n1pty6EYGOhobZWqsMMyE5j4Zemr85txhztdRssHgba3QqzGCCN7EpoVY61VdXW14rNPpbt26xWMw\nyGqtOuXQDQ10NDjK1FhhGN14k0puCpHNib+xwiqa76Nk/eBtv9DLwMaDN1HoRQuw1qugoMDPz6+x\nenokIK095dRhxShTYmIiALi7u2/evJn59zbkGBsppDbh/v37tnOUbJANFnqZj9/gTRR60UKr9Zo+\nfTpzyzASuhaeciy9icyKUSbTTSqbUkhtxcBI08t52TpKsjiqw0uhF1u3H/JYpSbQQq/6n7xGoxk4\ncCCe4rQMLTnlWHoTmdWjTFu2bJk1a1ZlZeXixYudnJzc3NwaTDOmO9CYHzxb8TdWqNb0oyTrozoc\nF3qxe/shX1VqQiz0Mr3noBagJacc2sKbyCwaizcuDTCnSSVtsgONpcE3PX7TQ3BNP0oKfVSn5d1+\nKJRCL9N7DhI68dKlS3nrYt38Lly4UFFRERERMXv27Lt374aFhSkUCuYpsVhcZ2WJRNKpUyfDjwRB\njBgxYs6cOeXl5S+88AJBEIanNBpNSEgIRVEDBgwAADs7O41Gk5SUdO/ePTc3t127dg0aNKhPnz4u\nLi6Gl+Tk5EyaNCkhIcHf31+r1SYlJZWWlrZr1874bc0PvonxA8Ds2bODgoLmzZvXq1evf/7znxMn\nTpRKpYZn3d3d67+/RSwN3tZYtOfYspqaGjs7OwDo0KHDjBkznn/+eRufis30noMEj++c17yaPrbQ\n4ChT/SGppzapNKcDDevBNxY/B4Vq5gdvm7W8TY/fFgqRhVXoxfwvCHE8EJmvhaecOqwbW6gzytTY\nkJSJJpWWdqBhMfj68dN8lPM2FrxQanmtiN8WghdQoVdj98MJZTwQmal1pRyajZvILK1qs3RqLxOa\n7w44Dsp5GwteKFd9rIif4+DrpD1hFXqZ6LqEN362JK0i5Rjft1yfRQMjlg5JWTG1FyvBNxa/Cc1R\nzms6eIYt1/IKKH5B13rZ2gyqqPm0/JRjejjb0oERS4ekLJ3ai63gG4vfNHbLec28kGCzY/eCi1+4\ntV7Ym671aPkp56nD2U0cGDExJGXp1F7sBm9m/M3HugsJtjN2L7j4beR8y1K21psONasWlXLu3r37\nzjvv/Otf/9q7dy/Tdcac4eym/6M2NiRl6Xe3OvFzEzzNUm2VFcEb47dpo3V7jjEe4zcUPdrO+ZZF\nbKo3HWpuLSfllJSUeHl5xcbGLly4sEOHDv7+/t9++605F8NZ+UetPyRl6Xc36+JnpRa56bVVVn/4\nDH5reZsYPM1r/Cam17Od80VLCbqxHjKt5aScY8eOPf/888xjnU63bds2V1fXSZMmWdrIma1/VEu/\nu7ESv3W1yHSTh+CaGDy/tbxN/+R5jN/09HrCrfWywRlUEStaTsp59OiRs7NzSkqKYUl+fn5YWJgV\njZyb6R/V9Hc3tuK3rha5iSwK/u7du1FRUSEhIUuWLLFiCI51TQme5rUWuWUXeuEMqi1SC0k5zHD2\nV199JZPJvv76a8MxPS8vT6FQmPN1lcVCahMa/O5mGIu3On4ea3ktDT4vL8/b23v58uVM75+PP/6Y\nlTCsZtGeY2vBY6EXEpyWkHKMh7OPHj3q4eERFhaWmJio0Whu3rzp7+//1BFhdgupTavz3a3OWLwV\n8fNYy2tF8MuWLZszZw7z+PTp0/369WtKAE1k6Z5jU8FjoRcSopaQcuoMZ5eVla1cubJ9+/YEQbi5\nuRlfUGlMcxdSmx88bXn8PNbyWhH8G2+88cknnzCPi4uL5XI58zgzMzMmJobjC92W7jk2FbygC71s\ns7Ee4oDgU46J4WyVSlX/KGDdBFDmj0pZNARneiy+wfibeCGExVpeS4NnhrASExMVCkVycjJN08XF\nxYZ+P2PGjJk5c6bVwVjB0j2nsLDQzC2+0wAABmNJREFUdoJvkFAKvYTSWA81B8GnHIuGs61uCmLm\nqJSlQ3CWjsU38VoCu7W81s1ASlHUV199deHCBWaho6MjSZI///yzXC4vKChgJTAzWRH/mjVrbCT4\nxgil0MuWb2FGzUrYKceK4WxWCrdMjEqZPwRnRfBNvJbAYi0vKzOQUhTl5eWVkpLSo0ePL774gpXA\nzNT0+HkM3jRBFHoJtFECajphpxwrhrObsq+bMypl/vtbEbxF1xLi4+Pnz59fVlbG/MhuLS9bM5Cu\nWLFCJpMFBgZyfJRkJX6+gm8BBNooATWdsFNOfSaGs5s4ARQHhWFPDd6iawmJiYkA4O7uvnnzZub7\nu6V3xVrERPAmhrD0ev177713/Pjx5gvMTFbEbzvBC51wGyUgS7W0lEM3Mpzd9AmguCkMMx28RRdC\nSJIMCgpatmyZj49Pr169zp07N3LkyBs3blj6K5iPrxlI2SKI+G15EtKmEG6jBGSRFphy6IaGs5s4\nAVRTRqUs/V8yM3hzriVs2bJl1qxZlZWVixcvdnJycnNze/jwoRW/QlOCF1Ytr43Hb+OTkCL0VC0z\n5dRhaV+Q+k1NLGVOOwAzWXohxNAOgAkjICAgPz+/tra2e/fur7zyipOT09GjR9mKzTpCqeVtDL/x\n284kpAhZQXgpx4qbyCwqh216UxPTV30sHRix6EJI/fHDTz/9dNGiRfHx8cyo4C+//GKckyzVrI1/\nmpuNz6BqJqz1QoImsJRjxU1kVpTDNrGpiYmrPpYOjDS9lre0tNTX17dTp05Nr1Vr1sY/zc3GZ1A1\nH9Z6IUETWMqhLb+JzNKxeIsKkS9dutS3b9/27duvXLmSGWZ56lUfiwZGWKnlPXLkSEJCQmPxWETQ\nozot8vZDrPVCwiK8lNP0gQXTtcjmFyIXFBR4eXlt27YtMTGxc+fOe/fu5SB+62qR2SLoUR1BB28C\n1nohARFeymFlYMFELbL5TU3i4+NnzJjBPN68eXOfPn18fX0jIiIyMzObNX4ea3mbPgkpj7W8rMyg\nKvRaZIT4JbyUU4fVAwvm1CKbLkRetmzZxIkTmceTJk0aOXJkSkrKm2++GRQUZP6x3rr4baSW14pJ\nSG2nlte6GVRtJ36EhEjwKYdmaWDB0qYmhYWFWVlZSqVy69atNE3v3r2buX6j0+mkUqlFVWG8TELK\nFismIbWdqybWzaBqO/EjJDgtIeXUx2IhdYNNTQy1yBkZGXUuCdy6dcvX17eJI1qCrkVmCPrCiaCD\nR8iWtcCUw00hdf12AF988UVcXJy/v//OnTubKX6s5eWGoINHyJa1wJRDN3MhdYNDcDU1NbGxsePG\njdu3b1+zxi/QUR1B1/IKOniEbErLTDnNWkiNtcjWEXQtr6CDR8h2iKAlqqio+O9///vpp5/+9ttv\nOp0uKCho6dKlVVVV5r+Du7v76dOnN23adOLECePlly5dOnr06OrVqwHAw8Pjgw8+mDZtGkmSfMXf\n2KYfPnx45coVdqNqIjs7u169evEdhZUEHTxCtoOgaZrvGJrd2bNn582bR1HU7du3RSILsuz9+/c7\ndOjg4OBgWFJWVnbt2rWhQ4cyP1IUNXz48NDQUCYJNZPG4tfr9aGhofPmzZs6dapYLDZ+yTfffPPF\nF1/cvHnTot8XIYSaVatIOQCg1+t/++235viiWlxc3KdPn2nTpi1ZsoQgCNbfn9FY/GvXrl2wYEFo\naOh///vfQYMGGT+l1WqNkyVCCPGutXwFrj8wYmI0zKJRqcaG4NjV2MCOTqcLDw8fMWLE0KFDJ06c\n+PjxY8NTmG8QQramtaScOvR6fe/evb/++usGE8+vv/46Y8YMiqLMfLegoKDz589HRESwGqNZmn7V\nCiGEONNaBtbqMzEkBYIdlbL6qhVCCHGg9R6VTAxJgdmjUrZWGPbyyy/fuHFjz549mG8QQjao9R6Y\nml6IDJYPwXEAy3kRQjar9Q6s1WFFITJDoENwCCHEvdZ7llNHY0NSdnZ206ZNmz59elhY2IULF+q/\nEPMNQgiZCVPOX6woREYIIWQ+TDlPh4XICCHECryWYzEsREYIIetgyrFG87XPQQihFgxTDkIIIY7g\nuBBCCCGOYMpBCCHEEUw5CCGEOIIpByGEEEcw5SCEEOIIphyEEEIcwZSDEEKII5hyEEIIcQRTDkII\nIY5gykEIIcSR/wdrOBlJgXOXDgAAAABJRU5ErkJggg==\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -h 250 -w 550\n",
    "df.f = df %>%\n",
    "    filter(variables %in% c('Sensitivity', 'Specificity', 'Balanced Accuracy')) %>%\n",
    "    mutate(variables = gsub(' ', '\\n', variables))\n",
    "\n",
    "ggplot(df.f, aes(BD_window, values, color=variables)) +\n",
    "    geom_point(size=3, alpha=0.7) +\n",
    "    theme_bw() +\n",
    "    theme(\n",
    "        text = element_text(size=16),\n",
    "        axis.title.x = element_blank(),\n",
    "        axis.text.x = element_text(angle=60, hjust=1)\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%pushnote validation MW-HR-SIP complete"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "hide_input": true,
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
